From d0c9292e86362d00518cdcded043cff3f1d2711b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 6 May 2020 15:55:11 -0300
Subject: [PATCH] bc: add missing header

---
 bc/module.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bc/module.hpp b/bc/module.hpp
index d50cff0..5a35310 100644
--- a/bc/module.hpp
+++ b/bc/module.hpp
@@ -25,6 +25,7 @@
 #include <unordered_map>
 #include <utility>
 #include <vector>
+#include <string>
 
 // A reasonably small LLVM C++ API lookalike.
 
-- 
2.26.2

