diff -rupN --no-dereference mingw-w64-v9.0.0/mingw-w64-tools/widl/include/basetsd.h mingw-w64-v9.0.0-new/mingw-w64-tools/widl/include/basetsd.h
--- mingw-w64-v9.0.0/mingw-w64-tools/widl/include/basetsd.h	2021-05-22 09:34:25.000000000 -0300
+++ mingw-w64-v9.0.0-new/mingw-w64-tools/widl/include/basetsd.h	2021-05-24 18:20:59.552628797 -0300
@@ -36,7 +36,7 @@ extern "C" {
  * 64-bit.
  */
 
-#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__aarch64__)) && !defined(_WIN64)
+#if (defined(__x86_64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(__s390x__)) && !defined(_WIN64)
 #define _WIN64
 #endif
 
@@ -295,6 +295,8 @@ typedef ULONG_PTR KAFFINITY, *PKAFFINITY
 # define WORDS_BIGENDIAN
 #elif defined(__MIPSEL__)
 # undef  WORDS_BIGENDIAN
+#elif defined(__s390__)
+# define WORDS_BIGENDIAN
 #elif !defined(RC_INVOKED) && !defined(__WIDL__) && !defined(__midl)
 # error Unknown CPU architecture!
 #endif
diff -rupN --no-dereference mingw-w64-v9.0.0/mingw-w64-tools/widl/include/winnt.h mingw-w64-v9.0.0-new/mingw-w64-tools/widl/include/winnt.h
--- mingw-w64-v9.0.0/mingw-w64-tools/widl/include/winnt.h	2021-05-22 09:34:25.000000000 -0300
+++ mingw-w64-v9.0.0-new/mingw-w64-tools/widl/include/winnt.h	2021-05-24 18:21:30.740034653 -0300
@@ -2396,6 +2396,36 @@ typedef struct _STACK_FRAME_HEADER
 
 #endif  /* __powerpc__ */
 
+#ifdef __s390x__
+
+/*
+ * FIXME:
+ *
+ * There is no official CONTEXT structure defined for the S/390
+ * architecture, so I just made one up.
+ *
+ * This structure is completely dummy, made just to build widl.
+ *
+ */
+
+#define CONTEXT_S390X            0x10000000
+
+#define CONTEXT_CONTROL         (CONTEXT_S390X | 0x00000001)
+#define CONTEXT_FLOATING_POINT  (CONTEXT_S390X | 0x00000002)
+#define CONTEXT_INTEGER         (CONTEXT_S390X | 0x00000004)
+
+#define CONTEXT_FULL (CONTEXT_CONTROL | CONTEXT_FLOATING_POINT | CONTEXT_INTEGER)
+
+#define EXCEPTION_READ_FAULT    0
+#define EXCEPTION_WRITE_FAULT   1
+#define EXCEPTION_EXECUTE_FAULT 8
+
+typedef struct _CONTEXT
+{
+} CONTEXT, *PCONTEXT;
+
+#endif  /* __s390x__ */
+
 #if !defined(CONTEXT_FULL) && !defined(RC_INVOKED)
 #error You need to define a CONTEXT for your CPU
 #endif
diff -rupN --no-dereference mingw-w64-v9.0.0/mingw-w64-tools/widl/src/widl.c mingw-w64-v9.0.0-new/mingw-w64-tools/widl/src/widl.c
--- mingw-w64-v9.0.0/mingw-w64-tools/widl/src/widl.c	2021-05-22 09:34:25.000000000 -0300
+++ mingw-w64-v9.0.0-new/mingw-w64-tools/widl/src/widl.c	2021-05-24 18:23:09.814086829 -0300
@@ -106,6 +106,8 @@ enum target_cpu target_cpu = CPU_POWERPC
 enum target_cpu target_cpu = CPU_ARM;
 #elif defined(__aarch64__)
 enum target_cpu target_cpu = CPU_ARM64;
+#elif defined(__s390x__)
+enum target_cpu target_cpu = CPU_S390X;
 #else
 #error Unsupported CPU
 #endif
@@ -301,6 +303,7 @@ static void set_target( const char *targ
         { "armv7a",         CPU_ARM },
         { "arm64",          CPU_ARM64 },
         { "aarch64",        CPU_ARM64 },
+        { "s390x",          CPU_S390X },
     };
 
     unsigned int i;
@@ -787,6 +790,7 @@ int main(int argc,char *argv[])
       else pointer_size = 4;
       break;
   case CPU_x86_64:
+  case CPU_S390X:
       if (pointer_size == 4) target_cpu = CPU_x86;
       else pointer_size = 8;
       break;
diff -rupN --no-dereference mingw-w64-v9.0.0/mingw-w64-tools/widl/src/widl.h mingw-w64-v9.0.0-new/mingw-w64-tools/widl/src/widl.h
--- mingw-w64-v9.0.0/mingw-w64-tools/widl/src/widl.h	2021-05-22 09:34:25.000000000 -0300
+++ mingw-w64-v9.0.0-new/mingw-w64-tools/widl/src/widl.h	2021-05-24 18:23:50.564470727 -0300
@@ -76,7 +76,7 @@ extern int char_number;
 
 enum target_cpu
 {
-    CPU_x86, CPU_x86_64, CPU_POWERPC, CPU_ARM, CPU_ARM64, CPU_POWERPC64, CPU_LAST = CPU_POWERPC64
+    CPU_x86, CPU_x86_64, CPU_POWERPC, CPU_ARM, CPU_ARM64, CPU_POWERPC64, CPU_S390X, CPU_LAST = CPU_POWERPC64
 };
 
 extern enum target_cpu target_cpu;
