From 3def6da6b429cf6a3240759ce48cc424bbfd65cd Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 11 Oct 2021 18:12:25 -0300
Subject: [PATCH] Fix libgamemodeauto search name

---
 lutris/util/linux.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lutris/util/linux.py b/lutris/util/linux.py
index 2ac62a4..85d8808 100644
--- a/lutris/util/linux.py
+++ b/lutris/util/linux.py
@@ -74,7 +74,7 @@ SYSTEM_COMPONENTS = {
         "VULKAN": ["libvulkan.so.1"],
         "WINE": ["libsqlite3.so.0"],
         "RADEON": ["libvulkan_radeon.so"],
-        "GAMEMODE": ["libgamemodeauto.so"],
+        "GAMEMODE": ["libgamemodeauto.so.0"],
         "GNUTLS": ["libgnutls.so.30"],
     },
 }
-- 
2.33.0

