From f0ab6608ee09e6321bdcb900eeff4b7ab62a5b6e Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 12 Sep 2021 22:40:06 -0300
Subject: [PATCH] System data file

---
 desktop_version/src/FileSystemUtils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index 6caabdc..35b83ce 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -154,7 +154,7 @@ int FILESYSTEM_init(char *argvZero, char* baseDir, char *assetsPath)
     else
     {
         SDL_snprintf(output, sizeof(output), "%s%s",
-            basePath,
+            "_RPM_DATA_DIR_/VVVVVV/",
             "data.zip"
         );
     }
@@ -177,7 +177,7 @@ int FILESYSTEM_init(char *argvZero, char* baseDir, char *assetsPath)
         goto end;
     }
 
-    SDL_snprintf(output, sizeof(output), "%s%s", basePath, "gamecontrollerdb.txt");
+    SDL_snprintf(output, sizeof(output), "%s%s", "_RPM_DATA_DIR_/SDL_GameControllerDB/", "gamecontrollerdb.txt");
     if (SDL_GameControllerAddMappingsFromFile(output) < 0)
     {
         vlog_info("gamecontrollerdb.txt not found!");
-- 
2.31.1

