From 34d3cc0250c673f77e060f821c563c8eaf54a194 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 11 Jan 2020 19:11:33 -0300
Subject: [PATCH] System data file

---
 desktop_version/src/FileSystemUtils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index f8f415a..0ec4a5d 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -70,8 +70,8 @@ int FILESYSTEM_init(char *argvZero)
 	}
 
 	/* Mount the stock content last */
-	strcpy(output, PHYSFS_getBaseDir());
-	strcat(output, "data.zip");
+	strcpy(output, "_RPM_DATA_DIR_/");
+	strcat(output, "VVVVVV/data.zip");
 	if (!PHYSFS_mount(output, NULL, 1))
 	{
 		puts("Error: data.zip missing!");
-- 
2.24.1

