From b357fc4c94b8bb63cf5a711544c8be06232f1f0e Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 18 Jul 2020 17:07:26 -0300
Subject: [PATCH] System libraries

---
 desktop_version/CMakeLists.txt | 33 +++++++++++++++++++++++++++++----
 1 file changed, 29 insertions(+), 4 deletions(-)

diff --git a/desktop_version/CMakeLists.txt b/desktop_version/CMakeLists.txt
index 01dd28b..e122c32 100644
--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -3,9 +3,13 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
 
+find_package(PkgConfig REQUIRED)
+
 # CMake Options
 OPTION(ENABLE_WARNINGS "Enable compilation warnings" ON)
 OPTION(ENABLE_WERROR "Treat compilation warnings as errors" OFF)
+OPTION(USE_SYSTEM_PHYSFS "Dynamically link against system tinyxml" OFF)
+OPTION(USE_SYSTEM_TINYXML "Dynamically link against system tinyxml" OFF)
 
 SET(CUSTOM_LEVEL_SUPPORT ENABLED CACHE STRING "Optionally disable playing and/or editing of custom levels")
 SET_PROPERTY(CACHE CUSTOM_LEVEL_SUPPORT PROPERTY STRINGS ENABLED NO_EDITOR DISABLED)
@@ -68,12 +72,17 @@ ENDIF()
 # Include Directories
 INCLUDE_DIRECTORIES(
 	src
-	../third_party/tinyxml2
-	../third_party/physfs
 	../third_party/lodepng
 	../third_party/utfcpp/source
 )
 
+IF(NOT USE_SYSTEM_TINYXML)
+  INCLUDE_DIRECTORIES(../third_party/tinyxml2)
+ENDIF()
+IF(NOT USE_SYSTEM_PHYSFS)
+  INCLUDE_DIRECTORIES(../third_party/physfs)
+ENDIF()
+
 # Source Lists
 SET(VVV_SRC
 	src/BinaryBlob.cpp
@@ -186,8 +195,24 @@ SET_PROPERTY(TARGET VVVVVV PROPERTY CXX_STANDARD 98)
 SET_PROPERTY(TARGET VVVVVV PROPERTY CXX_EXTENSIONS FALSE)
 
 # Library information
-ADD_LIBRARY(tinyxml2-static STATIC ${XML2_SRC})
-ADD_LIBRARY(physfs-static STATIC ${PFS_SRC} ${PFSP_SRC})
+IF(USE_SYSTEM_TINYXML)
+  ADD_LIBRARY(tinyxml2-static INTERFACE IMPORTED GLOBAL)
+  pkg_check_modules(TINYXML2 REQUIRED tinyxml2)
+  target_include_directories(tinyxml2-static INTERFACE ${TINYXML2_INCLUDE_DIRS})
+  target_link_libraries(tinyxml2-static INTERFACE ${TINYXML2_LIBRARIES})
+ELSE()
+  ADD_LIBRARY(tinyxml2-static STATIC ${XML_SRC})
+ENDIF()
+
+IF(USE_SYSTEM_PHYSFS)
+  add_library(physfs-static INTERFACE IMPORTED GLOBAL)
+  pkg_check_modules(PHYSFS REQUIRED physfs)
+  target_include_directories(physfs-static INTERFACE ${PHYSFS_INCLUDE_DIRS})
+  target_link_libraries(physfs-static INTERFACE ${PHYSFS_LIBRARIES})
+ELSE()
+  ADD_LIBRARY(physfs-static STATIC ${PFS_SRC} ${PFSP_SRC})
+ENDIF()
+
 ADD_LIBRARY(lodepng-static STATIC ${PNG_SRC})
 
 # Static Dependencies
-- 
2.26.2

