From cdbdac6b68154ee628705a3f663b7ec42efd7d01 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 19 Apr 2021 10:41:10 -0300
Subject: [PATCH] System data file

---
 desktop_version/src/FileSystemUtils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index f7bd840..00364e4 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -159,7 +159,7 @@ int FILESYSTEM_init(char *argvZero, char* baseDir, char *assetsPath)
 	else
 	{
 		SDL_snprintf(output, sizeof(output), "%s%s",
-			basePath,
+			"_RPM_DATA_DIR_/VVVVVV/",
 			"data.zip"
 		);
 	}
@@ -182,7 +182,7 @@ int FILESYSTEM_init(char *argvZero, char* baseDir, char *assetsPath)
 		goto end;
 	}
 
-	SDL_snprintf(output, sizeof(output), "%s%s", basePath, "gamecontrollerdb.txt");
+	SDL_snprintf(output, sizeof(output), "%s%s", "_RPM_DATA_DIR_/SDL_GameControllerDB/", "gamecontrollerdb.txt");
 	if (SDL_GameControllerAddMappingsFromFile(output) < 0)
 	{
 		printf("gamecontrollerdb.txt not found!\n");
-- 
2.30.2

