From 6b712c2da577221c9d41a4d948b36945850a45da Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 2 Feb 2020 19:08:14 -0300
Subject: [PATCH] System data file

---
 desktop_version/src/FileSystemUtils.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index 2a1d2df..5e50f2c 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -79,8 +79,8 @@ int FILESYSTEM_init(char *argvZero)
 	}
 
 	/* Mount the stock content last */
-	strcpy(output, PHYSFS_getBaseDir());
-	strcat(output, "data.zip");
+	strcpy(output, "_RPM_DATA_DIR_/");
+	strcat(output, "VVVVVV/data.zip");
 	if (!PHYSFS_mount(output, NULL, 1))
 	{
 		puts("Error: data.zip missing!");
@@ -99,8 +99,7 @@ int FILESYSTEM_init(char *argvZero)
 		return 0;
 	}
 
-	strcpy(output, PHYSFS_getBaseDir());
-	strcpy(output, "gamecontrollerdb.txt");
+	strcpy(output, "_RPM_DATA_DIR_/SDL_GameControllerDB/gamecontrollerdb.txt");
 	if (SDL_GameControllerAddMappingsFromFile(output) < 0)
 	{
 		printf("gamecontrollerdb.txt not found!\n");
-- 
2.24.1

