From 26c7ea9a4d0f7bc3b879a32af210257d9226a62d Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 3 Feb 2020 14:32:48 -0300
Subject: [PATCH] System data file

---
 desktop_version/src/FileSystemUtils.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
index 9ab1e5c..76d1ed6 100644
--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -82,8 +82,7 @@ int FILESYSTEM_init(char *argvZero, char *assetsPath)
 	if (assetsPath) {
 		strcpy(output, assetsPath);
 	} else {
-		strcpy(output, PHYSFS_getBaseDir());
-		strcat(output, "data.zip");
+		strcpy(output, "_RPM_DATA_DIR_/VVVVVV/data.zip");
 	}
 	if (!PHYSFS_mount(output, NULL, 1))
 	{
@@ -103,8 +102,7 @@ int FILESYSTEM_init(char *argvZero, char *assetsPath)
 		return 0;
 	}
 
-	strcpy(output, PHYSFS_getBaseDir());
-	strcpy(output, "gamecontrollerdb.txt");
+	strcpy(output, "_RPM_DATA_DIR_/SDL_GameControllerDB/gamecontrollerdb.txt");
 	if (SDL_GameControllerAddMappingsFromFile(output) < 0)
 	{
 		printf("gamecontrollerdb.txt not found!\n");
-- 
2.24.1

