From 998975f3eff54e9259710be2994e2a17e5aaee0a Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 28 Jul 2022 19:22:32 -0300
Subject: [PATCH] System libraries

---
 desktop_version/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/desktop_version/CMakeLists.txt b/desktop_version/CMakeLists.txt
index 5751aa7..a739c88 100644
--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -330,6 +330,11 @@ if(BUNDLE_DEPENDENCIES)
 
     target_link_libraries(VVVVVV ${STATIC_LIBRARIES})
 else()
+    find_package(FAudio CONFIG)
+    add_library(faudio-shared INTERFACE)
+    target_link_libraries(faudio-shared INTERFACE FAudio::FAudio)
+    add_library(FAudio ALIAS faudio-shared)
+
     target_link_libraries(VVVVVV ${STATIC_LIBRARIES} physfs tinyxml2 FAudio)
 endif()
 
-- 
2.41.0

