diff -up lutris/lutris/util/wine.py.nofile lutris/lutris/util/wine.py
--- lutris/lutris/util/wine.py.nofile	2019-01-12 11:22:53.911442019 -0200
+++ lutris/lutris/util/wine.py	2019-01-12 11:23:14.615106729 -0200
@@ -11,7 +11,7 @@ from lutris.util.log import logger
 from lutris.util.strings import version_sort
 from lutris.runners.steam import steam
 
-MIN_NUMBER_FILES_OPEN = 1048576
+MIN_NUMBER_FILES_OPEN = 524888
 WINE_DIR = os.path.join(settings.RUNNER_DIR, "wine")
 WINE_PATHS = {
     'winehq-devel': '/opt/wine-devel/bin/wine',
