diff -up ./lutris/game.py.gamemodelib ./lutris/game.py
--- ./lutris/game.py.gamemodelib	2019-11-26 23:35:02.416147868 -0300
+++ ./lutris/game.py	2019-11-26 23:36:08.977179843 -0300
@@ -497,7 +497,7 @@ class Game(GObject.Object):
             env["LD_PRELOAD"] = ":".join(
                 [
                     path
-                    for path in [env.get("LD_PRELOAD"), "libgamemodeauto.so", ]
+                    for path in [env.get("LD_PRELOAD"), "libgamemodeauto.so.0", ]
                     if path
                 ]
             )
diff -up ./lutris/util/linux.py.gamemodelib ./lutris/util/linux.py
--- ./lutris/util/linux.py.gamemodelib	2019-11-26 23:35:26.428108887 -0300
+++ ./lutris/util/linux.py	2019-11-26 23:36:09.705027103 -0300
@@ -71,7 +71,7 @@ SYSTEM_COMPONENTS = {
         "VULKAN": ["libvulkan.so.1"],
         "WINE": ["libsqlite3.so.0"],
         "RADEON": ["libvulkan_radeon.so"],
-        "GAMEMODE": ["libgamemodeauto.so"],
+        "GAMEMODE": ["libgamemodeauto.so.0"],
     },
 }
 
