From f91174fcf42e6edbb646309f79befa8e99cbaf42 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 5 Aug 2024 20:02:16 -0300
Subject: [PATCH] GeneralPane: add missing ANALYTICS definitions

---
 Source/Core/DolphinQt/Settings/GeneralPane.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/Core/DolphinQt/Settings/GeneralPane.cpp b/Source/Core/DolphinQt/Settings/GeneralPane.cpp
index 55dc4f0..db00c32 100644
--- a/Source/Core/DolphinQt/Settings/GeneralPane.cpp
+++ b/Source/Core/DolphinQt/Settings/GeneralPane.cpp
@@ -425,6 +425,7 @@ void GeneralPane::AddDescriptions()
   static constexpr char TR_FALLBACK_REGION_DESCRIPTION[] =
       QT_TR_NOOP("Sets the region used for titles whose region cannot be determined automatically."
                  "<br><br>This setting cannot be changed while emulation is active.");
+#if defined(USE_ANALYTICS) && USE_ANALYTICS
   static constexpr char TR_ENABLE_ANALYTICS_DESCRIPTION[] = QT_TR_NOOP(
       "If selected, Dolphin can collect data on its performance, feature usage, emulated games, "
       "and configuration, as well as data on your system's hardware and operating system."
@@ -434,6 +435,7 @@ void GeneralPane::AddDescriptions()
   static constexpr char TR_GENERATE_NEW_IDENTITY_DESCRIPTION[] =
       QT_TR_NOOP("Generate a new anonymous ID for your usage statistics. This will cause any "
                  "future statistics to be unassociated with your previous statistics.");
+#endif
 
   m_checkbox_dualcore->SetDescription(tr(TR_DUALCORE_DESCRIPTION));
 
@@ -459,8 +461,10 @@ void GeneralPane::AddDescriptions()
   m_combobox_fallback_region->SetTitle(tr("Fallback Region"));
   m_combobox_fallback_region->SetDescription(tr(TR_FALLBACK_REGION_DESCRIPTION));
 
+#if defined(USE_ANALYTICS) && USE_ANALYTICS
   m_checkbox_enable_analytics->SetDescription(tr(TR_ENABLE_ANALYTICS_DESCRIPTION));
 
   m_button_generate_new_identity->SetTitle(tr("Generate a New Statistics Identity"));
   m_button_generate_new_identity->SetDescription(tr(TR_GENERATE_NEW_IDENTITY_DESCRIPTION));
+#endif
 }
-- 
2.45.2

