From e7e715b4698006b66a3274080b8e6f3be7769758 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 29 Jun 2024 17:57:59 -0300
Subject: [PATCH] cmake: Downgrade minizip detection

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8eb4f7..ab784db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -680,8 +680,9 @@ if(TARGET PkgConfig::ZLIB)
   add_definitions(-DUSE_SHARED_ZLIBNG)
 endif()
 
+add_definitions(-DMZ_COMPAT_VERSION=110)  # This macro is for forwards compatibility with 4.0.4+
 dolphin_find_optional_system_library_pkgconfig(MINIZIP
-  "minizip>=4.0.4" minizip::minizip Externals/minizip-ng
+  "minizip>=3.0.10" minizip::minizip Externals/minizip-ng
 )
 
 dolphin_find_optional_system_library(LZO Externals/LZO)
-- 
2.45.2

