From 44c6b396831dc1ded3d94197721b612f4a812a46 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 19 Jun 2020 15:51:01 -0300
Subject: [PATCH] DSP interrupt hack for RE 2 and 3

From https://github.com/MoArtis/dolphin, MoArtis <mathieuphilippe.dev@gmail.com>
---
 Source/Core/Core/HW/DSP.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/Core/Core/HW/DSP.cpp b/Source/Core/Core/HW/DSP.cpp
index 4e431dd..aada59f 100644
--- a/Source/Core/Core/HW/DSP.cpp
+++ b/Source/Core/Core/HW/DSP.cpp
@@ -476,6 +476,10 @@ static void Do_ARAM_DMA()
 {
   s_dspState.DMAState = 1;
 
+  // RESHDP Team
+  // Re-implementation of an old hack to ""fix"" the music of RE 2 and 3
+  GenerateDSPInterrupt(INT_ARAM);
+
   // ARAM DMA transfer rate has been measured on real hw
   int ticksToTransfer = (s_arDMA.Cnt.count / 32) * 246;
   CoreTiming::ScheduleEvent(ticksToTransfer, s_et_CompleteARAM);
-- 
2.26.2

