From ac61952ee98ba808ad98adb07eee5fa67b61324d Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 3 Jan 2021 17:10:35 -0300
Subject: [PATCH] RESHDP Edition label

From https://github.com/MoArtis/dolphin, MoArtis <mathieuphilippe.dev@gmail.com>
---
 Source/Core/Common/scmrev.h.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/Core/Common/scmrev.h.in b/Source/Core/Common/scmrev.h.in
index df99e55..399a51d 100644
--- a/Source/Core/Common/scmrev.h.in
+++ b/Source/Core/Common/scmrev.h.in
@@ -1,5 +1,5 @@
 #define SCM_REV_STR "${DOLPHIN_WC_REVISION}"
-#define SCM_DESC_STR "${DOLPHIN_WC_DESCRIBE}"
+#define SCM_DESC_STR "${DOLPHIN_WC_DESCRIBE} - RESHDP Edition"
 #define SCM_BRANCH_STR "${DOLPHIN_WC_BRANCH}"
 #define SCM_IS_MASTER ${DOLPHIN_WC_IS_STABLE}
 #define SCM_DISTRIBUTOR_STR "${DISTRIBUTOR}" 
-- 
2.29.2

