From 95f677869777c99f8494ebd5e56a7da7768bb9a8 Mon Sep 17 00:00:00 2001
From: Mystro256 <alexjnewt@hotmail.com>
Date: Sun, 5 Apr 2020 17:46:52 -0400
Subject: [PATCH 3/4] soundtouch: disable exceptions

---
 Externals/soundtouch/STTypes.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Externals/soundtouch/STTypes.h b/Externals/soundtouch/STTypes.h
index 0f98c889fb..d5440c3b74 100644
--- a/Externals/soundtouch/STTypes.h
+++ b/Externals/soundtouch/STTypes.h
@@ -162,7 +162,7 @@ namespace soundtouch
 };
 
 // define ST_NO_EXCEPTION_HANDLING switch to disable throwing std exceptions:
-// #define ST_NO_EXCEPTION_HANDLING    1
+#define ST_NO_EXCEPTION_HANDLING    1
 #ifdef ST_NO_EXCEPTION_HANDLING
     // Exceptions disabled. Throw asserts instead if enabled.
     #include <assert.h>
-- 
2.29.2

