--- htop-3.0.5/linux/LibSensors.c	2021-01-11 18:18:21.000000000 +0100
+++ htop-3.0.5-/linux/LibSensors.c	2021-02-18 14:04:22.284009236 +0100
@@ -89,6 +89,7 @@
       if (!String_startsWith(buffer, "coretemp") &&
           !String_startsWith(buffer, "cpu_thermal") &&
           !String_startsWith(buffer, "k10temp") &&
+          !String_startsWith(buffer, "acpitz") &&
           !String_startsWith(buffer, "zenpower"))
          continue;
 
