From 1ac7d99a73c33d06972458fb56cc94beb80e9a55 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 11 Apr 2020 10:32:27 -0300
Subject: [PATCH] dxvk.conf: async option

---
 dxvk.conf | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/dxvk.conf b/dxvk.conf
index 1b5d8b5..dc9f945 100644
--- a/dxvk.conf
+++ b/dxvk.conf
@@ -314,4 +314,12 @@
 # Supported values:
 # - True/False
 
-# d3d9.longMad = False
\ No newline at end of file
+# d3d9.longMad = False
+
+# Sets the async pipeline compiler
+# 
+# Behaves like the DXVK_ASYNC environment variable.
+# Like DXVK_ASYNC, always take precedence, if enabled.
+# Supported values: True, False
+
+# dxvk.useAsync = False
-- 
2.26.0

