From 9a7d6ae0b608c27303e90fce5a3ccb0469c91a6b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 5 May 2020 10:16:54 -0300
Subject: [PATCH] dxvk.conf: async options

---
 dxvk.conf | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/dxvk.conf b/dxvk.conf
index 99f0d30..e024413 100644
--- a/dxvk.conf
+++ b/dxvk.conf
@@ -331,4 +331,22 @@
 # Supported values:
 # - True/False
 
-# d3d9.longMad = False
+# d3d9.longMad = False
+
+
+# Sets the async pipeline compiler
+# 
+# Behaves like the DXVK_ASYNC environment variable.
+# Like DXVK_ASYNC, always take precedence, if enabled.
+# Supported values: True, False
+
+# dxvk.useAsync = False
+
+
+# Specify the number of async pipeline compiler threads
+
+# Supported values:
+# - 0 to automatically determine the number of threads to use
+# - any positive number to enforce the thread count
+
+# dxvk.numAsyncCompilerThreads = 0
-- 
2.26.2

