From 6c5c97e463eac57f5347d02524051441c37f1188 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 8 Mar 2025 19:42:37 -0300
Subject: [PATCH] gplasync fixup 1

---
 src/dxvk/dxvk_context.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/dxvk/dxvk_context.cpp b/src/dxvk/dxvk_context.cpp
index 6a81f12..26221fc 100644
--- a/src/dxvk/dxvk_context.cpp
+++ b/src/dxvk/dxvk_context.cpp
@@ -6932,12 +6932,11 @@ namespace dxvk {
         return false;
     }
 
-    // End render pass if there are pending resolves
-    if (m_flags.any(DxvkContextFlag::GpDirtyFramebuffer,
-                    DxvkContextFlag::GpRenderPassNeedsFlush))
+    if (m_flags.test(DxvkContextFlag::GpDirtyFramebuffer))
       this->updateFramebuffer();
 
-    if (m_flags.test(DxvkContextFlag::GpXfbActive)) {
+    if (!m_flags.test(DxvkContextFlag::GpRenderPassBound))
+      this->startRenderPass();
       // If transform feedback is active and there is a chance that we might
       // need to rebind the pipeline, we need to end transform feedback and
       // issue a barrier. End the render pass to do that. Ignore dirty vertex
-- 
2.48.1

