From 85c1086e17cb501780b3040bc2921b7869fde3fc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 1 May 2023 12:32:50 -0300
Subject: [PATCH] [util] Add d3d9.deferSurfaceCreation to some games

---
 src/util/config/config.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/util/config/config.cpp b/src/util/config/config.cpp
index 7403ab9..8af2674 100644
--- a/src/util/config/config.cpp
+++ b/src/util/config/config.cpp
@@ -570,10 +570,6 @@ namespace dxvk {
     { R"(\\RichardBurnsRally_SSE\.exe$)", {{
       { "d3d9.floatEmulation",              "Strict" },
     }} },
-    /* BlazBlue Centralfiction                  */
-    { R"(\\BBCF\.exe$)", {{
-      { "d3d9.floatEmulation",              "Strict" },
-    }} },
     /* Resident Evil games using MT Framework   */
     { R"(\\(rerev|re0hd|bhd)\.exe$)", {{
       { "d3d9.allowDirectBufferMapping",                "False" },
@@ -720,6 +716,7 @@ namespace dxvk {
     /* BlazBlue Centralfiction                 *
      * Temporary crash workaround              */
     { R"(\\BBCF\.exe$)", {{
+      { "d3d9.floatEmulation",              "Strict" },
       { "d3d9.textureMemory",               "0"   },
     }} },
     /* Battle Fantasia Revised Edition         *
@@ -750,6 +747,14 @@ namespace dxvk {
     { R"(\\DCGAME\.EXE$)", {{
       { "d3d9.deviceLost",              "True"   },
     }} },
+    /* Bejeweled 3  */
+    { R"(\\Bejeweled3\.exe$)", {{
+      { "d3d9.deferSurfaceCreation",        "True" },
+    }} },
+    /* Melty Blood Type Lumina  */
+    { R"(\\MBTL\.exe$)", {{
+      { "d3d9.deferSurfaceCreation",        "True" },
+    }} },
     
     /**********************************************/
     /* D3D12 GAMES (vkd3d-proton with dxvk dxgi)  */
-- 
2.40.1

