From ee1527ad3ff62f441bada378ce03c8ea2adf981f Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 13 Jan 2024 17:00:31 -0300
Subject: [PATCH] dxvk.conf: gplasync options

---
 dxvk.conf | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/dxvk.conf b/dxvk.conf
index 96e686a..50d466e 100644
--- a/dxvk.conf
+++ b/dxvk.conf
@@ -851,3 +851,21 @@
 # - True/False
 
 # d3d8.forceLegacyDiscard = False
+
+
+# Sets the async pipeline compiler
+# 
+# Behaves like the DXVK_ASYNC environment variable.
+# Like DXVK_ASYNC, always take precedence, if enabled.
+# Supported values: True, False
+
+# dxvk.useAsync = False
+
+
+# Sets state cache with gpl and fixes for async
+#
+# Behaves like the DXVK_GPLASYNCCACHE environment variable.
+# Like DXVK_GPLASYNCCACHE, always take precedence, if enabled.
+# Supported values: True, False
+
+# dxvk.gplAsyncCache = False
-- 
2.48.1

