From de2a37727d891018ff4dcbcf605c2105038b75fb Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 7 Jun 2021 14:59:20 -0300
Subject: [PATCH] PR1582: add optional include

---
 src/spirv/spirv_module.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/spirv/spirv_module.h b/src/spirv/spirv_module.h
index a1bd587..6795367 100644
--- a/src/spirv/spirv_module.h
+++ b/src/spirv/spirv_module.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <unordered_set>
+#include <optional>
 
 #include "spirv_code_buffer.h"
 #include "spirv_compression.h"
@@ -1348,4 +1349,4 @@ namespace dxvk {
     
   };
   
-}
\ No newline at end of file
+}
-- 
2.31.1

