From 85469659e4d9e886024f1e74002796a532a050bc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 6 May 2022 14:17:52 -0300
Subject: [PATCH] mangoapp libdir install

---
 src/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/meson.build b/src/meson.build
index fcbb1f8..3249e84 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -282,6 +282,7 @@ if get_option('mangoapp_layer')
     gnu_symbol_visibility : 'hidden',
     include_directories : [inc_common],
     link_args : link_args,
+    install_dir : libdir_mangohud,
     install : true
   )
 endif
-- 
2.36.0

