From e78f95102e3fddf2a704e0cae8613d2abc02d9e9 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 2 Mar 2020 11:42:22 -0300
Subject: [PATCH] VNC: Fix passwords with TigerVNC

---
 lib/asbru_conn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/asbru_conn b/lib/asbru_conn
index 0c3c371..4dc32ff 100755
--- a/lib/asbru_conn
+++ b/lib/asbru_conn
@@ -1211,7 +1211,7 @@ if (defined $METHOD) {
         _setProxy($method);
         if ($method eq 'TigerVNC') {
             my $pfile = "$CFG_DIR/tmp/asbru_conn_{$$}_$UUID";
-            system("echo \"$PASS\" | vncpasswd -f > $pfile");
+            system("echo \'$PASS\' | vncpasswd -f > $pfile");
             if (defined $$CFG{'tmp'}{'xid'}) {
                 $connection_cmd = "$METHOD $CONNECT_OPTS -PasswordFile=$pfile -Parent=$$CFG{'tmp'}{'xid'} $IP:$PORT";
                 $connection_txt = "$METHOD $CONNECT_OPTS -PasswordFile=$pfile -Parent=$$CFG{'tmp'}{'xid'} $IP:$PORT";
-- 
2.24.1

