diff -rupN mingw-w64-v6.0.0/mingw-w64-headers/configure mingw-w64-v6.0.0-new/mingw-w64-headers/configure
--- mingw-w64-v6.0.0/mingw-w64-headers/configure	2018-09-14 16:16:43.000000000 +0200
+++ mingw-w64-v6.0.0-new/mingw-w64-headers/configure	2019-05-07 17:04:47.100390886 +0200
@@ -2576,7 +2576,7 @@ fi
 $as_echo "$with_widl" >&6; }
 
 if test "x$with_widl" = xno; then :
-
+  unset WIDL
 else
 
   if test "x$srcdir" = x.; then :
diff -rupN mingw-w64-v6.0.0/mingw-w64-headers/configure.ac mingw-w64-v6.0.0-new/mingw-w64-headers/configure.ac
--- mingw-w64-v6.0.0/mingw-w64-headers/configure.ac	2018-09-14 16:16:43.000000000 +0200
+++ mingw-w64-v6.0.0-new/mingw-w64-headers/configure.ac	2019-05-07 17:04:47.100390886 +0200
@@ -27,7 +27,7 @@ AC_ARG_WITH([widl],
   [AS_VAR_SET([with_widl],[no])])
 AC_MSG_RESULT([$with_widl])
 
-AS_VAR_IF([with_widl],[no],[],[
+AS_VAR_IF([with_widl],[no],[unset WIDL],[
   AS_VAR_IF([srcdir],[.],[],
     [AC_MSG_WARN([--with-widl used in out of the tree compilation.  Existing generated files will not be modified.])])
   AC_PATH_PROGS([WIDL], [$host-widl widl], [], [$with_widl])
