Index: qt5ct/src/qt5ct/main.cpp
===================================================================
--- qt5ct/src/qt5ct/main.cpp	(revisão 531)
+++ qt5ct/src/qt5ct/main.cpp	(revisão 532)
@@ -46,6 +46,8 @@
     qt_translator.load(QLibraryInfo::location (QLibraryInfo::TranslationsPath) + "/qtbase_" + locale);
     app.installTranslator(&qt_translator);
 
+    Qt5CT::initConfig();
+
     qDebug() << "Configuration path:" << Qt5CT::configPath();
     qDebug() << "Shared QSS paths:" << Qt5CT::sharedStyleSheetPaths();
     qDebug() << "Shared color scheme paths:" << Qt5CT::sharedColorSchemePaths();
Index: qt5ct/src/qt5ct/qt5ct.cpp
===================================================================
--- qt5ct/src/qt5ct/qt5ct.cpp	(revisão 531)
+++ qt5ct/src/qt5ct/qt5ct.cpp	(revisão 532)
@@ -31,6 +31,8 @@
 #include <QLatin1String>
 #include <QStandardPaths>
 #include <QRegularExpression>
+#include <QFile>
+#include <QtDebug>
 #include "qt5ct.h"
 
 #ifndef QT5CT_DATADIR
@@ -38,6 +40,19 @@
 #endif
 
 
+void Qt5CT::initConfig()
+{
+    if(QFile::exists(configFile()))
+        return;
+
+    QString globalConfig = QStandardPaths::locate(QStandardPaths::GenericConfigLocation, "qt5ct/qt5ct.conf");
+    if(globalConfig.isEmpty())
+        return;
+
+    QDir("/").mkpath(configPath());
+    QFile::copy(globalConfig, configFile());
+}
+
 QString Qt5CT::configPath()
 {
     return QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + QLatin1String("/qt5ct");
Index: qt5ct/src/qt5ct/qt5ct.h
===================================================================
--- qt5ct/src/qt5ct/qt5ct.h	(revisão 531)
+++ qt5ct/src/qt5ct/qt5ct.h	(revisão 532)
@@ -44,6 +44,7 @@
 class Qt5CT
 {
 public:
+    static void initConfig();
     static QString configPath();
     static QString configFile();
     static QStringList iconPaths();
Index: qt5ct/src/qt5ct-qtplugin/qt5ctplatformtheme.cpp
===================================================================
--- qt5ct/src/qt5ct-qtplugin/qt5ctplatformtheme.cpp	(revisão 531)
+++ qt5ct/src/qt5ct-qtplugin/qt5ctplatformtheme.cpp	(revisão 532)
@@ -65,6 +65,7 @@
 
 Qt5CTPlatformTheme::Qt5CTPlatformTheme()
 {
+    Qt5CT::initConfig();
     if(QGuiApplication::desktopSettingsAware())
     {
         readSettings();
