From 484229a4b37cc5b71b7cf6b13bb417630159ed2b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 21 Feb 2023 16:14:45 -0300
Subject: [PATCH] Add missing DEBUG_FILEDATA conditional

---
 src/filedata.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/filedata.cc b/src/filedata.cc
index daaabf7..22d2024 100644
--- a/src/filedata.cc
+++ b/src/filedata.cc
@@ -704,7 +704,9 @@ void file_data_dump()
 		{
 		list = g_hash_table_get_values(file_data_pool);
 
+#ifdef DEBUG_FILEDATA
 		log_printf("%d", global_file_data_count);
+#endif
 		log_printf("%d", g_list_length(list));
 
 		while (list)
-- 
2.39.2

