From 990bd8336a4d4fa2bab1a02a6b257220f782e70d Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 4 Apr 2023 23:11:23 -0300
Subject: [PATCH] rcfile: fix build

---
 src/rcfile.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rcfile.cc b/src/rcfile.cc
index 35905c8..22ed00e 100644
--- a/src/rcfile.cc
+++ b/src/rcfile.cc
@@ -225,7 +225,7 @@ gboolean read_int_option_clamp(const gchar *option, const gchar *label, const gc
 	return ret;
 }
 
-void write_int_unit_option(GString *str, gint UNUSED(indent), gchar *label, gint n, gint subunits)
+void write_int_unit_option(GString *str, gint UNUSED(indent), const gchar *label, gint n, gint subunits)
 {
 	gint l, r;
 
-- 
2.40.0

