--- mps-youtube-0.2.8/mps_youtube/config.py.orig	2018-02-18 01:54:43.000000000 -0300
+++ mps-youtube-0.2.8/mps_youtube/config.py	2018-07-06 18:03:58.772184742 -0300
@@ -300,7 +300,7 @@ class _Config:
             ConfigItem("playerargs", ""),
             ConfigItem("encoder", 0, minval=0, check_fn=check_encoder),
             ConfigItem("notifier", ""),
-            ConfigItem("checkupdate", True),
+            ConfigItem("checkupdate", False),
             ConfigItem("show_mplayer_keys", True, require_known_player=True),
             ConfigItem("fullscreen", False, require_known_player=True),
             ConfigItem("show_status", True),
