From 969d4d70c7539d1d2a336011038224e776b49105 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 8 Jun 2023 23:23:17 -0300
Subject: [PATCH] Lower the SDL2 requirement a bit

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b52125..047b5b5 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -506,7 +506,7 @@ if (ENABLE_SDL2)
     elseif (YUZU_USE_EXTERNAL_SDL2)
         message(STATUS "Using SDL2 from externals.")
     else()
-        find_package(SDL2 2.26.4 REQUIRED)
+        find_package(SDL2 2.26.3 REQUIRED)
     endif()
 endif()
 
-- 
2.40.1

