From c77c85a470e085cbc3668b6f7d9053366a2dbcc5 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 13 Sep 2022 14:28:06 -0300
Subject: [PATCH] system minizip fix

---
 src/arcade/imageprovider.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/arcade/imageprovider.h b/src/arcade/imageprovider.h
index bee46ee..557521b 100644
--- a/src/arcade/imageprovider.h
+++ b/src/arcade/imageprovider.h
@@ -8,7 +8,7 @@
 #include <QUuid>
 #include <QQuickImageProvider>
 
-#include "../minizip/mz_compat.h"
+#include "mz_compat.h"
 #include "../sevenzipfile.h"
 #if defined(QMC2_ARCADE_LIBARCHIVE_ENABLED)
 #include "../archivefile.h"
-- 
2.37.3

