From ce970cd290435eae4213d96329803d48a77a8cc8 Mon Sep 17 00:00:00 2001
From: Khalil Gharbaoui <kaygeee@gmail.com>
Date: Tue, 10 Aug 2021 20:18:06 +0000
Subject: [PATCH] Fix the 60 second delay

---
 src/usb_gsource.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/usb_gsource.cpp b/src/usb_gsource.cpp
index 00bf131..afb38f6 100644
--- a/src/usb_gsource.cpp
+++ b/src/usb_gsource.cpp
@@ -174,7 +174,10 @@ USBGSource::on_source_dispatch(GSource* source, GSourceFunc callback, gpointer u
 gboolean
 USBGSource::on_source()
 {
-  libusb_handle_events(NULL);
+  struct timeval to;
+  to.tv_sec = 0;
+  to.tv_usec = 0;
+  libusb_handle_events_timeout_completed(NULL, &to, NULL);
   return TRUE;
 }
 
-- 
GitLab

