From 57f2db4f2312ba70dd4edbd5e02d6e5d8a72b80b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 12 Jul 2023 21:33:21 -0300
Subject: [PATCH] RMG-Core: shared library fixes

---
 Source/RMG-Core/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Source/RMG-Core/CMakeLists.txt b/Source/RMG-Core/CMakeLists.txt
index 6ef9fea..900c637 100644
--- a/Source/RMG-Core/CMakeLists.txt
+++ b/Source/RMG-Core/CMakeLists.txt
@@ -1,7 +1,7 @@
 #
 # RMG-Core CMakeLists.txt
 #
-project(RMG-Core)
+project(RMG-Core VERSION _RPMVERSION_)
 
 set(CMAKE_CXX_STANDARD 20)
 
@@ -91,6 +91,8 @@ endif(USE_LIBFMT)
 
 add_library(RMG-Core SHARED ${RMG_CORE_SOURCES})
 
+set_target_properties(RMG-Core PROPERTIES VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}" SOVERSION ${PROJECT_VERSION_MAJOR})
+
 if(UNIX)
     target_link_libraries(RMG-Core dl)
 endif(UNIX)
-- 
2.41.0

