From 01525cf3fcaf98b73a63f9fbe5f7c0c2f43595ec Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 15 Mar 2023 23:27:09 -0300
Subject: [PATCH] angrylion-rdp-plus: gcc 13 build fix

---
 CMakeLists.txt        | 2 ++
 src/core/parallel.cpp | 1 +
 2 files changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b13397..0e89eb2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ option(USE_QT5 "Enables Qt5 instead of Qt6 (will be removed in the future)" OFF)
 
 project(angrylion-plus)
 
+set(CMAKE_CXX_STANDARD 11)
+
 include(GNUInstallDirs)
 
 set(OpenGL_GL_PREFERENCE GLVND)
diff --git a/src/core/parallel.cpp b/src/core/parallel.cpp
index b9e7efa..d4fb6a4 100644
--- a/src/core/parallel.cpp
+++ b/src/core/parallel.cpp
@@ -8,6 +8,7 @@
 #include <mutex>
 #include <thread>
 #include <vector>
+#include <stdexcept>
 
 class Parallel
 {
-- 
2.39.2

