From 30ba616c31f7f5b20eaa0c96ff5147d452354643 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 26 Aug 2021 08:20:38 -0300
Subject: [PATCH] Use system vulkan headers

---
 Common/GPU/Vulkan/VulkanLoader.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Common/GPU/Vulkan/VulkanLoader.h b/Common/GPU/Vulkan/VulkanLoader.h
index 83836b6..98b1386 100644
--- a/Common/GPU/Vulkan/VulkanLoader.h
+++ b/Common/GPU/Vulkan/VulkanLoader.h
@@ -36,7 +36,7 @@
 #define VK_ENABLE_BETA_EXTENSIONS				1		// VK_KHR_portability_subset
 #endif
 
-#include "ext/vulkan/vulkan.h"
+#include <vulkan/vulkan.h>
 #include <string>
 
 // Hacky X11 header workaround
-- 
2.52.0

