From fd8827d88200ad305c7990ae1ecfca57d10aeb34 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 26 Jan 2021 08:32:08 -0300
Subject: [PATCH] Fix undefined references

---
 CMakeLists.txt | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c120608..0be0115 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,7 +128,7 @@ PRIVATE
 )
 
 if (NOT WIN32 AND NOT APPLE)
-    target_link_libraries(tg_owt PRIVATE tg_owt::libevent)
+    target_link_libraries(tg_owt PRIVATE tg_owt::libevent X11 Xtst)
 endif()
 if (APPLE)
     target_link_libraries(tg_owt PUBLIC tg_owt::libsdkmacos)
@@ -1985,7 +1985,22 @@ PRIVATE
     stats/rtc_stats.cc
     stats/rtcstats_objects.cc
 
-
+    api/numerics/samples_stats_counter.cc
+    api/numerics/samples_stats_counter.h
+    modules/desktop_capture/linux/shared_x_display.cc
+    modules/desktop_capture/linux/shared_x_display.h
+    modules/desktop_capture/mouse_cursor_monitor_linux.cc
+    modules/desktop_capture/screen_drawer_linux.cc
+    modules/desktop_capture/screen_drawer_lock_posix.cc
+    modules/desktop_capture/screen_drawer_lock_posix.h
+    modules/desktop_capture/screen_capturer_linux.cc
+    modules/desktop_capture/window_capturer_linux.cc
+    modules/video_coding/deprecated/nack_module.cc
+    modules/video_coding/deprecated/nack_module.h
+    modules/rtp_rtcp/source/deprecated/deprecated_rtp_sender_egress.cc
+    modules/rtp_rtcp/source/deprecated/deprecated_rtp_sender_egress.h
+    video/video_stream_decoder_impl.cc
+    video/video_stream_decoder_impl.h
 )
 
 if (is_x86 OR is_x64)
-- 
2.29.2

