diff -rupN --no-dereference binutils-2.37/bfd/elfnn-aarch64.c binutils-2.37-new/bfd/elfnn-aarch64.c
--- binutils-2.37/bfd/elfnn-aarch64.c	2021-07-08 13:37:19.000000000 +0200
+++ binutils-2.37-new/bfd/elfnn-aarch64.c	2021-07-24 21:59:12.398868364 +0200
@@ -5473,6 +5473,7 @@ elfNN_aarch64_final_link_relocate (reloc
      it here if it is defined in a non-shared object.  */
   if (h != NULL
       && h->type == STT_GNU_IFUNC
+      && (input_section->flags & SEC_ALLOC)
       && h->def_regular)
     {
       asection *plt;
