diff -rupN --no-dereference binutils-2.36.1/bfd/configure binutils-2.36.1-new/bfd/configure
--- binutils-2.36.1/bfd/configure	2021-02-06 10:02:09.000000000 +0100
+++ binutils-2.36.1-new/bfd/configure	2021-02-13 18:13:19.528228597 +0100
@@ -13461,11 +13461,13 @@ if test "x${ac_cv_sizeof_long}" = "x8";
   BFD_HOST_64BIT_LONG=1
   test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
   test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
-elif test "x${ac_cv_sizeof_long_long}" = "x8"; then
+fi
+if test "x${ac_cv_sizeof_long_long}" = "x8"; then
   BFD_HOST_64BIT_LONG_LONG=1
   test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
   test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
-  if test "x${ac_cv_sizeof_void_p}" = "x8"; then
+  if test "x${ac_cv_sizeof_void_p}" = "x8" \
+          -a "x${ac_cv_sizeof_long}" != "x8"; then
     BFD_HOSTPTR_T="unsigned long long"
   fi
 fi
diff -rupN --no-dereference binutils-2.36.1/bfd/configure.ac binutils-2.36.1-new/bfd/configure.ac
--- binutils-2.36.1/bfd/configure.ac	2021-01-09 11:47:33.000000000 +0100
+++ binutils-2.36.1-new/bfd/configure.ac	2021-02-13 18:13:19.526228607 +0100
@@ -219,11 +219,13 @@ if test "x${ac_cv_sizeof_long}" = "x8";
   BFD_HOST_64BIT_LONG=1
   test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
   test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
-elif test "x${ac_cv_sizeof_long_long}" = "x8"; then
+fi
+if test "x${ac_cv_sizeof_long_long}" = "x8"; then
   BFD_HOST_64BIT_LONG_LONG=1
   test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
   test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
-  if test "x${ac_cv_sizeof_void_p}" = "x8"; then
+  if test "x${ac_cv_sizeof_void_p}" = "x8" \
+          -a "x${ac_cv_sizeof_long}" != "x8"; then
     BFD_HOSTPTR_T="unsigned long long"
   fi
 fi
