From 49d17f63ced79b866c8b48ebe8f74da961fefbbe Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 29 Mar 2022 11:11:47 -0300
Subject: [PATCH] skip some tests

---
 sdl2/test/joystick_test.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sdl2/test/joystick_test.py b/sdl2/test/joystick_test.py
index bcb3e3c..199290f 100644
--- a/sdl2/test/joystick_test.py
+++ b/sdl2/test/joystick_test.py
@@ -275,7 +275,7 @@ def test_SDL_JoystickGetProductVersion(joysticks):
         assert SDL_GetError() == b""
         assert pver >= 0
 
-@pytest.mark.skipif(sdl2.dll.version < 2014, reason="not available")
+@pytest.mark.skip("not implemented")
 def test_SDL_JoystickGetSerial(joysticks):
     for stick in joysticks:
         serial = joystick.SDL_JoystickGetSerial(stick)
@@ -418,7 +418,7 @@ def test_SDL_JoystickSendEffect(joysticks):
     # to each controller type, so can't easily test this.
     pass
 
-@pytest.mark.skipif(sdl2.dll.version < 2014, reason="not available")
+@pytest.mark.skip("not implemented")
 def test_SDL_JoystickVirtual(with_sdl):
     jcount = joystick.SDL_NumJoysticks()
     jtype = joystick.SDL_JOYSTICK_TYPE_GAMECONTROLLER
-- 
2.35.1

