From b2fdc0e9852fcf60825ed474b17353ac7367e6bd Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 2 Oct 2021 08:19:03 -0300
Subject: [PATCH] gtk3 dialogs

---
 src/qt6ct/appearancepage.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/qt6ct/appearancepage.cpp b/src/qt6ct/appearancepage.cpp
index 2f2d6ef..5791c33 100644
--- a/src/qt6ct/appearancepage.cpp
+++ b/src/qt6ct/appearancepage.cpp
@@ -48,6 +48,7 @@ AppearancePage::AppearancePage(QWidget *parent) :
 {
     m_ui->setupUi(this);
     QStringList keys = QStyleFactory::keys();
+    keys.removeAll("gtk2"); //hide gtk2 style
     keys.removeAll("qt6ct-style"); //hide qt6ct proxy style
     keys.removeAll("qt5gtk2"); //hide qt5gtk2 alias
     keys.removeAll("gtk2"); //hide gtk2 alias
@@ -82,10 +83,10 @@ AppearancePage::AppearancePage(QWidget *parent) :
 
     keys = QPlatformThemeFactory::keys();
     m_ui->dialogComboBox->addItem(tr("Default"), "default");
-    if(keys.contains("gtk2") || keys.contains("qt6gtk2"))
-        m_ui->dialogComboBox->addItem("GTK2", "gtk2");
-    else if(keys.contains("gtk3") || keys.contains("qt6gtk3"))
+    if(keys.contains("gtk3") || keys.contains("qt6gtk3"))
         m_ui->dialogComboBox->addItem("GTK3", "gtk3");
+    else if(keys.contains("gtk2") || keys.contains("qt6gtk2"))
+        m_ui->dialogComboBox->addItem("GTK2", "gtk2");
     if(keys.contains("kde"))
         m_ui->dialogComboBox->addItem("KDE", "kde");
     if (keys.contains("xdgdesktopportal"))
-- 
2.38.1

