--- gamemode-1.0/meson.build.orig	2018-04-04 06:06:23.000000000 -0300
+++ gamemode-1.0/meson.build	2018-04-11 10:16:05.948436170 -0300
@@ -83,8 +83,7 @@ subdir('lib')
 # main library
 if with_daemon == true
     # inih currently only needed by the daemon
-    inih = subproject('inih')
-    inih_dependency = inih.get_variable('inih_dependency')
+    inih_dependency = cc.find_library('inih')
 
     subdir('daemon')
 
