diff -up gamemode-1.1/lib/gamemode_client.h.soname gamemode-1.1/lib/gamemode_client.h
--- gamemode-1.1/lib/gamemode_client.h.soname	2018-05-11 10:25:02.000000000 -0300
+++ gamemode-1.1/lib/gamemode_client.h	2018-05-11 23:27:40.727626547 -0300
@@ -152,7 +152,7 @@ __attribute__((always_inline)) static in
 	void *libgamemode = NULL;
 
 	/* Try and load libgamemode */
-	libgamemode = dlopen("libgamemode.so", RTLD_NOW);
+	libgamemode = dlopen("libgamemode.so.0", RTLD_NOW);
 	if (!libgamemode) {
 		snprintf(internal_gamemode_client_error_string,
 		         sizeof(internal_gamemode_client_error_string),
diff -up gamemode-1.1/lib/meson.build.soname gamemode-1.1/lib/meson.build
--- gamemode-1.1/lib/meson.build.soname	2018-05-11 10:25:02.000000000 -0300
+++ gamemode-1.1/lib/meson.build	2018-05-11 23:28:20.823205724 -0300
@@ -1,6 +1,7 @@
 # Main client library to message the daemon
 shared_library(
     'gamemode',
+    version: '0.0.1',
     sources: [
         'client_impl.c',
     ],
@@ -17,6 +18,7 @@ libgamemode_includes = [
 # Small library to automatically use gamemode
 shared_library(
     'gamemodeauto',
+    version: '0.0.1',
     sources: [
         'client_loader.c',
     ],
