From 52118bc8f2eed79cba4afbbcd188d947b8a2fbf4 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 8 Jan 2023 12:08:32 -0300
Subject: [PATCH] Use system libraries

---
 yabause/src/CMakeLists.txt | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/yabause/src/CMakeLists.txt b/yabause/src/CMakeLists.txt
index 5a37d68..fd91bd9 100755
--- a/yabause/src/CMakeLists.txt
+++ b/yabause/src/CMakeLists.txt
@@ -181,12 +181,28 @@ set(kronos_SOURCES
 	)
 add_definitions(-DIMPROVED_SAVESTATES)
 
+find_package(PkgConfig REQUIRED)
+
+pkg_check_modules(CHDR IMPORTED_TARGET libchdr)
+if(CHDR_FOUND)
+  message(STATUS "Using shared libchdr")
+  include_directories(${CHDR_INCLUDE_DIRS})
+  set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} PkgConfig::CHDR)
+else()
 add_subdirectory(tools/libchdr)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/tools/libchdr/src)
 set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} chdr-static)
+endif()
 
+find_package(ZLIB)
+if(ZLIB_FOUND)
+  message(STATUS "Using shared zlib")
+  include_directories(${ZLIB_INCLUDE_DIR})
+  set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} ZLIB::ZLIB)
+else()
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/tools/libchdr/deps/zlib-1.2.11)
 set(KRONOS_LIBRARIES ${KRONOS_LIBRARIES} zlibstatic)
+endif()
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/tools/nanovg)
 set(kronos_SOURCES ${kronos_SOURCES} tools/nanovg/nanovg.c)
-- 
2.39.0

