From 849787a7d6e3685d829df3aed13f6e9ba30845d7 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 7 Jan 2024 18:14:11 -0300
Subject: [PATCH] format security

---
 Src/OSD/Logger.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Src/OSD/Logger.cpp b/Src/OSD/Logger.cpp
index 59e3c79..36c20f5 100644
--- a/Src/OSD/Logger.cpp
+++ b/Src/OSD/Logger.cpp
@@ -344,7 +344,7 @@ void CSystemLogger::DebugLog(const char *fmt, va_list vl)
 #ifdef _WIN32
   OutputDebugString(string2);
 #else
-  syslog(LOG_DEBUG, string2);
+  syslog(LOG_DEBUG, "%s", string2);
 #endif
 }
 
@@ -364,7 +364,7 @@ void CSystemLogger::InfoLog(const char *fmt, va_list vl)
 #ifdef _WIN32
   OutputDebugString(string2);
 #else
-  syslog(LOG_INFO, string2);
+  syslog(LOG_INFO, "%s", string2);
 #endif
 }
 
@@ -384,11 +384,11 @@ void CSystemLogger::ErrorLog(const char *fmt, va_list vl)
  #ifdef _WIN32
   OutputDebugString(string2);
 #else
-  syslog(LOG_ERR, string2);
+  syslog(LOG_ERR, "%s", string2);
 #endif
 }
 
 CSystemLogger::CSystemLogger(CLogger::LogLevel level)
   : m_logLevel(level)
 {
-}
\ No newline at end of file
+}
-- 
2.43.0

