From 9afa35e04f45aef479b24a37c590334ed7d36576 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 12 Jun 2020 03:08:33 -0300
Subject: [PATCH] antilib: add versioned soname

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 63c3680..dd2f65d 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -492,6 +492,11 @@ endif(UNIX)
                 ${antimicroX_RESOURCES_RCC}
             )
 
+            set_target_properties(antilib PROPERTIES
+                VERSION "${ANTIMICROX_MAJOR_VERSION}.${ANTIMICROX_MINOR_VERSION}.${ANTIMICROX_PATCH_VERSION}"
+                SOVERSION "${ANTIMICROX_MAJOR_VERSION}"
+            )
+
             #target_link_libraries (antilib Qt5::Widgets Qt5::Core Qt5::Test Qt5::Gui Qt5::Network Qt5::Concurrent ${SDL_LIBRARY} ${LIBS})
             target_link_libraries (antilib Qt5::Widgets Qt5::Core Qt5::Gui Qt5::Network Qt5::Concurrent ${SDL_LIBRARY} ${LIBS})
 
-- 
2.26.2

