From bdae1b3015d5f5437b910d2396a32be3209b17ba Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 12 Nov 2025 07:33:33 -0300
Subject: [PATCH] Offline build fixes

---
 hatch_build.py | 10 ----------
 pyproject.toml |  1 +
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/hatch_build.py b/hatch_build.py
index 54fffb6..684b736 100755
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -11,16 +11,6 @@ except ImportError:
 
 class CustomBuildHook(BuildHookInterface):
     def initialize(self, version, build_data):
-        name, cmds, env = build_bundle_cmds()
-        if cmds is None:
-            raise RuntimeError(
-                "One of 'pnpm', 'deno', 'bun', or 'npm' could not be found. "
-                "Please install one of them to proceed with the build."
-            )
-        print(f"Building with {name}...")
-
-        for cmd in cmds:
-            subprocess.run(cmd, env=env, check=True)
 
         build_data["force_include"].update(
             {
diff --git a/pyproject.toml b/pyproject.toml
index a2968e1..073b3dd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,6 +42,7 @@ Source = "https://github.com/yt-dlp/ejs"
 
 [tool.hatch.version]
 source = "vcs"
+fallback-version = "_RPM_VERSION_"
 
 [tool.hatch.build.targets.sdist]
 exclude = [
-- 
2.52.0

