From b239c50c6cca56b141a2351c8284c5f5078f8bed Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 12 Nov 2025 07:33:33 -0300
Subject: [PATCH] Offline build fixes

---
 hatch_build.py | 23 -----------------------
 pyproject.toml |  1 +
 2 files changed, 1 insertion(+), 23 deletions(-)

diff --git a/hatch_build.py b/hatch_build.py
index 70e0b3b..bb62a75 100644
--- a/hatch_build.py
+++ b/hatch_build.py
@@ -7,29 +7,6 @@ from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 
 class CustomBuildHook(BuildHookInterface):
     def initialize(self, version, build_data):
-        if shutil.which("deno"):
-            print("Building with deno...", flush=True)
-            os.environ["DENO_NO_UPDATE_CHECK"] = "1"
-            subprocess.run(["deno", "install", "--frozen"], check=True)
-            subprocess.run(["deno", "task", "bundle"], check=True)
-
-        elif shutil.which("bun"):
-            print("Building with bun...", flush=True)
-            subprocess.run(["bun", "install"], check=True)
-            subprocess.run(["bun", "--bun", "run", "bundle"], check=True)
-
-        elif shutil.which("npm"):
-            print("Building with npm...", flush=True)
-            # npm is a batch file (`npm.cmd`) on windows, which requires `shell=True`
-            requires_shell = os.name == "nt"
-            subprocess.run(["npm", "install"], check=True, shell=requires_shell)
-            subprocess.run(["npm", "run", "bundle"], check=True, shell=requires_shell)
-
-        else:
-            raise RuntimeError(
-                "One of 'deno', 'bun', or 'npm' could not be found. "
-                "Please install one of them to proceed with the build."
-            )
 
         build_data["force_include"].update(
             {
diff --git a/pyproject.toml b/pyproject.toml
index a2968e1..073b3dd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,6 +42,7 @@ Source = "https://github.com/yt-dlp/ejs"
 
 [tool.hatch.version]
 source = "vcs"
+fallback-version = "_RPM_VERSION_"
 
 [tool.hatch.build.targets.sdist]
 exclude = [
-- 
2.51.1

