From db5b78f39d94bd826462361fb1a7f671d9d1ff33 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 26 Apr 2022 17:02:45 -0300
Subject: [PATCH] cmake: fix data files install

---
 gtk/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gtk/CMakeLists.txt b/gtk/CMakeLists.txt
index 247c738..5c9d6a6 100644
--- a/gtk/CMakeLists.txt
+++ b/gtk/CMakeLists.txt
@@ -24,8 +24,8 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "One of: Debug Release RelWithDebInfo MinSizeRel" FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
-string(APPEND DATADIR ${CMAKE_INSTALL_PREFIX} "/" ${CMAKE_INSTALL_DATAROOTDIR} "/" ${CMAKE_INSTALL_DATADIR})
-string(APPEND LOCALEDIR ${CMAKE_INSTALL_PREFIX} "/" ${CMAKE_INSTALL_DATAROOTDIR} "/" ${CMAKE_INSTALL_LOCALEDIR})
+string(APPEND DATADIR ${SHARE_INSTALL_PREFIX} "/" ${CMAKE_INSTALL_DATADIR})
+string(APPEND LOCALEDIR ${SHARE_INSTALL_PREFIX} "/" ${CMAKE_INSTALL_LOCALEDIR})
 add_compile_definitions(HAVE_LIBPNG
                         ZLIB SNES9X_GTK
                         NETPLAY_SUPPORT
-- 
2.40.0

