From 41df51de1d87d38da1d8146003fe48a4538056f7 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 19 Mar 2025 20:25:46 -0300
Subject: [PATCH] glslang: gcc 15 build fix

---
 external/glslang/SPIRV/SpvBuilder.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/external/glslang/SPIRV/SpvBuilder.h b/external/glslang/SPIRV/SpvBuilder.h
index 02e9cf4..e3afe9f 100644
--- a/external/glslang/SPIRV/SpvBuilder.h
+++ b/external/glslang/SPIRV/SpvBuilder.h
@@ -60,6 +60,7 @@ namespace spv {
 #include <memory>
 #include <set>
 #include <sstream>
+#include <cstdint>
 #include <stack>
 #include <unordered_map>
 #include <map>
-- 
2.49.0

