From 142826e846e479c2081ac507461ce13d628c82a0 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 1 Feb 2025 12:22:26 -0300
Subject: [PATCH] cmake: fix when discord-rpc is disabled

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a0e2d9..902b6db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -679,4 +679,6 @@ if(ENABLE_QT AND UNIX AND NOT APPLE)
             DESTINATION "share/metainfo")
 endif()
 
+if (USE_DISCORD_PRESENCE)
 target_compile_options(discord-rpc PRIVATE -fpermissive)
+endif()
-- 
2.48.1

