From 1857868f47efe996a482db7721f75aa7e3438d57 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 23 Mar 2026 19:57:47 -0300
Subject: [PATCH] Add smaller game icon sizes

---
 src/yuzu/main_window.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/yuzu/main_window.cpp b/src/yuzu/main_window.cpp
index 2e7cef0..394b852 100644
--- a/src/yuzu/main_window.cpp
+++ b/src/yuzu/main_window.cpp
@@ -384,6 +384,8 @@ namespace {
 
 constexpr std::array<std::pair<u32, const char*>, 5> default_game_icon_sizes{
     std::make_pair(0, QT_TRANSLATE_NOOP("MainWindow", "None")),
+    std::make_pair(16, QT_TRANSLATE_NOOP("MainWindow", "Mini (16x16)")),
+    std::make_pair(24, QT_TRANSLATE_NOOP("MainWindow", "Smaller (24x24)")),
     std::make_pair(32, QT_TRANSLATE_NOOP("MainWindow", "Small (32x32)")),
     std::make_pair(64, QT_TRANSLATE_NOOP("MainWindow", "Standard (64x64)")),
     std::make_pair(128, QT_TRANSLATE_NOOP("MainWindow", "Large (128x128)")),
-- 
2.53.0

