From c959f994cc93591d6f88c4df5f58064a5df3b0b9 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 20 Sep 2024 23:02:16 -0300
Subject: [PATCH] Bundled fmt support

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2455795..9665622 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -398,6 +398,12 @@ endif()
 
 add_subdirectory(externals)
 
+if(fmt_FOUND)
+    message(STATUS "Using the system fmt")
+else()
+    add_subdirectory("externals/fmt" EXCLUDE_FROM_ALL)
+endif()
+
 if (ENABLE_QT)
     if (YUZU_USE_BUNDLED_QT)
         download_qt(6.8.3)
-- 
2.49.0

