From 292d8b730c1c571461c52abdb6193056d9258d03 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 23 Mar 2026 19:57:47 -0300
Subject: [PATCH] Add smaller game icon sizes

---
 src/yuzu/main_window.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/yuzu/main_window.cpp b/src/yuzu/main_window.cpp
index 2e7cef0..bd9e8b7 100644
--- a/src/yuzu/main_window.cpp
+++ b/src/yuzu/main_window.cpp
@@ -382,8 +382,10 @@ static QString PrettyProductName() {
 
 namespace {
 
-constexpr std::array<std::pair<u32, const char*>, 5> default_game_icon_sizes{
+constexpr std::array<std::pair<u32, const char*>, 7> default_game_icon_sizes{
     std::make_pair(0, QT_TRANSLATE_NOOP("MainWindow", "None")),
+    std::make_pair(16, QT_TRANSLATE_NOOP("MainWindow", "Mini (16x16)")),
+    std::make_pair(24, QT_TRANSLATE_NOOP("MainWindow", "Smaller (24x24)")),
     std::make_pair(32, QT_TRANSLATE_NOOP("MainWindow", "Small (32x32)")),
     std::make_pair(64, QT_TRANSLATE_NOOP("MainWindow", "Standard (64x64)")),
     std::make_pair(128, QT_TRANSLATE_NOOP("MainWindow", "Large (128x128)")),
-- 
2.53.0

