From 9ba55aec4456a424367e572b7955393166b96b7a Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 4 Oct 2022 23:14:00 -0300
Subject: [PATCH] vfs-lzma: update bool for lzma-sdk

---
 src/util/vfs/vfs-lzma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/vfs/vfs-lzma.c b/src/util/vfs/vfs-lzma.c
index 65a6fe7..db13f1a 100644
--- a/src/util/vfs/vfs-lzma.c
+++ b/src/util/vfs/vfs-lzma.c
@@ -134,7 +134,7 @@ struct VDir* VDirOpen7z(const char* path, int flags) {
 	vd->allocTempImp.Free = _vd7zFreeTemp;
 
 	FileInStream_CreateVTable(&vd->archiveStream);
-	LookToRead2_CreateVTable(&vd->lookStream, False);
+	LookToRead2_CreateVTable(&vd->lookStream, False7z);
 
 	vd->lookStream.realStream = &vd->archiveStream.vt;
 	vd->lookStream.buf = malloc(BUFFER_SIZE);
-- 
2.43.0

