--- wine-mono-4.5.6/build-winemono.sh.orig	2015-04-20 10:34:32.150114394 -0500
+++ wine-mono-4.5.6/build-winemono.sh	2015-04-20 10:33:13.699274928 -0500
@@ -72,6 +72,7 @@
     cd "$BUILDDIR/build-cross-$ARCH"
     if test 1 != $REBUILD || test ! -e Makefile; then
         CPPFLAGS="-gdwarf-2 -gstrict-dwarf" "$SRCDIR"/mono/configure --prefix="$BUILDDIR/build-cross-$ARCH-install" --build=$BUILD --target=$MINGW --host=$MINGW --with-tls=none --disable-mcs-build --enable-win32-dllmain=yes --with-libgc-threads=win32 PKG_CONFIG=false mono_cv_clang=no || exit 1
+        sed -e 's+compiler_flags="-specs=/usr/lib/rpm/redhat/redhat-hardened-ld"+compiler_flags="-static"+' -i libtool
         sed -e 's/-lgcc_s//' -i libtool
     fi
     WINEPREFIX=/dev/null make $MAKEOPTS || exit 1
@@ -118,6 +119,7 @@
     cd "$BUILDDIR/build-cross-cli"
     if test 1 != $REBUILD || test ! -e Makefile; then
         "$SRCDIR"/mono/configure --prefix="$BUILDDIR/build-cross-cli-install" --with-mcs-docs=no --disable-system-aot || exit 1
+        sed -e 's+compiler_flags="-specs=/usr/lib/rpm/redhat/redhat-hardened-ld"+compiler_flags="-static"+' -i libtool
     fi
     if test 1 = $USE_MONOLITE; then
         make get-monolite-latest || exit 1
@@ -172,6 +174,7 @@
     # build mono-basic
     cd "$SRCDIR/mono-basic"
     ./configure --prefix="$BUILDDIR/build-cross-cli-install" || exit 1
+    sed -e 's+compiler_flags="-specs=/usr/lib/rpm/redhat/redhat-hardened-ld"+compiler_flags="-static"+' -i libtool
     make $MAKEOPTS || exit 1
     make install || exit 1
 
