From d285eb1c21eca399ac23ee7e33d98b98d56707ff Mon Sep 17 00:00:00 2001
From: Fedora Kernel Team <kernel-team@fedoraproject.org>
Date: Wed, 8 Apr 2020 16:46:49 -0300
Subject: [PATCH] postfactum: fix build without BMQ

---
 kernel/sched/sched.h | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 6a56ebd..db3bcde 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2502,17 +2502,4 @@ static inline int task_running_nice(struct task_struct *p)
 	return (task_nice(p) > 0);
 }
 
-#ifdef CONFIG_SMP
-static inline bool is_per_cpu_kthread(struct task_struct *p)
-{
-	if (!(p->flags & PF_KTHREAD))
-		return false;
-
-	if (p->nr_cpus_allowed != 1)
-		return false;
-
-	return true;
-}
-#endif
-
 #endif /* !CONFIG_SCHED_BMQ */
-- 
2.26.0

