--- _upstream/kernel/kernel-fdist/kernel.spec	2017-02-02 19:30:42.767075982 -0200
+++ kernel/kernel.spec	2017-02-02 19:33:22.137390488 -0200
@@ -24,7 +24,7 @@ Summary: The Linux kernel
 %global zipsed -e 's/\.ko$/\.ko.xz/'
 %endif
 
-# define buildid .local
+%define buildid .chinfo
 
 # baserelease defines which build revision of this kernel version we're
 # building.  We used to call this fedora_build, but the magical name
@@ -42,7 +42,7 @@ Summary: The Linux kernel
 # For non-released -rc kernels, this will be appended after the rcX and
 # gitX tags, so a 3 here would become part of release "0.rcX.gitX.3"
 #
-%global baserelease 200
+%global baserelease 500
 %global fedora_build %{baserelease}
 
 # base_sublevel is the kernel version we're starting with and patching
@@ -89,7 +89,7 @@ Summary: The Linux kernel
 # kernel PAE (only valid for i686 (PAE) and ARM (lpae))
 %define with_pae       %{?_without_pae:       0} %{?!_without_pae:       1}
 # kernel-debug
-%define with_debug     %{?_without_debug:     0} %{?!_without_debug:     1}
+%define with_debug     0
 # kernel-headers
 %define with_headers   %{?_without_headers:   0} %{?!_without_headers:   1}
 %define with_cross_headers   %{?_without_cross_headers:   0} %{?!_without_cross_headers:   1}
@@ -111,7 +111,7 @@ Summary: The Linux kernel
 # Only build the pae kernel (--with paeonly):
 %define with_paeonly   %{?_with_paeonly:      1} %{?!_with_paeonly:      0}
 # Only build the debug kernel (--with dbgonly):
-%define with_dbgonly   %{?_with_dbgonly:      1} %{?!_with_dbgonly:      0}
+%define with_dbgonly   0
 #
 # should we do C=1 builds with sparse
 %define with_sparse    %{?_with_sparse:       1} %{?!_with_sparse:       0}
@@ -468,7 +468,7 @@ Source2001: cpupower.config
 %if 0%{?stable_update}
 %if 0%{?stable_base}
 %define    stable_patch_00  patch-4.%{base_sublevel}.%{stable_base}.xz
-Source5000: %{stable_patch_00}
+Source5000: ftp://ftp.kernel.org/pub/linux/kernel/v4.x/%{stable_patch_00}
 %endif
 
 # non-released_kernel case
@@ -639,6 +639,31 @@ Patch854: 0001-x86-efi-always-map-first-
 # CVE-2017-2596 rhbz 1417812 1417813
 Patch855: kvm-fix-page-struct-leak-in-handle_vmon.patch
 
+### Extra
+
+# Add additional cpu gcc optimization support
+# https://github.com/graysky2/kernel_gcc_patch (20160728)
+Source4000: https://github.com/graysky2/kernel_gcc_patch/raw/master/enable_additional_cpu_optimizations_for_gcc_v4.9+_kernel_v3.15+.patch
+
+### openSUSE patches.suse
+# Patches to export btrfs anonymous devices (VFS portion)
+Patch1010: vfs-add-super_operations-get_inode_dev.patch
+Patch1011: btrfs-provide-super_operations-get_inode_dev.patch
+Patch1012: revert-btrfs-fix-lockdep-warning-on-deadlock-against-an-inode-s-log-mutex.patch
+Patch1013: revert-btrfs-improve-performance-on-fsync-against-new-inode-after-rename-unlink.patch
+Patch1014: btrfs-fix-extent-tree-corruption-due-to-relocation.patch
+
+### openSUSE patches.arch
+Patch1015: perf_timechart_fix_zero_timestamps.patch
+
+# BFQ disk scheduler - http://algo.ing.unimo.it/people/paolo/disk_sched/
+Patch1030: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/0001-block-cgroups-kconfig-build-bits-for-BFQ-v7r11-4.5.0.patch
+Patch1031: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/0002-block-introduce-the-BFQ-v7r11-I-O-sched-for-4.5.0.patch
+Patch1032: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/0003-block-bfq-add-Early-Queue-Merge-EQM-to-BFQ-v7r11-for.patch
+Patch1033: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/0004-Turn-into-BFQ-v8r7-for-4.9.0.patch
+Source3000: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/COPYING.BFQ
+Source3001: http://algo.ing.unimo.it/people/paolo/disk_sched/patches/4.9.0-v8r7/README.BFQ
+
 # END OF PATCH DEFINITIONS
 
 %endif
@@ -815,7 +840,7 @@ This package provides debug information
 # symlinks because of the trailing nonmatching alternation and
 # the leading .*, because of find-debuginfo.sh's buggy handling
 # of matching the pattern against the symlinks file.
-%{expand:%%global debuginfo_args %{?debuginfo_args} -p '.*%%{_bindir}/centrino-decode(\.debug)?|.*%%{_bindir}/powernow-k8-decode(\.debug)?|.*%%{_bindir}/cpupower(\.debug)?|.*%%{_libdir}/libcpupower.*|.*%%{_bindir}/turbostat(\.debug)?|.*%%{_bindir}/x86_energy_perf_policy(\.debug)?|.*%%{_bindir}/tmon(\.debug)?|.*%%{_bindir}/lsgpio(\.debug)?|.*%%{_bindir}/gpio-hammer(\.debug)?|.*%%{_bindir}/gpio-event-mon(\.debug)?|.*%%{_bindir}/iio_event_monitor(\.debug)?|.*%%{_bindir}/iio_generic_buffer(\.debug)?|.*%%{_bindir}/lsiio(\.debug)?|XXX' -o kernel-tools-debuginfo.list}
+%{expand:%%global debuginfo_args %{?debuginfo_args} -p '.*%%{_bindir}/centrino-decode(\.debug)?|.*%%{_bindir}/powernow-k8-decode(\.debug)?|.*%%{_bindir}/cpupower(\.debug)?|.*%%{_libdir}/libcpupower.*|.*%%{_bindir}/turbostat(\.debug)?|.*%%{_bindir}/x86_energy_perf_policy(\.debug)?|.*%%{_bindir}/tmon(\.debug)?|.*%%{_bindir}/iio_event_monitor(\.debug)?|.*%%{_bindir}/iio_generic_buffer(\.debug)?|.*%%{_bindir}/lsiio(\.debug)?|XXX' -o kernel-tools-debuginfo.list}
 
 %endif # with_tools
 
@@ -1242,6 +1267,8 @@ git am %{SOURCE5005}
 
 git am %{patches}
 
+patch -p1 -F1 -i %{SOURCE4000}
+
 # END OF PATCH APPLICATIONS
 
 %endif
@@ -2186,50 +2213,35 @@ fi
 #
 #
 %changelog
-* Thu Feb 02 2017 Laura Abbott <labbott@fedoraproject.org> - 4.9.7-200
+* Thu Feb 02 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.7-500.chinfo
 - Linux v4.9.7
+- f25 sync
 
-* Tue Jan 31 2017 Justin M. Forbes <jforbes@fedoraproject.org>
-- Fix kvm nested virt CVE-2017-2596 (rhbz 1417812 1417813)
-
-* Tue Jan 31 2017 Peter Robinson <pbrobinson@fedoraproject.org>
-- Fix CMA compaction regression (Raspberry Pi and others)
-
-* Thu Jan 26 2017 Peter Robinson <pbrobinson@fedoraproject.org>
-- arm64: dma-mapping: Fix dma_mapping_error() when bypassing SWIOTLB
-
-* Thu Jan 26 2017 Laura Abbott <labbott@redhat.com> - 4.9.6-200
+* Thu Jan 26 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.6-500.chinfo
 - Linux v4.9.6
-- Bring in fix for bogus EFI firmware
-- Fixes CVE-2017-5547, CVE-2016-10153, CVE-2017-5548, CVE-2017-5551
-  (rhbz 1416096 1416101 1416110 1416126 1416128)
-
-* Wed Jan 25 2017 Justin M. Forbes <jforbes@fedoraproject.org>
-- CVE-2017-5576 CVE-2017-5577 vc4 overflows (rhbz 1416436 1416437 1416439)
-
-* Mon Jan 23 2017 Justin M. Forbes <jforbes@fedoraproject.org>
-- Enable CONFIG_IPV6_GRE (rhbz 1405398)
+- f25 sync
 
-* Fri Jan 20 2017 Laura Abbott <labbott@redhat.com> - 4.9.5-200
+* Fri Jan 20 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.5-500.chinfo
 - Linux v4.9.5
+- Insanely high build number to escape from Fedora packages. ;)
 
-* Tue Jan 17 2017 Laura Abbott <labbott@fedoraproject.org>
-- Fix kubernetes networking issue (rhbz 1414068)
-
-* Tue Jan 17 2017 Laura Abbott <labbott@fedoraproject.org> - 4.9.4-201
-- Add possible ATI fixes
-
-* Mon Jan 16 2017 Laura Abbott <labbott@fedoraproject.org> - 4.9.4-200
-- Linux v4.9.4
-
-* Mon Jan 16 2017 Peter Robinson <pbrobinson@fedoraproject.org>
+* Tue Jan 17 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.4-202.chinfo
+- Sync with Fedora default:
 - Minor updates for Raspberry Pi 3 support
 - Re-enable /sys/class/gpio/
 - Build gpio tools
+- Add possible ATI fixes
 
-* Thu Jan 12 2017 Laura Abbott <labbott@fedoraproject.org> - 4.9.3-200
+* Sun Jan 15 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.4-201.chinfo
+- Linux v4.9.4
+
+* Thu Jan 12 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.3-201.chinfo
 - Linux v4.9.3
 
+* Thu Jan 12 2017 Phantom X <megaphantomx at bol dot com dot br> - 4.9.2-201.chinfo
+- BFQ patches
+- Assorted fixes from openSUSE
+
 * Wed Jan 11 2017 Laura Abbott <labbott@fedoraproject.org>
 - Add fix for some thinkpads missed during the rebase.
 
