From e557e4aeb169cdcc1469c9cf802f20b2f7ab6ee3 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 30 Mar 2022 18:23:12 -0300
Subject: [PATCH] gcc 12 build fix

---
 src/teakra/src/interpreter.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/teakra/src/interpreter.h b/src/teakra/src/interpreter.h
index 72724ee..2106a6c 100644
--- a/src/teakra/src/interpreter.h
+++ b/src/teakra/src/interpreter.h
@@ -5,6 +5,7 @@
 #include <type_traits>
 #include <unordered_map>
 #include <unordered_set>
+#include <utility>
 #include "bit.h"
 #include "core_timing.h"
 #include "crash.h"
-- 
2.35.1

