From 265abdc069736723e1f7745a2ccd7dcffc56b6a4 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 20 Sep 2024 23:02:16 -0300
Subject: [PATCH] Bundled fmt support

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3395c1a..e2d2e5f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -728,6 +728,12 @@ endif()
 add_subdirectory(externals)
 add_subdirectory(src)
 
+if(fmt_FOUND)
+  message(STATUS "Using the system fmt")
+else()
+add_subdirectory("externals/fmt" EXCLUDE_FROM_ALL)
+endif()
+
 # Set citron project or citron-cmd project as default StartUp Project in Visual Studio depending on whether QT is enabled or not
 if(ENABLE_QT)
     set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT citron)
-- 
2.47.1

