From b7c3d3f628fd7709b12e90ef77a7a13b1392a34c Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 20 Apr 2021 11:54:23 -0300
Subject: [PATCH] look-for-libdl.so.2

---
 src/opengl/real_dlsym.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/opengl/real_dlsym.c b/src/opengl/real_dlsym.c
index 91d05de..926071f 100644
--- a/src/opengl/real_dlsym.c
+++ b/src/opengl/real_dlsym.c
@@ -36,7 +36,8 @@ typedef struct {
 	ElfW(Word) bucket[];
 } dt_hash_t;
 
-const char* target_libs[3] = {
+const char* target_libs[4] = {
+	"libdl.so.2",
 	"libdl.so",
 	"libc.so",
 	NULL
-- 
2.52.0

