From 5c96287a98f6258fab98a928ae55868e5cb31e3c Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 17 Aug 2020 08:22:21 -0300
Subject: [PATCH] tgvoip: system json11

---
 Telegram/cmake/lib_tgvoip.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Telegram/cmake/lib_tgvoip.cmake b/Telegram/cmake/lib_tgvoip.cmake
index 328ec03..3ef880f 100644
--- a/Telegram/cmake/lib_tgvoip.cmake
+++ b/Telegram/cmake/lib_tgvoip.cmake
@@ -72,8 +72,6 @@ else()
         video/VideoSource.h
         video/VideoRenderer.cpp
         video/VideoRenderer.h
-        json11.cpp
-        json11.hpp
 
         # Windows
         os/windows/NetworkSocketWinsock.cpp
@@ -824,17 +822,20 @@ else()
     if (LINUX)
         find_package(PkgConfig REQUIRED)
         pkg_check_modules(ALSA REQUIRED alsa)
+        pkg_check_modules(JSON11 REQUIRED json11)
         pkg_check_modules(PULSE REQUIRED libpulse)
 
         target_include_directories(lib_tgvoip_bundled
         PRIVATE
             ${ALSA_INCLUDE_DIRS}
+            ${JSON11_INCLUDE_DIRS}
             ${PULSE_INCLUDE_DIRS}
         )
 
         target_link_libraries(lib_tgvoip_bundled
         PRIVATE
             ${CMAKE_DL_LIBS}
+            ${JSON11_LIBRARIES}
             pthread
         )
     endif()
-- 
2.26.2

