From 8baa7d3294e4f2838135ba8ff299e6aad8940290 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 7 Jan 2020 09:43:34 -0300
Subject: [PATCH] Plugin loading dirty fix with system Qt

---
 Telegram/SourceFiles/core/launcher.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 2547342..972a37a 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -17,6 +17,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "core/sandbox.h"
 #include "base/concurrent_timer.h"
 #include "facades.h"
+#include <QApplication>
 
 namespace Core {
 namespace {
@@ -278,6 +279,9 @@ int Launcher::exec() {
 	Platform::start();
 	Ui::DisableCustomScaling();
 
+	QCoreApplication::addLibraryPath("_RPM_QT_PLUGIN_PATH_");
+	qunsetenv("QT_QPA_PLATFORMTHEME");
+
 	auto result = executeApplication();
 
 	DEBUG_LOG(("Telegram finished, result: %1").arg(result));
-- 
2.24.1

