From 48fa8f43d2d88ec480c00360e0df2bbc0d454575 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 7 Oct 2020 18:38:16 -0300
Subject: [PATCH] tgvoip: system json11

---
 Telegram/cmake/lib_tgvoip.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Telegram/cmake/lib_tgvoip.cmake b/Telegram/cmake/lib_tgvoip.cmake
index 684f927b..7eb33336 100644
--- a/Telegram/cmake/lib_tgvoip.cmake
+++ b/Telegram/cmake/lib_tgvoip.cmake
@@ -76,8 +76,6 @@ if (NOT TGVOIP_FOUND)
         video/VideoSource.h
         video/VideoRenderer.cpp
         video/VideoRenderer.h
-        json11.cpp
-        json11.hpp
 
         # Windows
         os/windows/NetworkSocketWinsock.cpp
@@ -829,16 +827,19 @@ if (NOT TGVOIP_FOUND)
         find_package(PkgConfig REQUIRED)
         find_package(ALSA REQUIRED)
         pkg_check_modules(PULSE REQUIRED libpulse)
+        pkg_check_modules(JSON11 REQUIRED json11)
 
         target_include_directories(lib_tgvoip_bundled
         PRIVATE
             ${ALSA_INCLUDE_DIRS}
             ${PULSE_INCLUDE_DIRS}
+            ${JSON11_INCLUDE_DIRS}
         )
 
         target_link_libraries(lib_tgvoip_bundled
         PRIVATE
             ${CMAKE_DL_LIBS}
+            ${JSON11_LIBRARIES}
             pthread
         )
     endif()
-- 
2.28.0

