From c9125dc7641c73c4a2849cde20294e82050da383 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 8 Oct 2024 18:02:08 -0300
Subject: [PATCH] ltdp: 0005 fixup 2

---
 Telegram/SourceFiles/data/data_stories.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Telegram/SourceFiles/data/data_stories.cpp b/Telegram/SourceFiles/data/data_stories.cpp
index f67b6ff..f53ff88 100644
--- a/Telegram/SourceFiles/data/data_stories.cpp
+++ b/Telegram/SourceFiles/data/data_stories.cpp
@@ -7,7 +7,6 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 #include "data/data_stories.h"
 
-#include "api/api_report.h"
 #include "base/options.h"
 #include "base/unixtime.h"
 #include "apiwrap.h"
-- 
2.46.2

