diff -up tdesktop-1.5.1/Telegram/SourceFiles/dialogs/dialogs_layout.cpp.realmute tdesktop-1.5.1/Telegram/SourceFiles/dialogs/dialogs_layout.cpp
--- tdesktop-1.5.1/Telegram/SourceFiles/dialogs/dialogs_layout.cpp.realmute	2018-12-11 17:09:11.104794469 -0200
+++ tdesktop-1.5.1/Telegram/SourceFiles/dialogs/dialogs_layout.cpp	2018-12-11 17:12:04.733689761 -0200
@@ -121,7 +121,9 @@ int PaintWideCounter(
 		UnreadBadgeStyle st;
 		st.active = active;
 		st.muted = unreadMuted;
-		paintUnreadCount(p, counter, unreadRight, unreadTop, st, &unreadWidth);
+		if (!st.muted) {
+			paintUnreadCount(p, counter, unreadRight, unreadTop, st, &unreadWidth);
+		}
 		availableWidth -= unreadWidth + st.padding;
 
 		hadOneBadge = true;
@@ -801,13 +803,13 @@ void paintImportantSwitch(Painter &p, Mo
 		const auto unreadRight = fullWidth - st::dialogsPadding.x();
 		UnreadBadgeStyle st;
 		st.muted = true;
-		paintUnreadCount(
-			p,
-			QString::number(unread),
-			unreadRight,
-			unreadTop,
-			st,
-			nullptr);
+		//paintUnreadCount(
+			//p,
+			//QString::number(unread),
+			//unreadRight,
+			//unreadTop,
+			//st,
+			//nullptr);
 	}
 }
 
