diff --git a/Telegram/CMakeLists.txt b/Telegram/CMakeLists.txt
index d8c623a..e5b6833 100644
--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -63,7 +63,7 @@ generate_styles(Telegram ${src_loc} "${style_files}" "${dependent_style_files}")
     nuget_add_winrt(Telegram)
 endif()
 
-set_target_properties(Telegram PROPERTIES AUTOMOC ON)
+set_target_properties(Telegram PROPERTIES AUTOMOC ON SKIP_BUILD_RPATH ON)
 target_prepare_qrc(Telegram)
 
 target_link_libraries(Telegram
