diff -up tdesktop-1.5.1/Telegram/gyp/telegram_linux.gypi.orig tdesktop-1.5.1/Telegram/gyp/telegram_linux.gypi
--- tdesktop-1.5.1/Telegram/gyp/telegram_linux.gypi.orig	2018-12-11 17:19:50.239623621 -0200
+++ tdesktop-1.5.1/Telegram/gyp/telegram_linux.gypi	2018-12-11 17:19:59.971599010 -0200
@@ -66,18 +66,9 @@
       ['not_need_gtk!="True"', {
         'cflags_cc': [
           '<!(pkg-config 2> /dev/null --cflags gtk+-3.0)',
-          '<!(pkg-config 2> /dev/null --cflags dee-1.0)',
         ],
-      }], ['<!(pkg-config ayatana-appindicator3-0.1; echo $?) == 0', {
-        'cflags_cc': [ '<!(pkg-config --cflags ayatana-appindicator3-0.1)' ],
-        'defines': [ 'TDESKTOP_USE_AYATANA_INDICATORS' ],
-      }], ['<!(pkg-config ayatana-appindicator-0.1; echo $?) == 0', {
-        'cflags_cc': [ '<!(pkg-config --cflags ayatana-appindicator-0.1)' ],
-        'defines': [ 'TDESKTOP_USE_AYATANA_INDICATORS' ],
       }], ['<!(pkg-config appindicator3-0.1; echo $?) == 0', {
         'cflags_cc': [ '<!(pkg-config --cflags appindicator3-0.1)' ],
-      }], ['<!(pkg-config appindicator-0.1; echo $?) == 0', {
-        'cflags_cc': [ '<!(pkg-config --cflags appindicator-0.1)' ],
       }]
     ],
   }]],
diff -up tdesktop-1.5.1/Telegram/SourceFiles/platform/linux/linux_libs.cpp.orig tdesktop-1.5.1/Telegram/SourceFiles/platform/linux/linux_libs.cpp
--- tdesktop-1.5.1/Telegram/SourceFiles/platform/linux/linux_libs.cpp.orig	2018-12-11 17:20:13.295565317 -0200
+++ tdesktop-1.5.1/Telegram/SourceFiles/platform/linux/linux_libs.cpp	2018-12-11 17:21:45.420332318 -0200
@@ -234,30 +234,18 @@ void start() {
 	bool gtkLoaded = false;
 	bool indicatorLoaded = false;
 	QLibrary lib_gtk, lib_indicator;
-	if (loadLibrary(lib_indicator, "ayatana-appindicator3", 1) || loadLibrary(lib_indicator, "appindicator3", 1)) {
+	if (loadLibrary(lib_indicator, "appindicator3", 1)) {
 		if (loadLibrary(lib_gtk, "gtk-3", 0)) {
 			gtkLoaded = setupGtkBase(lib_gtk);
 			indicatorLoaded = setupAppIndicator(lib_indicator);
 		}
 	}
-	if (!gtkLoaded || !indicatorLoaded) {
-		if (loadLibrary(lib_indicator, "ayatana-appindicator", 1) || loadLibrary(lib_indicator, "appindicator", 1)) {
-			if (loadLibrary(lib_gtk, "gtk-x11-2.0", 0)) {
-				gtkLoaded = indicatorLoaded = false;
-				gtkLoaded = setupGtkBase(lib_gtk);
-				indicatorLoaded = setupAppIndicator(lib_indicator);
-			}
-		}
-	}
 
 	// If no appindicator, try at least load gtk.
 	if (!gtkLoaded && !indicatorLoaded) {
 		if (loadLibrary(lib_gtk, "gtk-3", 0)) {
 			gtkLoaded = setupGtkBase(lib_gtk);
 		}
-		if (!gtkLoaded && loadLibrary(lib_gtk, "gtk-x11-2.0", 0)) {
-			gtkLoaded = setupGtkBase(lib_gtk);
-		}
 	}
 
 	if (gtkLoaded) {
