--- src/psxrip.cpp.orig	2014-05-10 06:54:00.000000000 -0300
+++ src/psxrip.cpp	2018-04-15 04:47:00.384228993 -0300
@@ -107,7 +107,7 @@ static void dumpFilesystem(CdIo_t * imag
 	cdio_info("Dumping '%s' as '%s'", inputPath.c_str(), dirName.c_str());
 
 	// Read the directory entries
-	CdioList_t * entries = iso9660_fs_readdir(image, inputPath.c_str(), false);
+	CdioList_t * entries = iso9660_fs_readdir(image, inputPath.c_str());
 	if (!entries) {
 		throw runtime_error((format("Error reading ISO 9660 directory '%1%'") % inputPath).str());
 	}
@@ -227,7 +227,7 @@ static void dumpFilesystem(CdIo_t * imag
 	// Close the catalog record for the directory
 	catalog << string(level * 2, ' ') << "}\n";
 
-	_cdio_list_free(entries, true);
+	iso9660_filelist_free(entries);
 }
 
 
@@ -297,7 +297,7 @@ static void dumpImage(CdIo_t * image, co
 static void dumpLBNTable(CdIo_t * image, const string & inputPath = "", ostream & output = cout)
 {
 	// Read the directory entries
-	CdioList_t * entries = iso9660_fs_readdir(image, inputPath.c_str(), false);
+	CdioList_t * entries = iso9660_fs_readdir(image, inputPath.c_str());
 	if (!entries) {
 		throw runtime_error((format("Error reading ISO 9660 directory '%1%'") % inputPath).str());
 	}
