From: Michael Chapman <mike@very.puzzling.org>
Date: Tue, 17 Sep 2019 17:03:57 +1000
Subject: [PATCH] remote: fix registration of TLS socket
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Daniel P. Berrangé <berrange@redhat.com>
Signed-off-by: Michael Chapman <mike@very.puzzling.org>
---
 src/remote/remote_daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/remote/remote_daemon.c b/src/remote/remote_daemon.c
index 546328b24d..8f85d09dd3 100644
--- a/src/remote/remote_daemon.c
+++ b/src/remote/remote_daemon.c
@@ -505,7 +505,7 @@ daemonSetupNetworking(virNetServerPtr srv,
                                   config->max_client_requests) < 0)
         goto cleanup;
 
-    if (((ipsock && config->listen_tls) || (act && virSystemdActivationHasName(act, "ip-tls")))) {
+    if (((ipsock && config->listen_tls) || (act && virSystemdActivationHasName(act, DAEMON_NAME "-tls.socket")))) {
         virNetTLSContextPtr ctxt = NULL;
 
         if (config->ca_file ||
