From 647ab1a0aac3740a79b2d502a8bd35630d949102 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 16 Aug 2021 13:32:37 -0300
Subject: [PATCH] libdraw: staticinline

---
 modules/libdraw/libdraw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/libdraw/libdraw.c b/modules/libdraw/libdraw.c
index 905bd67..1dec910 100644
--- a/modules/libdraw/libdraw.c
+++ b/modules/libdraw/libdraw.c
@@ -45,7 +45,7 @@ uint32_t drawing_stipple = 0xFFFFFFFF;
 /* --------------------------------------------------------------------------- */
 
 #ifdef __GNUC__
-#define _inline inline
+#define _inline static inline
 #endif
 
 _inline void _HLine8_nostipple( uint8_t * ptr, uint32_t length )
-- 
2.31.1

