From 857a18f4e0cb1aedf2a0dfe22f161319f8184348 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 25 Nov 2025 14:39:22 -0300
Subject: [PATCH] Remove git requirement

---
 lact-daemon/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lact-daemon/Cargo.toml b/lact-daemon/Cargo.toml
index 8b84391..23e476e 100644
--- a/lact-daemon/Cargo.toml
+++ b/lact-daemon/Cargo.toml
@@ -46,7 +46,7 @@ os-release = "0.1.0"
 notify = { version = "8.0.0", default-features = false }
 libcopes = "1.0.0"
 libloading = "0.8.6"
-cl3 = { git = "https://github.com/kenba/cl3", branch = "develop", features = [
+cl3 = { version = "_RPM_CL3VER_", features = [
     "static",
 ] }
 
-- 
2.52.0

