--- whdd-3.0/CMakeLists.txt.orig	2017-01-07 21:30:06.000000000 -0200
+++ whdd-3.0/CMakeLists.txt	2019-01-05 20:43:00.429769098 -0200
@@ -5,7 +5,7 @@ project(whdd)
 option(STATIC "Build static binaries" OFF)
 option(CLI "Build whdd-cli" OFF)
 
-set(CMAKE_C_FLAGS "-std=gnu99 -pthread -Wall -Wextra -Wno-missing-field-initializers ${CFLAGS}")
+set(CMAKE_C_FLAGS "-std=gnu99 -pthread -Wall -Wextra -Wno-missing-field-initializers ${CMAKE_C_FLAGS}")
 set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS}")
 set(CMAKE_C_FLAGS_DEBUG "-O0 -g -ggdb ${CMAKE_C_FLAGS}")
 
