diff -up vmware-horizon-client-2006-8.0.0.16522670/x86_64/usr/lib/vmware/view/env/env_utils.sh.orig vmware-horizon-client-2006-8.0.0.16522670/x86_64/usr/lib/vmware/view/env/env_utils.sh
--- vmware-horizon-client-2006-8.0.0.16522670/x86_64/usr/lib/vmware/view/env/env_utils.sh.orig	2020-07-06 15:18:11.000000000 +0200
+++ vmware-horizon-client-2006-8.0.0.16522670/x86_64/usr/lib/vmware/view/env/env_utils.sh	2020-09-01 14:15:15.155024947 +0200
@@ -35,6 +35,7 @@ is_glibcxx_compatible_distr() {
    # For distribution after Ubuntu 20.04 and RHEL 8, glibcxx version is compatible
    [ $DISTRO == "ubuntu" ] && [ $DISTRO_VERSION -ge 2004 ] && result="true"
    [ $DISTRO == "rhel" ] && [ $DISTRO_VERSION -ge 8 ] && result="true"
+   [ $DISTRO == "fedora" ] && [ $DISTRO_VERSION -ge 31 ] && result="true"
 
    echo $result
 }
