From e36c06f18346aeac4f6708269845ea911842ca6d Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 4 Dec 2025 07:19:59 -0300
Subject: [PATCH] Remove git requirement

---
 crates/libamdgpu_top/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crates/libamdgpu_top/Cargo.toml b/crates/libamdgpu_top/Cargo.toml
index 76825ad..746d170 100644
--- a/crates/libamdgpu_top/Cargo.toml
+++ b/crates/libamdgpu_top/Cargo.toml
@@ -15,6 +15,6 @@ libdrm_dynamic_loading = ["libdrm_amdgpu_sys/dynamic_loading"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-libdrm_amdgpu_sys = { version = "0.8.12", git = "https://github.com/Umio-Yasuno/libdrm-amdgpu-sys-rs", default-features = false, rev = "5f4d7ea6d4d51b2536ac7605a4c325cac15bdfd9" }
+libdrm_amdgpu_sys = { version = "0.8.12", default-features = false }
 nix = { version = "0", default-features = false, features = [ "ioctl" ] }
 num_enum = { workspace = true }
-- 
2.53.0

