diff -up mp4v2/src/mp4file.cpp.fmt mp4v2/src/mp4file.cpp
--- mp4v2/src/mp4file.cpp.fmt	2017-01-25 00:10:21.329225703 -0200
+++ mp4v2/src/mp4file.cpp	2017-01-25 00:11:23.490051348 -0200
@@ -2580,7 +2580,7 @@ MP4ChapterType MP4File::ConvertChapters(
     GetChapters(&chapters, &chapterCount, sourceType);
     if (0 == chapterCount)
     {
-        VERBOSE_READ(GetVerbosity(), printf(errMsg));
+        VERBOSE_READ(GetVerbosity(), printf("%s", errMsg));
         return MP4ChapterTypeNone;
     }
 
diff -up mp4v2/src/mp4property.cpp.fmt mp4v2/src/mp4property.cpp
--- mp4v2/src/mp4property.cpp.fmt	2017-01-25 00:09:31.772728700 -0200
+++ mp4v2/src/mp4property.cpp	2017-01-25 00:09:33.240417638 -0200
@@ -619,7 +619,7 @@ void MP4BytesProperty::Dump(FILE* pFile,
 
         oss << "  |" << text.str() << "|\n";
 
-        fprintf( pFile, oss.str().c_str() );
+        fprintf( pFile, "%s", oss.str().c_str() );
         fflush( pFile );
         return;
     }
