From b6c8a1c1820b0419bd2844a1b7dad0024d5d808f Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 23 Oct 2025 17:27:09 -0300
Subject: [PATCH] openSUSE fixup 1

---
 kernel/events/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/events/core.c b/kernel/events/core.c
index 6e9427c..330cc15 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -9408,7 +9408,7 @@ static void perf_event_mmap_event(struct perf_mmap_event *mmap_event)
 			name = "//toolong";
 			goto cpy_name;
 		}
-		inode = file_user_inode(vma->vm_file);
+		inode = file_inode(vma->vm_file);
 		dev = inode->i_sb->s_dev;
 		ino = inode->i_ino;
 		gen = inode->i_generation;
-- 
2.51.0

