From 910a75620098cfa1a3d7fcc63d402ff813b0bd83 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 23 Oct 2025 17:27:49 -0300
Subject: [PATCH] openSUSE fixup 2

---
 kernel/events/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/events/core.c b/kernel/events/core.c
index 30384e4..b0d7394 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -9408,7 +9408,7 @@ static void perf_event_mmap_event(struct perf_mmap_event *mmap_event)
 			name = "//toolong";
 			goto cpy_name;
 		}
-		inode = file_inode(vma->vm_file);
+		inode = file_user_inode(vma->vm_file);
 		dev = inode_get_dev(inode);
 		ino = inode->i_ino;
 		gen = inode->i_generation;
-- 
2.51.0

