From 2d75a064118aa9f507488f3282b91aa72fb79f19 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 19 Oct 2025 17:04:11 -0300
Subject: [PATCH] openSUSE fixup 1

---
 include/linux/fs.h | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index ed02715..74f2bfc 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -4025,18 +4025,4 @@ static inline bool vfs_empty_path(int dfd, const char __user *path)
 
 int generic_atomic_write_valid(struct kiocb *iocb, struct iov_iter *iter);
 
-static inline bool extensible_ioctl_valid(unsigned int cmd_a,
-					  unsigned int cmd_b, size_t min_size)
-{
-	if (_IOC_DIR(cmd_a) != _IOC_DIR(cmd_b))
-		return false;
-	if (_IOC_TYPE(cmd_a) != _IOC_TYPE(cmd_b))
-		return false;
-	if (_IOC_NR(cmd_a) != _IOC_NR(cmd_b))
-		return false;
-	if (_IOC_SIZE(cmd_a) < min_size)
-		return false;
-	return true;
-}
-
 #endif /* _LINUX_FS_H */
-- 
2.51.0

