From be8479fc44064b5624a33004b9312cc5d51a0918 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 19 Oct 2025 17:04:56 -0300
Subject: [PATCH] openSUSE fixup 2

---
 include/linux/fs.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index 83e252b..8c9834b 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -4026,6 +4026,20 @@ static inline bool vfs_empty_path(int dfd, const char __user *path)
 
 int generic_atomic_write_valid(struct kiocb *iocb, struct iov_iter *iter);
 
+static inline bool extensible_ioctl_valid(unsigned int cmd_a,
+					  unsigned int cmd_b, size_t min_size)
+{
+	if (_IOC_DIR(cmd_a) != _IOC_DIR(cmd_b))
+		return false;
+	if (_IOC_TYPE(cmd_a) != _IOC_TYPE(cmd_b))
+		return false;
+	if (_IOC_NR(cmd_a) != _IOC_NR(cmd_b))
+		return false;
+	if (_IOC_SIZE(cmd_a) < min_size)
+		return false;
+	return true;
+}
+
 static inline dev_t inode_get_dev(const struct inode *inode)
 {
 	if (inode->i_sb->s_op->get_inode_dev)
-- 
2.51.0

