From f026b519814cccff6775eafe2aaaf109d348c865 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 5 Dec 2024 13:37:02 -0300
Subject: [PATCH] openSUSE: vfs fixup 1

---
 include/linux/fs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index 4b5cad4..3559446 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -3726,6 +3726,6 @@ static inline bool vfs_empty_path(int dfd, const char __user *path)
 	return !c;
 }
 
-int generic_atomic_write_valid(struct kiocb *iocb, struct iov_iter *iter);
+bool generic_atomic_write_valid(struct iov_iter *iter, loff_t pos);
 
 #endif /* _LINUX_FS_H */
-- 
2.47.1

