From b9effe3579b1df7fe2d59c5151caf4d50e0b9e66 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 21 Apr 2021 23:31:04 -0300
Subject: [PATCH] fix mountinfo import when installed

---
 ninfs/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ninfs/main.py b/ninfs/main.py
index 49b7bbd..21cb271 100755
--- a/ninfs/main.py
+++ b/ninfs/main.py
@@ -12,7 +12,7 @@ from os import environ, makedirs
 from os.path import basename, dirname, expanduser, join as pjoin, realpath
 from sys import exit, argv, path, platform, hexversion, version_info
 
-import mountinfo
+from . import mountinfo
 
 windows = platform in {'win32', 'cygwin'}
 
-- 
2.31.1

