From efd0226fac38db346dd4e8c9d7f0ea262395c1dc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 17 Mar 2022 13:53:39 -0300
Subject: [PATCH] Makefile: package build fixes

---
 Makefile | 61 +++++++++++++++++++++++++++++++++++++++++++-------------
 1 file changed, 47 insertions(+), 14 deletions(-)

diff --git a/Makefile b/Makefile
index be7128b..9da0723 100644
--- a/Makefile
+++ b/Makefile
@@ -4,11 +4,11 @@ MT       := 1
 # 0: not need zlib;  1: compile zlib source code;  2: used -lz to link zlib lib;
 ZLIB     := 2
 # 0: not need lzma;  1: compile lzma source code;  2: used -llzma to link lzma lib;
-LZMA     := 1
+LZMA     := 2
 ARM64ASM := 0
 RISCV32  := 0
 # 0: not need zstd;  1: compile zstd source code;  2: used -lzstd to link zstd lib;
-ZSTD     := 1
+ZSTD     := 2
 MD5      := 1
 STATIC_CPP := 0
 # used clang?
@@ -35,6 +35,15 @@ ifeq ($(BZIP2),0)
   endif
 endif
 
+PREFIX ?= /usr/local
+BINDIR ?= $(PREFIX)/bin
+LIBDIR ?= $(PREFIX)/lib
+INCDIR ?= $(PREFIX)/include
+
+LIBRARY = libhdiffpatch.so
+LIBRARYMINOR = $(LIBRARY)._RPM_MINOR_
+LIBRARYMAJOR = $(LIBRARYMINOR)._RPM_MAJOR_
+
 HDIFF_OBJ  := 
 HPATCH_OBJ := \
     libHDiffPatch/HPatch/patch.o \
@@ -57,7 +66,7 @@ else
 	HPATCH_OBJ += bsdiff_wrapper/bspatch_wrapper.o
 endif
 
-MD5_PATH := ../libmd5
+MD5_PATH := libmd5
 ifeq ($(DIR_DIFF),0)
 else
   ifeq ($(MD5),0)
@@ -98,6 +107,7 @@ else
   HDIFF_OBJ  += vcdiff_wrapper/vcdiff_wrapper.o
   ifeq ($(LZMA),0)
   else
+    ifeq ($(LZMA),1)
 	HPATCH_OBJ+=$(LZMA_PATH)/7zCrc.o \
 				$(LZMA_PATH)/7zCrcOpt.o \
 				$(LZMA_PATH)/7zStream.o \
@@ -112,6 +122,7 @@ else
 				$(LZMA_PATH)/XzCrc64Opt.o \
 				$(LZMA_PATH)/XzDec.o
 	HDIFF_OBJ +=$(LZMA_PATH)/XzEnc.o
+    endif
   endif
 endif
 
@@ -208,7 +219,7 @@ endif
 
 
 DEF_FLAGS := \
-    -O3 -DNDEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 \
+    -fPIC -DNDEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 \
     -D_IS_NEED_ALL_CompressPlugin=0 \
     -D_IS_NEED_DEFAULT_CompressPlugin=0 \
     -D_IS_NEED_ALL_ChecksumPlugin=0 \
@@ -238,6 +249,9 @@ else
 	ifeq ($(BZIP2),1)
         DEF_FLAGS += -I$(BZ2_PATH)
 	endif
+	ifeq ($(BZIP2),2)
+        DEF_FLAGS += $(shell pkg-config --cflags-only-I bzip2)
+	endif
 endif
 ifeq ($(ZLIB),0)
 else
@@ -245,6 +259,9 @@ else
 	ifeq ($(ZLIB),1)
         DEF_FLAGS += -I$(ZLIB_PATH)
 	endif
+	ifeq ($(ZLIB),2)
+        DEF_FLAGS += $(shell pkg-config --cflags-only-I zlib)
+	endif
 endif
 ifeq ($(DIR_DIFF),0)
   DEF_FLAGS += -D_IS_NEED_DIR_DIFF_PATCH=0
@@ -289,6 +306,9 @@ else
       DEF_FLAGS += -DUSE_CRC_EMU
     endif
   endif
+  ifeq ($(LZMA),2)
+    DEF_FLAGS += $(shell pkg-config --cflags-only-I lzmasdk-c)
+  endif
 endif
 ifeq ($(ZSTD),0)
 else
@@ -297,6 +317,9 @@ else
     DEF_FLAGS += -DZSTD_HAVE_WEAK_SYMBOLS=0 -DZSTD_TRACE=0 -DZSTD_DISABLE_ASM=1 -DZSTDLIB_VISIBLE= -DZSTDLIB_HIDDEN= \
 	               -I$(ZSTD_PATH) -I$(ZSTD_PATH)/common -I$(ZSTD_PATH)/compress -I$(ZSTD_PATH)/decompress
 	endif
+	ifeq ($(ZSTD),2)
+      DEF_FLAGS += $(shell pkg-config --cflags-only-I libzstd)
+	endif
 endif
 
 ifeq ($(MT),0)
@@ -321,7 +344,7 @@ ifeq ($(ZSTD),2)
   PATCH_LINK += -lzstd		# link zstd
 endif
 ifeq ($(LZMA),2)
-  PATCH_LINK += -llzma		# link lzma
+  PATCH_LINK += -llzmasdk		# link lzmasdk
 endif
 ifeq ($(MT),0)
 else
@@ -351,15 +374,15 @@ CXXFLAGS += $(DEF_FLAGS) -std=c++11
 
 .PHONY: all install clean
 
-all: libhdiffpatch.a hpatchz hdiffz mostlyclean
+all: $(LIBRARYMAJOR) hpatchz hdiffz
 
-libhdiffpatch.a: $(HDIFF_OBJ)
-	$(AR) rcs $@ $^
+$(LIBRARYMAJOR): $(HDIFF_OBJ)
+	$(CXX) -o $(LIBRARYMAJOR) -shared -Wl,-soname=$(LIBRARYMINOR) -DNDEBUG $(LDFLAGS) -Wl,-z,defs $(HDIFF_OBJ) $(PATCH_LINK)
 
 hpatchz: $(HPATCH_OBJ)
-	$(CC) hpatchz.c $(HPATCH_OBJ) $(CFLAGS) $(PATCH_LINK) -o hpatchz
-hdiffz: libhdiffpatch.a
-	$(CXX) hdiffz.cpp libhdiffpatch.a $(CXXFLAGS) $(DIFF_LINK) -o hdiffz
+	$(CC) hpatchz.c $(HPATCH_OBJ) $(CFLAGS) $(LDFLAGS) $(PATCH_LINK) -o hpatchz
+hdiffz: $(LIBRARYMAJOR)
+	$(CXX) hdiffz.cpp $(LIBRARYMAJOR) $(CXXFLAGS) $(LDFLAGS) $(DIFF_LINK) -o hdiffz
 
 ifeq ($(OS),Windows_NT) # mingw?
   RM := del /Q /F
@@ -368,17 +391,27 @@ else
   RM := rm -f
   DEL_HDIFF_OBJ := $(HDIFF_OBJ)
 endif
-INSTALL_X := install -m 0755
-INSTALL_BIN := $(DESTDIR)/usr/local/bin
+INSTALL_X := install -m 0755 -D
+INSTALL_F := install -m 0644 -D
+INSTALL_BIN := $(DESTDIR)/$(BINDIR)
+INSTALL_LIB := $(DESTDIR)/$(LIBDIR)
+INSTALL_INC := $(DESTDIR)/$(INCDIR)
 
 mostlyclean: hpatchz hdiffz
 	$(RM) $(DEL_HDIFF_OBJ)
 clean:
-	$(RM) libhdiffpatch.a hpatchz hdiffz $(DEL_HDIFF_OBJ)
+	$(RM) $(LIBRARYMAJOR) hpatchz hdiffz $(DEL_HDIFF_OBJ)
 
 install: all
 	$(INSTALL_X) hdiffz $(INSTALL_BIN)/hdiffz
 	$(INSTALL_X) hpatchz $(INSTALL_BIN)/hpatchz
+	$(INSTALL_X) $(LIBRARYMAJOR) $(INSTALL_LIB)/$(LIBRARYMAJOR)
+	ln -s $(LIBRARYMAJOR) $(INSTALL_LIB)/$(LIBRARYMINOR)
+	ln -s $(LIBRARYMINOR) $(INSTALL_LIB)/$(LIBRARY)
+	$(INSTALL_F) libHDiffPatch/HDiff/diff.h $(INSTALL_INC)/HDiffPatch/HDiff/diff.h
+	$(INSTALL_F) libHDiffPatch/HPatch/patch.h $(INSTALL_INC)/HDiffPatch/HPatch/patch.h
+	$(INSTALL_F) dirDiffPatch/dir_diff/dir_diff.h $(INSTALL_INC)/HDiffPatch/dir_diff/dir_diff.h
+	$(INSTALL_F) dirDiffPatch/dir_patch/dir_patch.h $(INSTALL_INC)/HDiffPatch/dir_patch/dir_patch.h
 
 uninstall:
 	$(RM)  $(INSTALL_BIN)/hdiffz  $(INSTALL_BIN)/hpatchz
-- 
2.44.0

