From 9ee5b902ed4f85c977664e9c928de8c64a95c2b1 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 17 Mar 2022 13:53:39 -0300
Subject: [PATCH] Makefile: package build fixes

---
 Makefile | 81 +++++++++++++++++++++++++++++++++++++++++---------------
 1 file changed, 59 insertions(+), 22 deletions(-)

diff --git a/Makefile b/Makefile
index 4049c2a..641af46 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 DIR_DIFF := 1
 MT       := 1
 # used libdeflate?
-LDEF     := 1
+LDEF     := 0
 # 0: not need zlib;  1: compile zlib source code;  2: used -lz to link zlib lib;
 ifeq ($(LDEF),0)
   ZLIB     := 2
@@ -10,16 +10,16 @@ else
   ZLIB     := 1
 endif
 # 0: not need lzma;  1: compile lzma source code;  2: used -llzma to link lzma lib;
-LZMA     := 1
+LZMA     := 2
 # lzma decompressor used arm64 asm optimize? 
 ARM64ASM := 0
 # lzma only can used software CRC? (no hardware CRC)
 USE_CRC_EMU := 0
 # supported atomic uint64?
 # 0: not need zstd;  1: compile zstd source code;  2: used -lzstd to link zstd lib;
-ZSTD     := 1
+ZSTD     := 2
 MD5      := 1
-XXH      := 1
+XXH      := 2
 # used clang?
 CL  	 := 0
 # build with -m32?
@@ -62,6 +62,15 @@ PIE :=0
 ATOMIC_U64 := 1
 
 
+PREFIX ?= /usr/local
+BINDIR ?= $(PREFIX)/bin
+LIBDIR ?= $(PREFIX)/lib
+INCDIR ?= $(PREFIX)/include
+
+LIBRARY = libhdiffpatch.so
+LIBRARYMINOR = $(LIBRARY)._RPM_MINOR_
+LIBRARYMAJOR = $(LIBRARYMINOR)._RPM_MAJOR_
+
 HDIFF_OBJ  := 
 HPATCH_OBJ := \
     libHDiffPatch/HPatch/patch.o \
@@ -94,7 +103,7 @@ else
 	HPATCH_OBJ += bsdiff_wrapper/bspatch_wrapper.o
 endif
 
-MD5_PATH := ../libmd5
+MD5_PATH := libmd5
 ifeq ($(DIR_DIFF),0)
 else
   ifeq ($(MD5),0)
@@ -103,7 +112,7 @@ else
   endif
 endif
 
-XXH_PATH := ../xxHash
+XXH_PATH := xxHash
 
 LZMA_PATH := ../lzma/C
 ifeq ($(LZMA),1)
@@ -138,6 +147,7 @@ else
   HDIFF_OBJ  += vcdiff_wrapper/vcdiff_wrapper.o
   ifeq ($(LZMA),0)
   else
+    ifeq ($(LZMA),1)
 	HPATCH_OBJ+=$(LZMA_PATH)/7zCrc.o \
 				$(LZMA_PATH)/7zCrcOpt.o \
 				$(LZMA_PATH)/Bra.o \
@@ -151,6 +161,7 @@ else
 				$(LZMA_PATH)/XzCrc64Opt.o \
 				$(LZMA_PATH)/XzDec.o
 	HDIFF_OBJ +=$(LZMA_PATH)/XzEnc.o
+    endif
   endif
 endif
 
@@ -266,7 +277,7 @@ UTEST_OBJ := \
     libhsync/sync_make/sync_make.o
 
 DEF_FLAGS := \
-    -O3 -DNDEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 \
+    -fPIC -DNDEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 \
     -D_IS_NEED_ALL_CompressPlugin=0 \
     -D_IS_NEED_DEFAULT_CompressPlugin=0 \
     -D_IS_NEED_ALL_ChecksumPlugin=0 \
@@ -298,6 +309,9 @@ else
 	ifeq ($(BZIP2),1)
         DEF_FLAGS += -I$(BZ2_PATH)
 	endif
+	ifeq ($(BZIP2),2)
+        DEF_FLAGS += $(shell pkg-config --cflags-only-I bzip2)
+	endif
 endif
 ifeq ($(ZLIB),0)
 else
@@ -305,6 +319,9 @@ else
 	ifeq ($(ZLIB),1)
     DEF_FLAGS += -I$(ZLIB_PATH)
 	endif
+	ifeq ($(ZLIB),2)
+        DEF_FLAGS += $(shell pkg-config --cflags-only-I zlib)
+	endif
 endif
 ifeq ($(LDEF),0)
 else
@@ -332,9 +349,13 @@ else
   else
     DEF_FLAGS += -D_ChecksumPlugin_md5 -I$(MD5_PATH)
   endif
-  ifeq ($(XXH),0)
-  else
-    DEF_FLAGS += -D_ChecksumPlugin_xxh3 -D_ChecksumPlugin_xxh128 -I$(XXH_PATH)
+
+  XXH_DEF_FLAGS = -D_ChecksumPlugin_xxh3 -D_ChecksumPlugin_xxh128
+  ifeq ($(XXH),1)
+    DEF_FLAGS += $(XXH_DEF_FLAGS) -I$(XXH_PATH)
+  endif
+  ifeq ($(XXH),2)
+    DEF_FLAGS += $(XXH_DEF_FLAGS) -I$(shell pkg-config --cflags-only-I libxxhash)
   endif
 endif
 ifeq ($(BSD),0)
@@ -366,6 +387,9 @@ else
       endif
     endif
   endif
+  ifeq ($(LZMA),2)
+    DEF_FLAGS += $(shell pkg-config --cflags-only-I lzmasdk-c)
+  endif
 endif
 ifeq ($(ZSTD),0)
 else
@@ -376,6 +400,9 @@ else
                  -DZSTD_LIB_DEPRECATED=0 -DZSTD_STRIP_ERROR_STRINGS=1 \
 	               -I$(ZSTD_PATH) -I$(ZSTD_PATH)/common -I$(ZSTD_PATH)/compress -I$(ZSTD_PATH)/decompress
 	endif
+	ifeq ($(ZSTD),2)
+      DEF_FLAGS += $(shell pkg-config --cflags-only-I libzstd)
+	endif
 endif
 
 ifeq ($(MT),0)
@@ -399,7 +426,7 @@ ifeq ($(ZSTD),2)
   PATCH_LINK += -lzstd		# link zstd
 endif
 ifeq ($(LZMA),2)
-  PATCH_LINK += -llzma		# link lzma
+  PATCH_LINK += -llzmasdk		# link lzmasdk
 endif
 ifeq ($(MT),0)
 else
@@ -437,18 +464,18 @@ CXXFLAGS += $(DEF_FLAGS) -std=c++11
 
 .PHONY: all install clean
 
-all: libhdiffpatch.a hpatchz hdiffz unit_test mostlyclean
+all: $(LIBRARYMAJOR) hpatchz hdiffz
 
 _ALL_OBJs := $(HDIFF_OBJ) $(UTEST_OBJ)
-libhdiffpatch.a: $(_ALL_OBJs)
-	$(AR) rcs $@ $^
+$(LIBRARYMAJOR): $(HDIFF_OBJ)
+	$(CXX) -o $(LIBRARYMAJOR) -shared -Wl,-soname=$(LIBRARYMINOR) -DNDEBUG $(LDFLAGS) -Wl,-z,defs $(HDIFF_OBJ) $(PATCH_LINK)
 
 hpatchz: $(HPATCH_OBJ)
-	$(CC) hpatchz.c $(HPATCH_OBJ) $(CFLAGS) $(PATCH_LINK) -o hpatchz
-hdiffz: libhdiffpatch.a
-	$(CXX) hdiffz.cpp libhdiffpatch.a $(CXXFLAGS) $(DIFF_LINK) -o hdiffz
-unit_test: libhdiffpatch.a 
-	$(CXX) ./test/unit_test.cpp libhdiffpatch.a $(DIFF_LINK) -o unit_test
+	$(CC) hpatchz.c $(HPATCH_OBJ) $(CFLAGS) $(LDFLAGS) $(PATCH_LINK) -o hpatchz
+hdiffz: $(LIBRARYMAJOR)
+	$(CXX) hdiffz.cpp $(LIBRARYMAJOR) $(CXXFLAGS) $(LDFLAGS) $(DIFF_LINK) -o hdiffz
+unit_test: $(LIBRARYMAJOR)
+	$(CXX) ./test/unit_test.cpp $(LIBRARYMAJOR) $(DIFF_LINK) -o unit_test
 
 ifeq ($(OS),Windows_NT) # mingw?
   RM := del /Q /F
@@ -457,17 +484,27 @@ else
   RM := rm -f
   DEL_ALL_OBJ := $(_ALL_OBJs)
 endif
-INSTALL_X := install -m 0755
-INSTALL_BIN := $(DESTDIR)/usr/local/bin
+INSTALL_X := install -m 0755 -D
+INSTALL_F := install -m 0644 -D
+INSTALL_BIN := $(DESTDIR)/$(BINDIR)
+INSTALL_LIB := $(DESTDIR)/$(LIBDIR)
+INSTALL_INC := $(DESTDIR)/$(INCDIR)
 
 mostlyclean: hpatchz hdiffz unit_test
 	$(RM) $(DEL_ALL_OBJ)
 clean:
-	$(RM) libhdiffpatch.a hpatchz hdiffz unit_test $(DEL_ALL_OBJ)
+	$(RM) $(LIBRARYMAJOR) hpatchz hdiffz unit_test $(DEL_ALL_OBJ)
 
 install: all
 	$(INSTALL_X) hdiffz $(INSTALL_BIN)/hdiffz
 	$(INSTALL_X) hpatchz $(INSTALL_BIN)/hpatchz
+	$(INSTALL_X) $(LIBRARYMAJOR) $(INSTALL_LIB)/$(LIBRARYMAJOR)
+	ln -s $(LIBRARYMAJOR) $(INSTALL_LIB)/$(LIBRARYMINOR)
+	ln -s $(LIBRARYMINOR) $(INSTALL_LIB)/$(LIBRARY)
+	$(INSTALL_F) libHDiffPatch/HDiff/diff.h $(INSTALL_INC)/HDiffPatch/HDiff/diff.h
+	$(INSTALL_F) libHDiffPatch/HPatch/patch.h $(INSTALL_INC)/HDiffPatch/HPatch/patch.h
+	$(INSTALL_F) dirDiffPatch/dir_diff/dir_diff.h $(INSTALL_INC)/HDiffPatch/dir_diff/dir_diff.h
+	$(INSTALL_F) dirDiffPatch/dir_patch/dir_patch.h $(INSTALL_INC)/HDiffPatch/dir_patch/dir_patch.h
 
 uninstall:
 	$(RM)  $(INSTALL_BIN)/hdiffz  $(INSTALL_BIN)/hpatchz
-- 
2.53.0

