diff -Naur claws-mail-4.1.1-orig/src/plugins/litehtml_viewer/litehtml/os_types.h claws-mail-4.1.1/src/plugins/litehtml_viewer/litehtml/os_types.h
--- claws-mail-4.1.1-orig/src/plugins/litehtml_viewer/litehtml/os_types.h	2021-09-16 12:20:13.000000000 +0200
+++ claws-mail-4.1.1/src/plugins/litehtml_viewer/litehtml/os_types.h	2023-01-16 18:08:23.201525654 +0100
@@ -1,6 +1,8 @@
 #ifndef LH_OS_TYPES_H
 #define LH_OS_TYPES_H
 
+#include <cstdint>
+
 namespace litehtml
 {
 #if defined( WIN32 ) || defined( _WIN32 ) || defined( WINCE )
