diff -Naur claws-mail-3.16.0-orig/configure.ac claws-mail-3.16.0/configure.ac
--- claws-mail-3.16.0-orig/configure.ac	2017-12-17 12:49:16.000000000 +0100
+++ claws-mail-3.16.0/configure.ac	2018-03-13 21:03:46.767302725 +0100
@@ -875,7 +875,7 @@
 dnl #######################################################################
 if test x"$enable_dbus" = xyes; then
 	if test x"$enable_networkmanager" = xyes; then
-		PKG_CHECK_MODULES(NETWORKMANAGER_SUPPORT, NetworkManager >= 0.6.2,
+		PKG_CHECK_MODULES(NETWORKMANAGER_SUPPORT, libnm,
 		[
 			AC_DEFINE(HAVE_NETWORKMANAGER_SUPPORT, 1, [Define if NetworkManager support is to be included.])
 			echo "Building with NetworkManager support"
diff -Naur claws-mail-3.16.0-orig/src/main.c claws-mail-3.16.0/src/main.c
--- claws-mail-3.16.0-orig/src/main.c	2017-12-17 12:48:41.000000000 +0100
+++ claws-mail-3.16.0/src/main.c	2018-03-13 15:04:00.029759255 +0100
@@ -152,9 +152,6 @@
 static gboolean went_offline_nm;
 #endif
 
-#if !defined(NM_CHECK_VERSION)
-#define NM_CHECK_VERSION(x,y,z) 0
-#endif
 
 #ifdef HAVE_DBUS_GLIB
 static DBusGProxy *awn_proxy = NULL;
@@ -1106,17 +1103,10 @@
 			"/org/freedesktop/NetworkManager",
 			"org.freedesktop.NetworkManager");
 		if (nm_proxy) {
-#if NM_CHECK_VERSION(0,8,992)
 			dbus_g_proxy_add_signal(nm_proxy, "StateChanged", G_TYPE_UINT, G_TYPE_INVALID);
 			dbus_g_proxy_connect_signal(nm_proxy, "StateChanged",
 				G_CALLBACK(networkmanager_state_change_cb),
 				NULL,NULL);
-#else
-			dbus_g_proxy_add_signal(nm_proxy, "StateChange", G_TYPE_UINT, G_TYPE_INVALID);
-			dbus_g_proxy_connect_signal(nm_proxy, "StateChange",
-				G_CALLBACK(networkmanager_state_change_cb),
-				NULL,NULL);
-#endif
 		}
 #endif
 		install_dbus_status_handler();
@@ -2971,14 +2961,9 @@
 		g_propagate_error(error, tmp_error);
 		return TRUE;
 	}
-#if NM_CHECK_VERSION(0,8,992)
     	return (state == NM_STATE_CONNECTED_LOCAL ||
 		state == NM_STATE_CONNECTED_SITE ||
 		state == NM_STATE_CONNECTED_GLOBAL ||
 		state == NM_STATE_UNKNOWN);
-#else
-    	return (state == NM_STATE_CONNECTED ||
-		state == NM_STATE_UNKNOWN);
-#endif
 }
 #endif
