From 5264d490bf1bdee972f1cc48b5eed4b43c056699 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 2 Aug 2023 17:53:52 -0300
Subject: [PATCH] Fix futex_waitv struct definition

---
 Utilities/sync.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Utilities/sync.h b/Utilities/sync.h
index 4e60c0f..952a12a 100644
--- a/Utilities/sync.h
+++ b/Utilities/sync.h
@@ -47,12 +47,6 @@ constexpr NTSTATUS NTSTATUS_TIMEOUT = 0x102;
 
 #ifndef FUTEX_32
 #define FUTEX_32 2
-#endif
-
-#ifndef FUTEX_WAITV_MAX
-#define FUTEX_WAITV_MAX 128
-#endif
-
 struct futex_waitv
 {
 	__u64 val;
@@ -60,6 +54,12 @@ struct futex_waitv
 	__u32 flags;
 	__u32 __reserved;
 };
+#endif
+
+#ifndef FUTEX_WAITV_MAX
+#define FUTEX_WAITV_MAX 128
+#endif
+
 #else
 enum
 {
-- 
2.41.0

