From ca7708824efe025b3c2d47051b26aec9c1970ed5 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 9 Feb 2022 16:11:12 -0300
Subject: [PATCH] Use system libraries

---
 3rdparty/CMakeLists.txt | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 3d84d3f..4893749 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -73,7 +73,15 @@ endif()
 
 
 # hidapi
-add_subdirectory(hidapi)
+find_package(hidapi)
+if(hidapi_FOUND)
+	message(STATUS "Using the system hidapi")
+	add_library(3rdparty_hidapi INTERFACE)
+	target_include_directories(3rdparty_hidapi SYSTEM INTERFACE ${HIDAPI_INCLUDEDIR})
+	target_link_libraries(3rdparty_hidapi INTERFACE hidapi::hidraw)
+else()
+	add_subdirectory(hidapi)
+endif()
 
 
 # Vulkan
@@ -134,7 +142,15 @@ target_include_directories(3rdparty_stblib INTERFACE stblib/include)
 add_subdirectory(discord-rpc)
 
 # Cubeb
-add_subdirectory(cubeb EXCLUDE_FROM_ALL)
+find_package(cubeb)
+if(cubeb_FOUND)
+	message(STATUS "Using the system cubeb")
+	add_library(cubeb INTERFACE)
+	target_link_libraries(cubeb INTERFACE cubeb::cubeb)
+	add_library(3rdparty::cubeb ALIAS cubeb)
+else()
+	add_subdirectory(cubeb EXCLUDE_FROM_ALL)
+endif()
 
 # SoundTouch
 add_subdirectory(SoundTouch EXCLUDE_FROM_ALL)
@@ -234,6 +250,7 @@ if(USE_FAUDIO)
 			message(STATUS "RPCS3: Using system FAudio")
 			find_package(FAudio REQUIRED CONFIGS FAudioConfig.cmake FAudio-config.cmake)
 			add_library(3rdparty_FAudio INTERFACE)
+			add_library(FAudio ALIAS FAudio::FAudio)
 			target_link_libraries(3rdparty_FAudio INTERFACE FAudio)
 			target_compile_definitions(3rdparty_FAudio INTERFACE -DHAVE_FAUDIO)
 			set(FAUDIO_TARGET 3rdparty_FAudio)
-- 
2.37.1

