From 953b524d3cc7391ee4d5dcd401f83e5c745c5f83 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 16 Mar 2023 17:25:53 -0300
Subject: [PATCH] llvm: gcc 13 build fix

---
 include/llvm/Support/Signals.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/llvm/Support/Signals.h b/include/llvm/Support/Signals.h
index 44f5a75..ccb429b 100644
--- a/include/llvm/Support/Signals.h
+++ b/include/llvm/Support/Signals.h
@@ -15,6 +15,7 @@
 #define LLVM_SUPPORT_SIGNALS_H
 
 #include <string>
+#include <cstdint>
 
 namespace llvm {
 class StringRef;
-- 
2.40.0

