From a4124de961cb1a4e4a6775f61d8ec5f5c2cc61c1 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 8 Dec 2022 23:33:46 -0300
Subject: [PATCH] ffmpeg 5.1.2 fix

---
 rpcs3/util/media_utils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rpcs3/util/media_utils.cpp b/rpcs3/util/media_utils.cpp
index 94a0db3..1ed259b 100644
--- a/rpcs3/util/media_utils.cpp
+++ b/rpcs3/util/media_utils.cpp
@@ -699,7 +699,7 @@ namespace utils
 			AVCodecID used_codec = static_cast<AVCodecID>(m_video_codec_id);
 
 			// Find specified codec first
-			if (AVCodec* encoder = avcodec_find_encoder(used_codec); !!encoder)
+			if (const AVCodec* encoder = avcodec_find_encoder(used_codec); !!encoder)
 			{
 				media_log.success("video_encoder: Found requested video_codec %d = %s", static_cast<int>(used_codec), encoder->name);
 				av_output_format = find_format(encoder);
@@ -724,7 +724,7 @@ namespace utils
 				void* opaque = nullptr;
 				for (const AVCodec* codec = av_codec_iterate(&opaque); !!codec; codec = av_codec_iterate(&opaque))
 				{
-					if (AVCodec* encoder = avcodec_find_encoder(codec->id); !!encoder)
+					if (const AVCodec* encoder = avcodec_find_encoder(codec->id); !!encoder)
 					{
 						media_log.notice("video_encoder: Found video_codec %d = %s", static_cast<int>(used_codec), encoder->name);
 						av_output_format = find_format(encoder);
-- 
2.38.1

