From 2bbf6451dcafce7f030e347880b5744d9b35727e Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 19 Mar 2025 18:40:03 -0300
Subject: [PATCH] glslang: gcc 15 build fix

---
 3rdparty/glslang/glslang/SPIRV/SpvBuilder.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/3rdparty/glslang/glslang/SPIRV/SpvBuilder.h b/3rdparty/glslang/glslang/SPIRV/SpvBuilder.h
index 2e1c07d..f822f94 100644
--- a/3rdparty/glslang/glslang/SPIRV/SpvBuilder.h
+++ b/3rdparty/glslang/glslang/SPIRV/SpvBuilder.h
@@ -60,6 +60,7 @@ namespace spv {
 #include <memory>
 #include <set>
 #include <sstream>
+#include <cstdint>
 #include <stack>
 #include <unordered_map>
 #include <map>
-- 
2.49.0

