From 699e3c2a8c5c282986cd63b5d1d8a399932fb047 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 19 Nov 2019 09:51:03 -0300
Subject: [PATCH] Remove some ugly paddings

---
 style/adwaita.h | 41 +++++++++++++++++++++--------------------
 1 file changed, 21 insertions(+), 20 deletions(-)

diff --git a/style/adwaita.h b/style/adwaita.h
index 09b970a..3a396f7 100644
--- a/style/adwaita.h
+++ b/style/adwaita.h
@@ -137,49 +137,50 @@ namespace Adwaita
         // line editors
         LineEdit_FrameWidth = 3,
         LineEdit_MarginHeight = 2,
-        LineEdit_MarginWidth = 8,
-        LineEdit_MinHeight = 36,
+        LineEdit_MarginWidth = 6,
+        LineEdit_MinHeight = 30,
         LineEdit_MinWidth = 80,
 
         // menu items
         Menu_FrameWidth = 0,
-        MenuItem_MarginWidth = 4,
-        MenuItem_ItemSpacing = 6,
+        MenuItem_MarginWidth = 3,
+        MenuItem_ItemSpacing = 3,
         MenuItem_AcceleratorSpace = 16,
         MenuButton_IndicatorWidth = 20,
 
         // combobox
-        ComboBox_FrameWidth = 4,
-        ComboBox_MarginHeight = 4,
-        ComboBox_MarginWidth = 8,
-        ComboBox_MinHeight = 36,
+        ComboBox_FrameWidth = 3,
+        ComboBox_MarginHeight = 3,
+        ComboBox_MarginWidth = 6,
+        ComboBox_MinHeight = 30,
         ComboBox_MinWidth = 80,
 
         // spinbox
         SpinBox_FrameWidth = LineEdit_FrameWidth,
-        SpinBox_ArrowButtonWidth = 20,
-        SpinBox_MinHeight = 36,
+        SpinBox_ArrowButtonWidth = 16,
+        SpinBox_MinHeight = 30,
         SpinBox_MinWidth = 80,
-        SpinBox_MarginHeight = 4,
-        SpinBox_MarginWidth = 8,
+        SpinBox_MarginHeight = 3,
+        SpinBox_MarginWidth = 6,
 
         // groupbox title margin
         GroupBox_TitleMarginWidth = 4,
 
         // buttons
-        Button_MinHeight = 36,
+        Button_FrameWidth = 3,
+        Button_MinHeight = 30,
         Button_MinWidth = 80,
-        Button_MarginHeight = 4,
-        Button_MarginWidth = 8,
+        Button_MarginHeight = 3,
+        Button_MarginWidth = 6,
         Button_ItemSpacing = 4,
 
         // tool buttons
         ToolButton_MarginWidth = 6,
-        ToolButton_ItemSpacing = 4,
+        ToolButton_ItemSpacing = 5,
         ToolButton_InlineIndicatorWidth = 12,
 
         // checkboxes and radio buttons
-        CheckBox_Size = 20,
+        CheckBox_Size = 16,
         CheckBox_FocusMarginWidth = 3,
         CheckBox_ItemSpacing = 4,
 
@@ -196,7 +197,7 @@ namespace Adwaita
         ScrollBar_DoubleButtonHeight = 0,
 
         // toolbars
-        ToolBar_FrameWidth = 2,
+        ToolBar_FrameWidth = 0,
         ToolBar_HandleExtent = 10,
         ToolBar_HandleWidth = 6,
         ToolBar_SeparatorWidth = 8,
@@ -218,10 +219,10 @@ namespace Adwaita
         Slider_ControlThickness = 24,
 
         // tabbar
-        TabBar_TabMarginHeight = 9,
+        TabBar_TabMarginHeight = 6,
         TabBar_TabMarginWidth = 8,
         TabBar_TabMinWidth = 80,
-        TabBar_TabMinHeight = 36,
+        TabBar_TabMinHeight = 32,
         TabBar_TabItemSpacing = 8,
         TabBar_TabOverlap = 1,
         TabBar_BaseOverlap = 0,
-- 
2.23.0

