From a49ea43bc2e7d1f9b28546ef21682c740f3380e5 Mon Sep 17 00:00:00 2001
From: Jacek Caban <jacek@codeweavers.com>
Date: Mon, 12 Feb 2018 23:01:28 +0100
Subject: [PATCH 1/7] Use make instead of mach in make_package.

---
 wine/make_package | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wine/make_package b/wine/make_package
index 4f105adc83c5..3269d44ad9c2 100755
--- a/wine/make_package
+++ b/wine/make_package
@@ -142,7 +142,7 @@ then
         MOZCONFIG=$build_dir/mozconfig ./mach configure
     fi
 
-    MOZCONFIG=$build_dir/mozconfig ./mach build
+    MOZCONFIG=$build_dir/mozconfig make $MAKEOPTS -C $build_dir
 fi
 
 echo "make_package: Preparing Wine Gecko files..."
@@ -150,7 +150,7 @@ rm -rf $build_dir/dist/wine_gecko
 
 if [ "$no_xulinstaller" != "yes" ]
 then
-    MOZCONFIG=$build_dir/mozconfig ./mach build browser/installer
+    MOZCONFIG=$build_dir/mozconfig make $MAKEOPTS -C $build_dir/browser/installer
 fi
 
 cp -r $build_dir/dist/firefox $build_dir/dist/wine_gecko
-- 
2.21.0

