From 4548fd0f533f233606ec66dd2ca1731744be73bc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 12 Nov 2023 20:04:07 -0300
Subject: [PATCH] Fix update settings

---
 vita3k/gui/src/settings_dialog.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vita3k/gui/src/settings_dialog.cpp b/vita3k/gui/src/settings_dialog.cpp
index a7349f8..727dcfe 100644
--- a/vita3k/gui/src/settings_dialog.cpp
+++ b/vita3k/gui/src/settings_dialog.cpp
@@ -1016,8 +1016,10 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::Checkbox(lang.emulator["log_compat_warn"].c_str(), &emuenv.cfg.log_compat_warn);
         SetTooltipEx(lang.emulator["log_compat_warn_description"].c_str());
         ImGui::Spacing();
+#ifdef USE_VITA3K_UPDATE
         ImGui::Checkbox(lang.emulator["check_for_updates"].c_str(), &emuenv.cfg.check_for_updates);
         SetTooltipEx(lang.emulator["check_for_updates_description"].c_str());
+#endif
         ImGui::Spacing();
         ImGui::SliderInt(lang.emulator["file_loading_delay"].c_str(), &config.file_loading_delay, 0, 30, "%d ms", ImGuiSliderFlags_AlwaysClamp);
         SetTooltipEx(lang.emulator["file_loading_delay_description"].c_str());
-- 
2.52.0

