From 691c4da5fca86d8b2a3ec635f1e381ce3f2c2999 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 24 Mar 2024 23:39:45 -0300
Subject: [PATCH] Disable ffmpeg download

---
 ffmpeg/CMakeLists.txt | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/ffmpeg/CMakeLists.txt b/ffmpeg/CMakeLists.txt
index 6c78b0c..d72e154 100644
--- a/ffmpeg/CMakeLists.txt
+++ b/ffmpeg/CMakeLists.txt
@@ -36,32 +36,6 @@ else ()
 	message(FATAL_ERROR "Unsupported architecture.")
 endif()
 
-# Compute current short git commit SHA
-execute_process(
-	COMMAND git rev-parse --short HEAD
-	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-	OUTPUT_VARIABLE FFMPEG_GIT_SHA
-	OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-# Download prebuilt ffmpeg
-if(NOT EXISTS "${CMAKE_BINARY_DIR}/external/ffmpeg.zip")
-message(STATUS "Downloading FFMPEG prebuilt ${FFMPEG_PREBUILTS_NAME} for commit ${FFMPEG_GIT_SHA}...")
-	file(DOWNLOAD https://github.com/Vita3K/ffmpeg-core/releases/download/${FFMPEG_GIT_SHA}/${FFMPEG_PREBUILTS_NAME}
-		"${CMAKE_BINARY_DIR}/external/ffmpeg.zip" SHOW_PROGRESS
-		STATUS FILE_STATUS)
-	list(GET FILE_STATUS 0 STATUS_CODE)
-	if (NOT STATUS_CODE EQUAL 0)
-		file(REMOVE "${CMAKE_BINARY_DIR}/external/ffmpeg.zip") # CMake create 0 byte file even if URL is invalid. So need to delete it.
-		message(FATAL_ERROR "No FFMPEG prebuilt found with corresponding commit SHA (${FFMPEG_GIT_SHA})")
-	endif()
-endif()
-
-if(NOT EXISTS "${CMAKE_BINARY_DIR}/external/ffmpeg/lib")
-	file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/external/ffmpeg/lib")
-	execute_process(COMMAND ${CMAKE_COMMAND} -E tar xzf "${CMAKE_BINARY_DIR}/external/ffmpeg.zip"
-		WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/external/ffmpeg/lib")
-endif()
-
 set(LIB_PREFIX "lib")
 set(LIB_EXT "a")
 if (WIN32 AND NOT MINGW)
-- 
2.52.0

