From de8c33a1e1c1583307896509e0092bc9fbfbd736 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 19 Apr 2019 02:20:07 -0300
Subject: [PATCH] Add soversion to library

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea48923..fcdb11d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,6 +111,11 @@ configure_package_config_file(
   INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
 )
 
+set_target_properties(cubeb PROPERTIES
+  SOVERSION 0
+  VERSION 0.0.0
+)
+
 install(TARGETS cubeb
   EXPORT "${PROJECT_NAME}Targets"
   DESTINATION ${CMAKE_INSTALL_PREFIX}
-- 
2.21.0

