--- steam/steam.orig	2017-01-04 19:08:38.510756913 -0200
+++ steam/steam	2017-01-04 19:09:42.958122555 -0200
@@ -19,6 +19,11 @@ export TEXTDOMAIN=steam
 
 # Make "X" window button close the program instead of minimizing like "_"
 export STEAM_FRAME_FORCE_CLOSE=1
+ 
+# check if pulseaudio is installed and if it isn't, use alsa for SDL_AUDIODRIVER
+if ! pulseaudio --check &>/dev/null; then
+    export SDL_AUDIODRIVER=${SDL_AUDIODRIVER:-alsa}
+fi
 
 function show_message()
 {
