--- _upstream/steam/steam-fdist/steam.spec	2017-01-10 23:12:22.939615013 -0200
+++ steam/steam.spec	2017-01-10 23:16:31.397873195 -0200
@@ -6,7 +6,8 @@
 
 Name:           steam
 Version:        1.0.0.54
-Release:        3%{?dist}
+Epoch:          1
+Release:        4.chinfo%{?dist}
 Summary:        Installer for the Steam software distribution service
 # Redistribution and repackaging for Linux is allowed, see license file
 License:        Steam License Agreement
@@ -46,6 +47,12 @@ Patch2:         %{name}-controller-gamep
 # Make "X" window button close the program instead of minimizing like "_"
 Patch3:         %{name}-3210.patch
 
+# Disable desktop files installation on desktop and create logs on user directory
+Patch4:         %{name}-launcher.patch
+
+# From Arch: Set alsa SDL audiodriver if pulseaudio is not installed
+Patch5:         alsa_sdl_audiodriver.patch
+
 BuildRequires:  desktop-file-utils
 BuildRequires:  systemd
 
@@ -70,6 +77,7 @@ Requires:       libXScrnSaver%{?_isa}
 Requires:       mesa-libGL%{?_isa}
 Requires:       nss%{?_isa}
 Requires:       pulseaudio-libs%{?_isa}
+Requires:       SDL2%{?_isa}
 
 # Required for sending out crash reports to Valve
 Requires:       libcurl%{?_isa}
@@ -106,6 +114,8 @@ savegame and screenshot functionality, a
 %patch1 -p1
 %patch2 -p1
 %patch3 -p1
+%patch4 -p1
+%patch5 -p1
 
 sed -i 's/\r$//' %{name}.desktop
 sed -i 's/\r$//' steam_install_agreement.txt
@@ -125,7 +135,13 @@ mkdir -p %{buildroot}%{_udevrulesdir}/
 install -m 644 -p lib/udev/rules.d/* \
     %{SOURCE8} %{SOURCE9} %{buildroot}%{_udevrulesdir}/
 
-desktop-file-validate %{buildroot}/%{_datadir}/applications/%{name}.desktop
+desktop-file-edit \
+  --remove-category="Network" \
+  --remove-category="FileTransfer" \
+  %{buildroot}/%{_datadir}/applications/%{name}.desktop
+
+ln -sf ../icons/hicolor/48x48/apps/steam.png \
+  %{buildroot}/%{_datadir}/pixmaps/%{name}_tray_mono.png
 
 install -D -m 644 -p %{SOURCE3} \
     %{buildroot}%{_prefix}/lib/firewalld/services/steam.xml
@@ -173,10 +189,20 @@ fi
 %{_udevrulesdir}/*
 
 %changelog
-* Sun Jan 08 2017 Simone Caronni <negativo17@gmail.com> - 1.0.0.54-3
+* Wed Jan 11 2017 Phantom X <megaphantomx at bol dot com dot br> - 1:1.0.0.54-4
+- Sync with rpmfusion.
+- Sun Jan 08 2017 Simone Caronni <negativo17@gmail.com> - 1.0.0.54-3
 - Microsoft keyboards have been fixed in kernel 4.9 and backported to other
   kernels.
 
+* Wed Jan 04 2017 Phantom X <megaphantomx at bol dot com dot br> - 1.0.0.54-3
+- Patches updates.
+- Disable desktop files installation on desktop.
+- Create logs on user directory.
+- R: SDL2
+- Remove extraneous categories from desktop file.
+- Use the good colorful notification icon.
+
 * Tue Dec 13 2016 Simone Caronni <negativo17@gmail.com> - 1.0.0.54-2
 - Re-add close functionality to X window button (#3210).
 
