From 1cda4699c9ae575dc5581b1470ee9fe3a0b83fa8 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 19 Dec 2020 16:44:01 -0300
Subject: [PATCH] format-security

---
 src/gui/sdl_ttf.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/gui/sdl_ttf.c b/src/gui/sdl_ttf.c
index fd00f26..00047c0 100644
--- a/src/gui/sdl_ttf.c
+++ b/src/gui/sdl_ttf.c
@@ -331,7 +331,7 @@ static void TTF_SetFTError(const char *msg, FT_Error error)
 	sprintf(buffer, "%s: %s", msg, err_msg);
 	TTF_SetError(buffer);
 #else
-	TTF_SetError(msg);
+	TTF_SetError("%s", msg);
 #endif /* USE_FREETYPE_ERRORS */
 }
 
@@ -533,7 +533,6 @@ TTF_Font* TTF_OpenFontIndex( const char *file, int ptsize, long index )
 {
 	SDL_RWops *rw = SDL_RWFromFile(file, "rb");
 	if ( rw == NULL ) {
-		TTF_SetError(SDL_GetError());
 		return NULL;
 	}
 	return TTF_OpenFontIndexRW(rw, 1, ptsize, index);
@@ -2093,4 +2092,4 @@ int TTF_GetFontKerningSize(TTF_Font* font, int prev_index, int index)
 	FT_Vector delta; 
 	FT_Get_Kerning( font->face, prev_index, index, ft_kerning_default, &delta ); 
 	return (delta.x >> 6);
-}
\ No newline at end of file
+}
-- 
2.29.2

