From adfad030e0ddfaefdba401835557d94a429a16c1 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 21 Feb 2021 10:16:49 -0300
Subject: [PATCH] Use system xxhash

---
 Makefile | 7 +++----
 jdupes.c | 2 +-
 jdupes.h | 2 +-
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index afebf3e..2b1eed0 100644
--- a/Makefile
+++ b/Makefile
@@ -152,20 +152,19 @@ INSTALL_DATA    = $(INSTALL) -m 0644
 OBJS += jdupes.o jody_paths.o jody_sort.o jody_win_unicode.o jody_strtoepoch.o string_malloc.o oom.o
 OBJS += jody_cacheinfo.o
 OBJS += act_deletefiles.o act_linkfiles.o act_printmatches.o act_summarize.o act_printjson.o
-OBJS += xxhash.o
 OBJS += $(ADDITIONAL_OBJECTS)
 
 all: $(PROGRAM_NAME)
 
 static: $(PROGRAM_NAME)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
 
 static_stripped: $(PROGRAM_NAME)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
 	strip $(PROGRAM_NAME)
 
 $(PROGRAM_NAME): $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
 
 winres.o: winres.rc winres.manifest.xml
 	./tune_winres.sh
diff --git a/jdupes.c b/jdupes.c
index 6b3737e..c4baefd 100644
--- a/jdupes.c
+++ b/jdupes.c
@@ -42,7 +42,7 @@
 #include <time.h>
 #include <sys/time.h>
 #include "jdupes.h"
-#include "xxhash.h"
+#include <xxhash.h>
 #include "oom.h"
 #ifdef ENABLE_DEDUPE
 #include <sys/utsname.h>
diff --git a/jdupes.h b/jdupes.h
index 38c0469..d6606dd 100644
--- a/jdupes.h
+++ b/jdupes.h
@@ -32,7 +32,7 @@ extern "C" {
 #include "jody_sort.h"
 #include "version.h"
 
-#include "xxhash.h"
+#include <xxhash.h>
 
 /* Set hash type (change this if swapping in a different hash function) */
  typedef XXH64_hash_t jdupes_hash_t;
-- 
2.29.2

