From 90eb703a9d390324eb260712e655ea758ca8da6e Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 14 Jan 2020 14:45:08 -0300
Subject: [PATCH] Use system xxhash

---
 Makefile | 3 +--
 jdupes.c | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 3296b77..2d54b88 100644
--- a/Makefile
+++ b/Makefile
@@ -138,7 +138,6 @@ INSTALL_DATA    = $(INSTALL) -m 0644
 OBJS += jdupes.o jody_paths.o jody_sort.o jody_win_unicode.o string_malloc.o
 OBJS += jody_cacheinfo.o
 OBJS += act_deletefiles.o act_linkfiles.o act_printmatches.o act_summarize.o act_printjson.o
-OBJS += xxhash.o
 OBJS += $(ADDITIONAL_OBJECTS)
 
 OBJS_CLEAN += jdupes-standalone
@@ -146,7 +145,7 @@ OBJS_CLEAN += jdupes-standalone
 all: $(PROGRAM_NAME)
 
 $(PROGRAM_NAME): $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
 
 winres.o : winres.rc winres.manifest.xml
 	./tune_winres.sh
diff --git a/jdupes.c b/jdupes.c
index 0292f04..3426cb8 100644
--- a/jdupes.c
+++ b/jdupes.c
@@ -41,7 +41,7 @@
 #include <sys/time.h>
 #include "jdupes.h"
 #include "string_malloc.h"
-#include "xxhash.h"
+#include <xxhash.h>
 #include "jody_sort.h"
 #include "jody_win_unicode.h"
 #include "jody_cacheinfo.h"
-- 
2.24.1

