From 445a79c325649a19fd3c60a1125269caf8b0bb73 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 21 Feb 2021 10:16:49 -0300
Subject: [PATCH] Use system xxhash

---
 Makefile | 8 ++++----
 jdupes.c | 2 +-
 jdupes.h | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index bc08be2..3cafc6e 100644
--- a/Makefile
+++ b/Makefile
@@ -182,20 +182,20 @@ INSTALL_DATA    = $(INSTALL) -m 0644
 OBJS += jdupes.o jody_paths.o jody_sort.o jody_win_unicode.o jody_strtoepoch.o string_malloc.o oom.o
 OBJS += jody_cacheinfo.o
 OBJS += act_deletefiles.o act_linkfiles.o act_printmatches.o act_summarize.o act_printjson.o
-OBJS += xxhash.o jody_hash.o
+OBJS += jody_hash.o
 OBJS += $(ADDITIONAL_OBJECTS)
 
 all: $(PROGRAM_NAME)
 
 static: $(PROGRAM_NAME)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
 
 static_stripped: $(PROGRAM_NAME)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS) -static
 	strip $(PROGRAM_NAME)
 
 $(PROGRAM_NAME): $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
+	$(CC) $(CFLAGS) -lxxhash $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
 
 winres.o: winres.rc winres.manifest.xml
 	./tune_winres.sh
diff --git a/jdupes.c b/jdupes.c
index b1faa1a..fbc98c1 100644
--- a/jdupes.c
+++ b/jdupes.c
@@ -42,7 +42,7 @@
 #include <sys/time.h>
 #include "jdupes.h"
 #ifndef USE_JODY_HASH
- #include "xxhash.h"
+ #include <xxhash.h>
 #else
  #include "jody_hash.h"
 #endif
diff --git a/jdupes.h b/jdupes.h
index 3b435a9..6578bc4 100644
--- a/jdupes.h
+++ b/jdupes.h
@@ -32,7 +32,7 @@ extern "C" {
 #include "version.h"
 
 #ifndef USE_JODY_HASH
-#include "xxhash.h"
+#include <xxhash.h>
 #else
 #include "jody_hash.h"
 #endif /* USE_JODY_HASH */
-- 
2.39.2

