From 2580ca4c7c0818c2a2b2fb8c8c271a5ffed59bcd Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 19 Aug 2023 22:41:05 -0300
Subject: [PATCH] log.h: ignore format-security

---
 src/common/log.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/common/log.h b/src/common/log.h
index d68b556..cec3d96 100644
--- a/src/common/log.h
+++ b/src/common/log.h
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: (GPL-3.0 OR CC-BY-NC-ND-4.0)
 
 #pragma once
+#pragma GCC diagnostic ignored "-Wformat-security"
 #include "types.h"
 #include <cinttypes>
 #include <cstdarg>
-- 
2.41.0

