From cbfb05928d817fe24a81e3e56469f1687ebe6bd9 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 25 Jun 2024 22:40:39 -0300
Subject: [PATCH] cmake: versioned spirv-cross-c-shared

---
 src/util/gpu_device.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/util/gpu_device.cpp b/src/util/gpu_device.cpp
index ef4d5d6..9d7afbf 100644
--- a/src/util/gpu_device.cpp
+++ b/src/util/gpu_device.cpp
@@ -1321,8 +1321,7 @@ bool dyn_libs::OpenSpirvCross(Error* error)
   // SPVC's build on Windows doesn't spit out a versioned DLL.
   const std::string libname = DynamicLibrary::GetVersionedFilename("spirv-cross-c-shared");
 #else
-  const std::string libname = DynamicLibrary::GetVersionedFilename("spirv-cross-c-shared", SPVC_C_API_VERSION_MAJOR,
-                                                                   SPVC_C_API_VERSION_MINOR, SPVC_C_API_VERSION_PATCH);
+  const std::string libname = DynamicLibrary::GetVersionedFilename("spirv-cross-c-shared", SPVC_C_API_VERSION_MAJOR);
 #endif
   if (!s_spirv_cross_library.Open(libname.c_str(), error))
   {
-- 
2.45.2

