From 68d22a320fadc21c6d9afe0a02aea1cdc9191e56 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 30 Aug 2023 22:52:57 -0300
Subject: [PATCH] Fix build without discord presence support

---
 src/core/system.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/system.cpp b/src/core/system.cpp
index e7025f1..a4b7cd2 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -3683,6 +3683,7 @@ void System::CheckForSettingsChanges(const Settings& old_settings)
 
   FullscreenUI::CheckForConfigChanges(old_settings);
 
+#ifdef WITH_DISCORD_PRESENCE
   if (g_settings.enable_discord_presence != old_settings.enable_discord_presence)
   {
     if (g_settings.enable_discord_presence)
@@ -3690,6 +3691,7 @@ void System::CheckForSettingsChanges(const Settings& old_settings)
     else
       ShutdownDiscordPresence();
   }
+#endif
 
   if (g_settings.log_level != old_settings.log_level || g_settings.log_filter != old_settings.log_filter ||
       g_settings.log_timestamps != old_settings.log_timestamps ||
-- 
2.41.0

