From 14fabbb8d45222eda0d95fb25f3e6e48355de208 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 22 Aug 2022 22:53:43 -0300
Subject: [PATCH] Set datadir to RPM packaging

---
 src/duckstation-qt/qthost.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/duckstation-qt/qthost.cpp b/src/duckstation-qt/qthost.cpp
index 8f64c0e..79f807a 100644
--- a/src/duckstation-qt/qthost.cpp
+++ b/src/duckstation-qt/qthost.cpp
@@ -550,7 +550,7 @@ void QtHost::SetResourcesDirectory()
 {
 #ifndef __APPLE__
   // On Windows/Linux, these are in the binary directory.
-  EmuFolders::Resources = Path::Combine(EmuFolders::AppRoot, "resources");
+  EmuFolders::Resources = "_RPM_DATADIR_/resources";
 #else
   // On macOS, this is in the bundle resources directory.
   EmuFolders::Resources = Path::Canonicalize(Path::Combine(EmuFolders::AppRoot, "../Resources"));
@@ -2272,14 +2272,14 @@ void QtHost::UpdateApplicationLanguage(QWidget* dialog_parent)
 
   // install the base qt translation first
 #ifndef __APPLE__
-  const QString base_dir = QStringLiteral("%1/translations").arg(qApp->applicationDirPath());
+  const QString base_dir = QStringLiteral("_RPM_DATADIR_/translations");
 #else
   const QString base_dir = QStringLiteral("%1/../Resources/translations").arg(qApp->applicationDirPath());
 #endif
 
   // Qt base uses underscores instead of hyphens.
   const QString qtbase_language = QString(qlanguage).replace(QChar('-'), QChar('_'));
-  QString base_path(QStringLiteral("%1/qt_%2.qm").arg(base_dir).arg(qtbase_language));
+  QString base_path(QStringLiteral("_RPM_QTTDIR_/qt_%1.qm").arg(qtbase_language));
   bool has_base_ts = QFile::exists(base_path);
   if (!has_base_ts)
   {
@@ -2287,7 +2287,7 @@ void QtHost::UpdateApplicationLanguage(QWidget* dialog_parent)
     const int index = qlanguage.lastIndexOf('-');
     if (index > 0)
     {
-      base_path = QStringLiteral("%1/qt_%2.qm").arg(base_dir).arg(qlanguage.left(index));
+      base_path = QStringLiteral("_RPM_QTTDIR_/qt_%1.qm").arg(qlanguage.left(index));
       has_base_ts = QFile::exists(base_path);
     }
   }
-- 
2.50.1

