From bb98d07dc64bbfb49af97cd422e46db33d4ab336 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 4 Jul 2024 18:50:51 -0300
Subject: [PATCH] cmake: versioned discord-rpc

---
 src/core/system.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/system.cpp b/src/core/system.cpp
index c015472..e559c1b 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -5486,7 +5486,7 @@ bool dyn_libs::OpenDiscordRPC(Error* error)
   if (s_discord_rpc_library.IsOpen())
     return true;
 
-  const std::string libname = DynamicLibrary::GetVersionedFilename("discord-rpc");
+  const std::string libname = DynamicLibrary::GetVersionedFilename("discord-rpc", 0);
   if (!s_discord_rpc_library.Open(libname.c_str(), error))
   {
     Error::AddPrefix(error, "Failed to load discord-rpc: ");
-- 
2.45.2

