From cd87df1215248e1659b98401fd92cab36c87cd5a Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 3 Sep 2023 12:05:58 -0300
Subject: [PATCH] Lower the SDL2 requirement a bit

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d0cc1b..f3f9632 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,7 +72,7 @@ endif()
 
 # Required libraries.
 if(USE_SDL2)
-  find_package(SDL2 2.28.2 REQUIRED)
+  find_package(SDL2 2.26.3 REQUIRED)
 endif()
 if(NOT WIN32 AND NOT ANDROID)
   find_package(CURL REQUIRED)
-- 
2.41.0

