From 83cf53976fbfd59a24dd29673acc1e39f7b7d193 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 13 Jul 2025 11:44:24 -0300
Subject: [PATCH] vulkan_device.cpp: replace VK_DRIVER_ID_MESA_AGXV

---
 src/util/vulkan_device.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/vulkan_device.cpp b/src/util/vulkan_device.cpp
index 31fa117..ae25697 100644
--- a/src/util/vulkan_device.cpp
+++ b/src/util/vulkan_device.cpp
@@ -2570,7 +2570,7 @@ GPUDriverType VulkanDevice::GuessDriverType(const VkPhysicalDeviceProperties& de
     {VK_DRIVER_ID_MESA_DOZEN, GPUDriverType::DozenMesa},
     {VK_DRIVER_ID_MESA_NVK, GPUDriverType::NVIDIAMesa},
     {VK_DRIVER_ID_IMAGINATION_OPEN_SOURCE_MESA, GPUDriverType::ImaginationMesa},
-    {VK_DRIVER_ID_MESA_AGXV, GPUDriverType::AppleMesa},
+    {VK_DRIVER_ID_MESA_HONEYKRISP, GPUDriverType::AppleMesa},
   };
 
   const auto iter = std::find_if(std::begin(table), std::end(table), [&driver_properties](const auto& it) {
-- 
2.50.1

