From f6885dccb8a65534fd7e3dee5303e711dea09dae Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 13 Nov 2022 12:06:55 -0300
Subject: [PATCH] Use system libraries

---
 thirdparty/GNUmakefile | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/thirdparty/GNUmakefile b/thirdparty/GNUmakefile
index 5390f9f..15da9cd 100644
--- a/thirdparty/GNUmakefile
+++ b/thirdparty/GNUmakefile
@@ -6,6 +6,12 @@ $(object.path)/sljitAllocator.o: $(thirdparty.path)/sljitAllocator.cpp
 
 flags += -DSLJIT_HAVE_CONFIG_PRE=1 -DSLJIT_HAVE_CONFIG_POST=1
 
+ifeq ($(system_chdr),true)
+  flags += $(shell pkg-config --cflags libchdr)
+  options += $(shell pkg-config --libs libchdr)
+  libchdr.objects :=
+  libchdr.options :=
+else
 libchdr.objects := libchdr_bitstream libchdr_cdrom libchdr_chd libchdr_flac libchdr_huffman
 libchdr.objects += lzma_alloc lzma_bra86 lzma_braia64 lzma_cpuarch lzma_delta lzma_lfind
 libchdr.objects += lzma_86dec lzma_dec lcma_enc lzma_sort
@@ -45,7 +51,7 @@ $(object.path)/zlib_zutil.o: $(libchdr.path)/deps/zlib-1.2.13/zutil.c
 
 flags += -I$(libchdr.path)/include
 flags += -I$(libchdr.path)/deps/lzma-19.00/include -D_7ZIP_ST
-flags += -I$(libchdr.path)/deps/zlib-1.2.13
+endif
 
 # instruct glibc to declare fseeko/ftello
 flags += -D_LARGEFILE_SOURCE
-- 
2.41.0

