diff -rupN --no-dereference mingw-w64-v8.0.0/mingw-w64-tools/widl/include/windef.h mingw-w64-v8.0.0-new/mingw-w64-tools/widl/include/windef.h
--- mingw-w64-v8.0.0/mingw-w64-tools/widl/include/windef.h	2020-09-17 04:20:15.000000000 +0200
+++ mingw-w64-v8.0.0-new/mingw-w64-tools/widl/include/windef.h	2021-01-16 00:00:46.860772591 +0100
@@ -107,7 +107,7 @@ extern "C" {
 #endif
 
 #ifndef __ms_va_list
-# if (defined(__x86_64__) || defined(__aarch64__)) && defined (__GNUC__)
+# if (defined(__x86_64__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
