--- mingw-w64-v9.0.0/mingw-w64-tools/widl/include/windef.h	2021-05-22 09:34:25.000000000 -0300
+++ mingw-w64-v9.0.0-new/mingw-w64-tools/widl/include/windef.h	2021-05-24 18:25:33.575690447 -0300
@@ -103,7 +103,7 @@ extern "C" {
 #endif
 
 #ifndef __ms_va_list
-# if (defined(__x86_64__) || (defined(__aarch64__) && __has_attribute(ms_abi))) && defined (__GNUC__)
+# if (defined(__x86_64__) && __has_attribute(ms_abi))) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
