diff -up nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine32.in.optflags nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine32.in
--- nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine32.in.optflags	2019-01-07 06:56:56.000000000 -0200
+++ nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine32.in	2019-01-07 16:28:31.627441594 -0200
@@ -5,8 +5,8 @@ strip = 'strip'
 pkgconfig = '@PKG_CONFIG@'
 
 [properties]
-c_args = ['-m32', '--no-gnu-unique']
-c_link_args = ['-m32', '-mwindows']
+c_args = ['-m32', '--no-gnu-unique', 'RPM_OPT_FLAGS']
+c_link_args = ['-m32', '-mwindows', 'RPM_LD_FLAGS']
 needs_exe_wrapper = true
 
 [host_machine]
diff -up nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine64.in.optflags nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine64.in
--- nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine64.in.optflags	2019-01-07 06:56:56.000000000 -0200
+++ nine-136dca65e0afa96be752ef6b8385a693bfe18279/tools/cross-wine64.in	2019-01-07 16:28:34.603436023 -0200
@@ -5,8 +5,8 @@ strip = 'strip'
 pkgconfig = '@PKG_CONFIG@'
 
 [properties]
-c_args = ['-m64', '--no-gnu-unique']
-c_link_args = ['-m64', '-mwindows']
+c_args = ['-m64', '--no-gnu-unique', 'RPM_OPT_FLAGS']
+c_link_args = ['-m64', '-mwindows', 'RPM_LD_FLAGS']
 needs_exe_wrapper = true
 
 [host_machine]
