From 9866efcabbf4219066ed47862059fd48c24b7ec0 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 20 Jul 2019 08:54:05 -0300
Subject: [PATCH] Fix format security error

---
 libswirl/rend/gui.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libswirl/rend/gui.cpp b/libswirl/rend/gui.cpp
index a3d9974..72be7f1 100644
--- a/libswirl/rend/gui.cpp
+++ b/libswirl/rend/gui.cpp
@@ -955,7 +955,7 @@ static void gui_display_settings()
 					if (ImGui::Selectable(backend.slug.c_str(), &is_selected))
 						settings.pvr.backend = backend.slug;
 					ImGui::SameLine(); ImGui::Text("-");
-					ImGui::SameLine(); ImGui::Text(backend.desc.c_str());
+					ImGui::SameLine(); ImGui::Text(backend.desc.c_str(), "%s");
 				}
 
 				ImGui::EndCombo();
@@ -1074,7 +1074,7 @@ static void gui_display_settings()
 					if (ImGui::Selectable(backend->slug.c_str(), &is_selected))
 						settings.audio.backend = backend->slug;
 					ImGui::SameLine(); ImGui::Text("-");
-					ImGui::SameLine(); ImGui::Text(backend->name.c_str());
+					ImGui::SameLine(); ImGui::Text(backend->name.c_str(), "%s");
 	                if (is_selected)
 	                    ImGui::SetItemDefaultFocus();
 				}
-- 
2.21.0

