From 885270d6673319965bd73a2d50dc0abb05fe10bc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 20 Jul 2019 00:42:54 -0300
Subject: [PATCH] Build fixes

---
 reicast/linux/Makefile | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/reicast/linux/Makefile b/reicast/linux/Makefile
index a2a1fe0..be0d424 100644
--- a/reicast/linux/Makefile
+++ b/reicast/linux/Makefile
@@ -4,8 +4,8 @@ FOR_LINUX :=1
 #NO_REC := 1
 #NO_REND := 1
 WEBUI :=1
-USE_OSS := 1
-#USE_PULSEAUDIO := 1
+#USE_OSS := 1
+#Crashy# USE_PULSEAUDIO := 1
 #USE_SDLAUDIO := 1
 #USE_LIBAO := 1
 USE_EVDEV := 1
@@ -24,12 +24,12 @@ CHD5_LZMA := 1
 CHD5_FLAC := 1
 
 MFLAGS :=
-ASFLAGS :=
-LDFLAGS :=
+#ASFLAGS :=
+#LDFLAGS :=
 INCS :=
 LIBS :=
-CFLAGS :=
-CXXFLAGS :=
+#CFLAGS :=
+#CXXFLAGS :=
 
 ifneq (, $(filter $(shell uname -s), FreeBSD OpenBSD NetBSD DragonFly))
     CFLAGS += -DTARGET_BSD
@@ -319,13 +319,13 @@ endif
 RZDCY_SRC_DIR = $(LOCAL_PATH)/../../libswirl
 include $(RZDCY_SRC_DIR)/core.mk
 
-LDFLAGS += -g -Wl,-Map,$(notdir $@).map,--gc-sections -Wl,-O3 -Wl,--sort-common -fopenmp
+LDFLAGS += -g -Wl,-Map,$(notdir $@).map,--gc-sections -Wl,--sort-common -fopenmp
 
-CFLAGS += $(RZDCY_CFLAGS) -g -O3 -D RELEASE -c -D USES_HOMEDIR -fopenmp #-D NO_REND
+CFLAGS += $(RZDCY_CFLAGS) -D RELEASE -c -D USES_HOMEDIR -fopenmp #-D NO_REND
 CFLAGS += -frename-registers -fno-strict-aliasing
 CFLAGS += -ffast-math -ftree-vectorize
 
-CXXFLAGS += $(RZDCY_CFLAGS) -fno-rtti -fpermissive -fno-operator-names -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++14
+CXXFLAGS += $(RZDCY_CFLAGS) -fno-rtti -fpermissive -fno-operator-names -std=gnu++14
 
 INCS += -I$(RZDCY_SRC_DIR) -I$(RZDCY_SRC_DIR)/deps -I$(RZDCY_SRC_DIR)/khronos
 
@@ -525,7 +525,7 @@ $(BUILDDIR)/%.build_obj : $(RZDCY_SRC_DIR)/%.S $(VERSION_HEADER)
 	mkdir -p $(dir $@)
 	$(AS) $(ASFLAGS) $(INCS) $< -o $@
 
-install: $(EXECUTABLE)
+install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(PREFIX)/share/reicast/mappings 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(MAN_DIR) 2>/dev/null || true
-- 
2.21.0

