From 6c0557b852a04670da88bc19241912bdf6ead5fe Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 5 Feb 2020 09:34:15 -0300
Subject: [PATCH] Disable setuptools_scm version check

---
 setup.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/setup.py b/setup.py
index 0ce0ba2..f1eaec9 100644
--- a/setup.py
+++ b/setup.py
@@ -16,9 +16,6 @@ URL = "https://github.com/Matoking/protontricks"
 
 setup(
     name="protontricks",
-    use_scm_version={
-        "write_to": "src/protontricks/_version.py"
-    },
     description=DESCRIPTION,
     long_description=LONG_DESCRIPTION,
     author=AUTHOR,
@@ -28,7 +25,6 @@ setup(
     packages=["protontricks"],
     package_data={"": ["LICENSE"]},
     package_dir={"protontricks": "src/protontricks"},
-    setup_requires=["setuptools_scm"],
     install_requires=["vdf>=3.2"],
     entry_points={
         "console_scripts": [
-- 
2.24.1

