diff -urN taglib-1.12.old/taglib-config.cmake taglib-1.12/taglib-config.cmake
--- taglib-1.12.old/taglib-config.cmake	2021-02-15 11:23:07.000000000 +0000
+++ taglib-1.12/taglib-config.cmake	2021-02-16 09:18:07.018627259 +0000
@@ -16,7 +16,6 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@LIB_INSTALL_DIR@
 includedir=@INCLUDE_INSTALL_DIR@
 
 flags=""
@@ -29,7 +28,7 @@
 do
   case $1 in
     --libs)
-	  flags="$flags -L$libdir -ltag @ZLIB_LIBRARIES_FLAGS@"
+	  flags="$flags -ltag @ZLIB_LIBRARIES_FLAGS@"
 	  ;;
     --cflags)
 	  flags="$flags -I$includedir -I$includedir/taglib"
