Index: gtk+-3.18.1/gtk/updateiconcache.c
===================================================================
--- gtk+-3.18.1.orig/gtk/updateiconcache.c	2015-10-08 00:19:36.650119362 +0200
+++ gtk+-3.18.1/gtk/updateiconcache.c	2015-10-08 00:19:36.646119308 +0200
@@ -678,7 +678,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 
 	      image = g_new0 (Image, 1);
