diff -up gtk+-3.22.7/gtk/gtkscrolledwindow.c.nooverlay gtk+-3.22.7/gtk/gtkscrolledwindow.c
--- gtk+-3.22.7/gtk/gtkscrolledwindow.c.nooverlay	2017-01-04 16:41:20.000000000 -0200
+++ gtk+-3.22.7/gtk/gtkscrolledwindow.c	2017-02-08 16:54:53.784752891 -0200
@@ -712,7 +712,7 @@ gtk_scrolled_window_class_init (GtkScrol
       g_param_spec_boolean ("overlay-scrolling",
                             P_("Overlay Scrolling"),
                             P_("Overlay scrolling mode"),
-                            TRUE,
+                            FALSE,
                             GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY);
 
   /**
@@ -2134,7 +2134,7 @@ gtk_scrolled_window_init (GtkScrolledWin
   priv->max_content_width = -1;
   priv->max_content_height = -1;
 
-  priv->overlay_scrolling = TRUE;
+  priv->overlay_scrolling = FALSE;
 
   priv->drag_gesture = gtk_gesture_drag_new (widget);
   gtk_gesture_single_set_touch_only (GTK_GESTURE_SINGLE (priv->drag_gesture), TRUE);
@@ -4412,8 +4412,8 @@ gtk_scrolled_window_update_use_indicator
 
   use_indicators = priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
-    use_indicators = FALSE;
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "1") == 0)
+    use_indicators = TRUE;
 
   if (priv->use_indicators != use_indicators)
     {
