From 8a0645d4f5c2077edd20450fda711433959c4bbe Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 28 Sep 2021 23:43:21 -0300
Subject: [PATCH] Use system SDL_GameControllerDB

---
 core/sdl/sdl.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/sdl/sdl.cpp b/core/sdl/sdl.cpp
index 62467d0..8f0de25 100644
--- a/core/sdl/sdl.cpp
+++ b/core/sdl/sdl.cpp
@@ -158,11 +158,11 @@ void input_sdl_init()
 			SDL_SetHint(SDL_HINT_XINPUT_ENABLED, "0");
 		}
 #endif
-		std::string db = get_readonly_data_path("gamecontrollerdb.txt");
+		std::string db = "_RPM_GCDBDIR_/gamecontrollerdb.txt";
 		int rv = SDL_GameControllerAddMappingsFromFile(db.c_str());
 		if (rv < 0)
 		{
-			db = get_readonly_config_path("gamecontrollerdb.txt");
+			db = "_RPM_GCDBDIR_/gamecontrollerdb.txt";
 			rv = SDL_GameControllerAddMappingsFromFile(db.c_str());
 		}
 		if (rv > 0)
-- 
2.31.1

