From 946b39e2eac67e2c83e9db3c467809b448cd333c Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 8 Apr 2020 11:44:54 -0300
Subject: [PATCH] Save logfile to writable_data_path

---
 core/log/LogManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/log/LogManager.cpp b/core/log/LogManager.cpp
index 50bbde5..a14d89f 100644
--- a/core/log/LogManager.cpp
+++ b/core/log/LogManager.cpp
@@ -132,7 +132,7 @@ LogManager::LogManager()
 #ifdef __ANDROID__
 		std::string logPath = get_writable_data_path("/flycast.log");
 #else
-		std::string logPath = "flycast.log";
+		std::string logPath = get_writable_data_path("/flycast.log");
 #endif
 		RegisterListener(LogListener::FILE_LISTENER, new FileLogListener(logPath));
 		EnableListener(LogListener::FILE_LISTENER, true);
-- 
2.26.0

