From bb0071c9487b2b3601c3e17e9fa4dd80044cd774 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 21 Oct 2024 20:46:23 -0300
Subject: [PATCH] vulkan_context.cpp: add missing header

---
 core/rend/vulkan/vulkan_context.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/rend/vulkan/vulkan_context.cpp b/core/rend/vulkan/vulkan_context.cpp
index 304bd0c..0fed035 100644
--- a/core/rend/vulkan/vulkan_context.cpp
+++ b/core/rend/vulkan/vulkan_context.cpp
@@ -42,6 +42,7 @@ VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE
 #endif
 
 #include <memory>
+#include <set>
 
 void ReInitOSD();
 
-- 
2.47.0

