From 2e6e2067151a073d61fffa634fe2a0d58869606b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 21 Mar 2020 09:33:23 -0300
Subject: [PATCH] Build fixes

---
 shell/linux/Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/shell/linux/Makefile b/shell/linux/Makefile
index 262838f..ffa2462 100644
--- a/shell/linux/Makefile
+++ b/shell/linux/Makefile
@@ -3,8 +3,8 @@ LOCAL_PATH := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 FOR_LINUX :=1
 #NO_REC := 1
 #NO_REND := 1
-USE_OSS := 1
-#USE_PULSEAUDIO := 1
+#USE_OSS := 1
+USE_PULSEAUDIO := 1
 #USE_SDLAUDIO := 1
 #USE_LIBAO := 1
 ifndef USE_SDL
@@ -28,8 +28,8 @@ ASFLAGS :=
 LDFLAGS ?=
 INCS ?=
 LIBS ?=
-CFLAGS :=
-CXXFLAGS :=
+CFLAGS ?=
+CXXFLAGS ?=
 
 ifneq (, $(filter $(shell uname -s), FreeBSD OpenBSD NetBSD DragonFly))
     CFLAGS += -DTARGET_BSD
@@ -291,13 +291,13 @@ endif
 RZDCY_SRC_DIR = $(LOCAL_PATH)/../../core
 include $(RZDCY_SRC_DIR)/core.mk
 
-LDFLAGS += -g -Wl,-Map,$(notdir $@).map,--gc-sections -Wl,-O3 -Wl,--sort-common -fopenmp
+LDFLAGS += -g -Wl,-Map,$(notdir $@).map,--gc-sections -Wl,--sort-common -fopenmp
 
-CFLAGS += $(RZDCY_CFLAGS) -g -O3 -D RELEASE -c -D USES_HOMEDIR -fopenmp #-D NO_REND
+CFLAGS += $(RZDCY_CFLAGS) -D RELEASE -c -D USES_HOMEDIR -fopenmp #-D NO_REND
 CFLAGS += -frename-registers -fno-strict-aliasing
 CFLAGS += -ffast-math -ftree-vectorize  -Wno-unused-result
 
-CXXFLAGS += $(RZDCY_CFLAGS) -fno-rtti -fno-operator-names -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++11
+CXXFLAGS += $(RZDCY_CFLAGS) -fno-rtti -fno-operator-names -std=gnu++11
 
 INCS += -I$(RZDCY_SRC_DIR) -I$(RZDCY_SRC_DIR)/deps -I$(RZDCY_SRC_DIR)/khronos
 
@@ -491,7 +491,7 @@ $(BUILDDIR)/%.build_obj : $(RZDCY_SRC_DIR)/%.S $(VERSION_HEADER)
 	mkdir -p $(dir $@)
 	$(AS) $(ASFLAGS) $(INCS) $< -o $@
 
-install: $(EXECUTABLE)
+install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(PREFIX)/share/reicast/mappings 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(MAN_DIR) 2>/dev/null || true
-- 
2.26.0.rc2

