From dd5feab44223483fd6914f187c44fca24a5e5c13 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 2 Sep 2021 07:37:02 -0300
Subject: [PATCH] Use system libraries

---
 Makefile       | 36 +++++++++++++++++++++++++++---------
 src/sector.cpp |  2 +-
 2 files changed, 28 insertions(+), 10 deletions(-)

diff --git a/Makefile b/Makefile
index 15cb2ca..971ace0 100644
--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,9 @@ PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 MANDIR ?= $(PREFIX)/share/man
 
+SYSTEM_LIBDEFLATE ?= 0
+SYSTEM_ZOPFLI ?= 0
+
 CC ?= gcc
 CXX ?= g++
 
@@ -11,14 +14,19 @@ CXXFLAGS ?= $(CFLAGS)
 LIBS ?= libuv liblz4 zlib
 
 SRC_CFLAGS += -W -Wall -Wextra -Wno-implicit-function-declaration -DNDEBUG=1
-SRC_CXXFLAGS += -W -Wall -Wextra -std=c++11 -Izopfli/src -I7zip -DNDEBUG=1 \
-	-Ilibdeflate -Wno-unused-parameter -Wno-unused-variable -pthread \
+SRC_CXXFLAGS += -W -Wall -Wextra -std=c++11 -I7zip -DNDEBUG=1 \
+	-Wno-unused-parameter -Wno-unused-variable -pthread \
 	`pkg-config --cflags $(LIBS)`
 
 SRC_CXX_SRC = $(wildcard src/*.cpp)
 SRC_CXX_OBJ = $(SRC_CXX_SRC:.cpp=.o)
 CLI_CXX_SRC = $(wildcard cli/*.cpp)
 CLI_CXX_OBJ = $(CLI_CXX_SRC:.cpp=.o)
+
+EXTRA_LIBS =
+
+ifeq ($(SYSTEM_ZOPFLI),0)
+SRC_CXXFLAGS += -Izopfli/src/zopfli
 ZOPFLI_C_SRC = zopfli/src/zopfli/blocksplitter.c zopfli/src/zopfli/cache.c \
                zopfli/src/zopfli/deflate.c zopfli/src/zopfli/gzip_container.c \
                zopfli/src/zopfli/hash.c zopfli/src/zopfli/katajainen.c \
@@ -26,13 +34,21 @@ ZOPFLI_C_SRC = zopfli/src/zopfli/blocksplitter.c zopfli/src/zopfli/cache.c \
                zopfli/src/zopfli/tree.c zopfli/src/zopfli/util.c \
                zopfli/src/zopfli/zlib_container.c zopfli/src/zopfli/zopfli_lib.c
 ZOPFLI_C_OBJ = $(ZOPFLI_C_SRC:.c=.o)
+else
+	EXTRA_LIBS += -lzopfli
+endif
 
-EXTRA_LIBS =
-ifeq ($(OS),Windows_NT)
-	LIBDEFLATE=libdeflatestatic.lib
-	EXTRA_LIBS += -luuid
+LIBDEFLATE =
+ifeq ($(SYSTEM_LIBDEFLATE),0)
+	SRC_CXXFLAGS += -Ilibdeflate
+	ifeq ($(OS),Windows_NT)
+		LIBDEFLATE=libdeflate/libdeflatestatic.lib
+		EXTRA_LIBS += -luuid
+	else
+		LIBDEFLATE=libdeflate/libdeflate.a
+	endif
 else
-	LIBDEFLATE=libdeflate.a
+	EXTRA_LIBS += -ldeflate
 endif
 
 %.o: %.cpp
@@ -42,14 +58,16 @@ endif
 	$(CC) -c $(SRC_CFLAGS) $(CFLAGS) -o $@ $<
 
 # TODO: Perhaps detect and use system libdeflate if available.
-maxcso: $(SRC_CXX_OBJ) $(CLI_CXX_OBJ) $(ZOPFLI_C_OBJ) 7zip/7zip.a libdeflate/$(LIBDEFLATE)
+maxcso: $(SRC_CXX_OBJ) $(CLI_CXX_OBJ) $(ZOPFLI_C_OBJ) 7zip/7zip.a $(LIBDEFLATE)
 	$(CXX) -o $@ $(SRC_CXXFLAGS) $(CXXFLAGS) $^ `pkg-config --libs $(LIBS)` $(EXTRA_LIBS)
 
 7zip/7zip.a:
 	$(MAKE) -C 7zip 7zip.a
 
-libdeflate/$(LIBDEFLATE):
+ifeq ($(SYSTEM_LIBDEFLATE),0)
+$(LIBDEFLATE):
 	$(MAKE) -C libdeflate $(LIBDEFLATE)
+endif
 
 install: all
 	mkdir -p $(DESTDIR)$(BINDIR)
diff --git a/src/sector.cpp b/src/sector.cpp
index ea4730f..cebc3a6 100644
--- a/src/sector.cpp
+++ b/src/sector.cpp
@@ -3,7 +3,7 @@
 #include "compress.h"
 #include "cso.h"
 #include "buffer_pool.h"
-#include "zopfli/zopfli.h"
+#include "zopfli.h"
 #include "libdeflate.h"
 #ifndef NO_DEFLATE7Z
 #include "deflate7z.h"
-- 
2.31.1

