From c4c5d068a9051fc556b6dee2022226131929f1aa Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 11 Dec 2021 20:54:01 -0300
Subject: [PATCH] Use system libraries

---
 Makefile       | 12 ++++++++++--
 src/sector.cpp |  2 +-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 85963a3..8bb0c22 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 SRCDIR := $(abspath $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST))))))
 
 USE_EXTERNAL_LIBDEFLATE ?= 0
+USE_EXTERNAL_ZOPFLI ?= 0
 
 PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
@@ -30,8 +31,8 @@ OBJDIR := obj
 MKDIRS := $(OBJDIR)/src $(OBJDIR)/cli $(OBJDIR)/zopfli/src/zopfli
 
 SRC_CFLAGS += -W -Wall -Wextra -Wno-implicit-function-declaration -DNDEBUG=1
-SRC_CXXFLAGS += -W -Wall -Wextra -std=c++11 -I$(SRCDIR)/zopfli/src -I$(SRCDIR)/7zip \
-	-DNDEBUG=1 -I$(SRCDIR)/libdeflate -Wno-unused-parameter -Wno-unused-variable \
+SRC_CXXFLAGS += -W -Wall -Wextra -std=c++11 -I$(SRCDIR)/7zip \
+	-DNDEBUG=1 -Wno-unused-parameter -Wno-unused-variable \
 	-pthread $(DEP_FLAGS)
 
 SRC_CXX_SRC := $(wildcard $(SRCDIR)/src/*.cpp)
@@ -42,6 +43,8 @@ CLI_CXX_SRC := $(wildcard $(SRCDIR)/cli/*.cpp)
 CLI_CXX_TMP := $(CLI_CXX_SRC:.cpp=.o)
 CLI_CXX_OBJ := $(patsubst $(SRCDIR)/%,$(OBJDIR)/%,$(CLI_CXX_TMP))
 
+ifeq ($(USE_EXTERNAL_ZOPFLI),0)
+SRC_CXXFLAGS += -I$(SRCDIR)/zopfli/src/zopfli
 ZOPFLI_C_DIR := $(SRCDIR)/zopfli/src/zopfli
 ZOPFLI_C_SRC := $(ZOPFLI_C_DIR)/blocksplitter.c $(ZOPFLI_C_DIR)/cache.c \
                $(ZOPFLI_C_DIR)/deflate.c $(ZOPFLI_C_DIR)/gzip_container.c \
@@ -51,6 +54,9 @@ ZOPFLI_C_SRC := $(ZOPFLI_C_DIR)/blocksplitter.c $(ZOPFLI_C_DIR)/cache.c \
                $(ZOPFLI_C_DIR)/zlib_container.c $(ZOPFLI_C_DIR)/zopfli_lib.c
 ZOPFLI_C_TMP := $(ZOPFLI_C_SRC:.c=.o)
 ZOPFLI_C_OBJ := $(patsubst $(SRCDIR)/%,$(OBJDIR)/%,$(ZOPFLI_C_TMP))
+else
+	LIBS += -lzopfli
+endif
 
 ifeq ($(OS),Windows_NT)
 	LIBDEFLATE := libdeflatestatic.lib
@@ -89,8 +95,10 @@ maxcso: $(OBJS)
 $(SRC_7ZIP):
 	$(MAKE) -f $(SRCDIR)/7zip/Makefile 7zip.a
 
+ifeq ($(USE_EXTERNAL_LIBDEFLATE),0)
 $(SRC_LIBDEFLATE):
 	$(MAKE) -C $(SRCDIR)/libdeflate $(LIBDEFLATE)
+endif
 
 $(OBJDIR)/.done:
 	@mkdir -p $(MKDIRS)
diff --git a/src/sector.cpp b/src/sector.cpp
index 81a3059..afefe19 100644
--- a/src/sector.cpp
+++ b/src/sector.cpp
@@ -3,7 +3,7 @@
 #include "compress.h"
 #include "cso.h"
 #include "buffer_pool.h"
-#include "zopfli/zopfli.h"
+#include "zopfli.h"
 #include "libdeflate.h"
 #ifndef NO_DEFLATE7Z
 #include "deflate7z.h"
-- 
2.44.0

