From 91e514c4c6e8cc3f770e9058b1864d7cf196d6af Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 20 Aug 2021 23:58:08 -0300
Subject: [PATCH] Update patch-bug1456512

---
 _patches/patch-bug1456512 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/_patches/patch-bug1456512 b/_patches/patch-bug1456512
index 4f52fe9..78a922b 100644
--- a/_patches/patch-bug1456512
+++ b/_patches/patch-bug1456512
@@ -62,9 +62,9 @@ index c14f8bfc2005e..340287e8322e9 100644
 +++ js/src/gc/Heap.h
 @@ -1381,7 +1381,7 @@ TenuredCell::readBarrier(TenuredCell* thing)
      if (thing->isMarkedGray()) {
-         // There shouldn't be anything marked grey unless we're on the active thread.
+         // There shouldn't be anything marked grey unless we're on the main thread.
          MOZ_ASSERT(CurrentThreadCanAccessRuntime(thing->runtimeFromAnyThread()));
--        if (!RuntimeFromActiveCooperatingThreadIsHeapMajorCollecting(shadowZone))
+-        if (!RuntimeFromMainThreadIsHeapMajorCollecting(shadowZone))
 +        if (!JS::CurrentThreadIsHeapCollecting())
              JS::UnmarkGrayGCThingRecursively(JS::GCCellPtr(thing, thing->getTraceKind()));
      }
-- 
2.31.1

