diff -up Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp.waterfoxdir Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp
--- Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp.waterfoxdir	2017-12-27 16:08:15.089424561 -0200
+++ Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp	2017-12-27 16:08:15.119424503 -0200
@@ -284,7 +284,7 @@ nsXREDirProvider::GetUserProfilesLocalDi
  * for extensions and native-messaing manifests.
  *
  * On OSX this is /Library/Application Support/Mozilla
- * On Linux this is /usr/{lib,lib64}/mozilla
+ * On Linux this is /usr/{lib,lib64}/waterfox
  *   (for 32- and 64-bit systems respsectively)
  */
 static nsresult
