From 36b535c9bfffcac87db2adfd3d6c4fd13e7d12b1 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 13 Dec 2021 15:23:51 -0300
Subject: [PATCH] cmake: library fixes

---
 src/CMakeLists.txt | 21 ++++++++-------------
 1 file changed, 8 insertions(+), 13 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ba62e48..db2f4f5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -85,9 +85,10 @@ set_target_properties(qtadvanceddocking PROPERTIES
     AUTORCC ON
     CXX_EXTENSIONS OFF
     VERSION ${VERSION_SHORT}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
     EXPORT_NAME "qtadvanceddocking"
     ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib"
-    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib${LIB_SUFFIX}"
     RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/bin"
 )
 if(QT_VERSION_MAJOR STREQUAL "5")
@@ -107,30 +108,24 @@ write_basic_package_version_file(
     COMPATIBILITY SameMajorVersion
 )
 install(FILES ${ads_HEADERS}
-    DESTINATION include
+    DESTINATION include/qtadvanceddocking
     COMPONENT headers
 )
-install(FILES
-    "${CMAKE_CURRENT_SOURCE_DIR}/../LICENSE"
-    "${CMAKE_CURRENT_SOURCE_DIR}/../gnu-lgpl-v2.1.md"
-    DESTINATION license/ads
-    COMPONENT license
-)
 install(TARGETS qtadvanceddocking
     EXPORT adsTargets
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
-    INCLUDES DESTINATION include
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
+    INCLUDES DESTINATION include/qtadvanceddocking
 )
 
 install(EXPORT adsTargets
     FILE adsTargets.cmake
     NAMESPACE ads::
-    DESTINATION lib/cmake/qtadvanceddocking
+    DESTINATION lib${LIB_SUFFIX}/cmake/qtadvanceddocking
 )
 install(FILES qtadvanceddockingConfig.cmake "${CMAKE_CURRENT_BINARY_DIR}/qtadvanceddockingConfigVersion.cmake"
-    DESTINATION lib/cmake/qtadvanceddocking
+    DESTINATION lib${LIB_SUFFIX}/cmake/qtadvanceddocking
 )
 
 target_include_directories(qtadvanceddocking PUBLIC
-- 
2.39.2

