From c6c17fdc38606c2530424ffef7a694c28328c084 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 13 Dec 2021 15:23:51 -0300
Subject: [PATCH] cmake: library fixes

---
 src/CMakeLists.txt      |  21 +++---
 3 files changed, 208 insertions(+), 13 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8224644..24c6a9d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -92,9 +92,10 @@ set_target_properties(${library_name} PROPERTIES
     AUTORCC ON
     CXX_EXTENSIONS OFF
     VERSION ${VERSION_SHORT}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
     EXPORT_NAME ${library_name}
-    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib"
-    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib"
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib${LIB_SUFFIX}"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/lib${LIB_SUFFIX}"
     RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${ads_PlatformDir}/bin"
 )
 if(QT_VERSION_MAJOR STREQUAL "5")
@@ -117,30 +118,24 @@ install(FILES ${ads_HEADERS}
     DESTINATION include/${library_name}
     COMPONENT headers
 )
-install(FILES
-    "${CMAKE_CURRENT_SOURCE_DIR}/../LICENSE"
-    "${CMAKE_CURRENT_SOURCE_DIR}/../gnu-lgpl-v2.1.md"
-    DESTINATION license/ads
-    COMPONENT license
-)
 install(TARGETS ${library_name}
     EXPORT adsTargets
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     INCLUDES DESTINATION include/${library_name}
 )
 
 install(EXPORT adsTargets
     FILE adsTargets.cmake
     NAMESPACE ads::
-    DESTINATION lib/cmake/${library_name}
+    DESTINATION lib${LIB_SUFFIX}/cmake/${library_name}
 )
 install(FILES qtadvanceddockingConfig.cmake RENAME ${library_name}Config.cmake
-    DESTINATION lib/cmake/${library_name}
+    DESTINATION lib${LIB_SUFFIX}/cmake/${library_name}
 )
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${library_name}ConfigVersion.cmake"
-    DESTINATION lib/cmake/${library_name}
+    DESTINATION lib${LIB_SUFFIX}/cmake/${library_name}
 )
 
 target_include_directories(${library_name} PUBLIC
-- 
2.41.0

