From 556c1bd267ac1997ecf08f164d47c100e0015962 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 19 Mar 2025 18:18:05 -0300
Subject: [PATCH] gcc 15 build fix

---
 src/tools.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tools.h b/src/tools.h
index 26af0e6..3f309eb 100644
--- a/src/tools.h
+++ b/src/tools.h
@@ -27,6 +27,7 @@ using namespace std;
 #endif
 #include <list>
 #include <string>
+#include <cstdint>
 #include <vector>
 #include <memory>
 #include <cstdio>
-- 
2.49.0

