From 3a329427da625254edf4b37970abf9acd2c03aec Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 18 Mar 2020 15:54:07 -0300
Subject: [PATCH] fstats: replace link flags to fix linking

---
 src/fstats/fstats.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fstats/fstats.pro b/src/fstats/fstats.pro
index a113523..214bdf0 100644
--- a/src/fstats/fstats.pro
+++ b/src/fstats/fstats.pro
@@ -1,6 +1,6 @@
 QMAKE_CXXFLAGS += $$(CXXFLAGS) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
 QMAKE_CFLAGS += $$(CFLAGS) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
-QMAKE_LFLAGS += $$(LDFLAGS) -Wl,-z,relro -Wl,-z,now
+QMAKE_LFLAGS += -Wl,-z,relro -Wl,-z,now $$(LDFLAGS)
 QMAKE_LIBS += $$(LIBS) -lrt
 QT += widgets
  HEADERS       = ../common/utils.h ../common/pid.h ../common/common.h \
-- 
2.25.1

