--- casperjs-1.1.4-2/bin/casperjs.orig	2017-12-29 15:51:44.081726000 -0200
+++ casperjs-1.1.4-2/bin/casperjs	2017-12-29 15:59:48.831692533 -0200
@@ -9,12 +9,6 @@ import logging
 logging.basicConfig(stream=sys.stdout, level=logging.INFO, format='%(levelname)s - %(message)s')
 logger = logging.getLogger(__name__)
 
-def resolve(path):
-    if os.path.islink(path):
-        path = os.path.join(os.path.dirname(path), os.readlink(path))
-        return resolve(path)
-    return path
-
 SUPPORTED_ENGINES = {
     'phantomjs' : {
         'native_args': [
@@ -114,8 +108,7 @@ ENGINE_NATIVE_ARGS = []
 ENGINE_EXECUTABLE = ''
 
 CASPER_ARGS = []
-CASPER_PATH = os.path.abspath(os.path.join(os.path.dirname(resolve(__file__)),
-                                           '..'))
+CASPER_PATH = '_RPM_DATADIR_/casperjs'
 SYS_ARGS = sys.argv[1:]
 
 # retrieve the engine name
@@ -163,7 +156,7 @@ for arg in arg_iter:
 CASPER_COMMAND = [ENGINE_EXECUTABLE]
 CASPER_COMMAND.extend(ENGINE_ARGS)
 CASPER_COMMAND.extend([
-    os.path.join(CASPER_PATH, 'bin', 'bootstrap.js'),
+    os.path.join(CASPER_PATH, 'bootstrap.js'),
     '--casper-path=%s' % CASPER_PATH,
     '--cli'
 ])
