From 81d8dbb800b7aa396b6b0a57add39c7e86d7f306 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 19 Sep 2021 16:55:00 -0300
Subject: [PATCH] TestAudioEventTimeline.h gtest: fix build

---
 dom/media/webaudio/gtest/TestAudioEventTimeline.cpp | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/dom/media/webaudio/gtest/TestAudioEventTimeline.cpp b/dom/media/webaudio/gtest/TestAudioEventTimeline.cpp
index da7c4c4..b79afa5 100644
--- a/dom/media/webaudio/gtest/TestAudioEventTimeline.cpp
+++ b/dom/media/webaudio/gtest/TestAudioEventTimeline.cpp
@@ -9,17 +9,6 @@
 #include <limits>
 #include "gtest/gtest.h"
 
-// Mock the MediaStream class
-namespace mozilla {
-class MediaStream
-{
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(MediaStream)
-private:
-  ~MediaStream() {
-  };
-};
-}
-
 using namespace mozilla;
 using namespace mozilla::dom;
 using std::numeric_limits;
-- 
2.31.1

