diff -up Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/caps/NullPrincipal.cpp.waterfoxdir Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/caps/NullPrincipal.cpp
--- Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/caps/NullPrincipal.cpp.waterfoxdir	2017-12-19 21:28:53.000000000 -0200
+++ Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/caps/NullPrincipal.cpp	2017-12-27 16:08:15.118424505 -0200
@@ -110,7 +110,7 @@ NullPrincipal::Init(const OriginAttribut
   if (aIsFirstParty) {
     // remove the '{}' characters from both ends.
     path.Mid(path, 1, path.Length() - 2);
-    path.AppendLiteral(".mozilla");
+    path.AppendLiteral(".waterfox");
     attrs.SetFirstPartyDomain(true, path);
   }
 
diff -up Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/js/src/old-configure.in.waterfoxdir Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/js/src/old-configure.in
--- Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/js/src/old-configure.in.waterfoxdir	2017-12-19 21:28:53.000000000 -0200
+++ Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/js/src/old-configure.in	2017-12-27 16:12:54.069891842 -0200
@@ -400,7 +400,7 @@ AR_EXTRACT='$(AR) x'
 AS='$(CC)'
 AS_DASH_C_FLAG='-c'
 DIRENT_INO=d_ino
-MOZ_USER_DIR=".mozilla"
+MOZ_USER_DIR=".waterfox"
 
 MOZ_FIX_LINK_PATHS="-Wl,-rpath-link,${DIST}/bin -Wl,-rpath-link,${prefix}/lib"
 
diff -up Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/old-configure.in.waterfoxdir Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/old-configure.in
--- Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/old-configure.in.waterfoxdir	2017-12-19 21:28:53.000000000 -0200
+++ Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/old-configure.in	2017-12-27 16:12:53.110893673 -0200
@@ -451,7 +451,7 @@ AR_EXTRACT='$(AR) x'
 AS='$(CC)'
 AS_DASH_C_FLAG='-c'
 DIRENT_INO=d_ino
-MOZ_USER_DIR=".mozilla"
+MOZ_USER_DIR=".waterfox"
 
 MOZ_FIX_LINK_PATHS="-Wl,-rpath-link,${DIST}/bin -Wl,-rpath-link,${prefix}/lib"
 
diff -up Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp.waterfoxdir Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp
--- Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp.waterfoxdir	2017-12-27 16:08:15.089424561 -0200
+++ Waterfox-388770ff37846abbe9f2eebbd6d71014523bb047/toolkit/xre/nsXREDirProvider.cpp	2017-12-27 16:08:15.119424503 -0200
@@ -284,7 +284,7 @@ nsXREDirProvider::GetUserProfilesLocalDi
  * for extensions and native-messaing manifests.
  *
  * On OSX this is /Library/Application Support/Mozilla
- * On Linux this is /usr/{lib,lib64}/mozilla
+ * On Linux this is /usr/{lib,lib64}/waterfox
  *   (for 32- and 64-bit systems respsectively)
  */
 static nsresult
@@ -300,11 +300,11 @@ GetSystemParentDirectory(nsIFile** aFile
 #else
   NS_NAMED_LITERAL_CSTRING(dirname,
 #ifdef HAVE_USR_LIB64_DIR
-                           "/usr/lib64/mozilla"
+                           "/usr/lib64/waterfox"
 #elif defined(__OpenBSD__) || defined(__FreeBSD__)
-                           "/usr/local/lib/mozilla"
+                           "/usr/local/lib/waterfox"
 #else
-                           "/usr/lib/mozilla"
+                           "/usr/lib/waterfox"
 #endif
                            );
   rv = NS_NewNativeLocalFile(dirname, false, getter_AddRefs(localDir));
@@ -421,7 +421,7 @@ nsXREDirProvider::GetFile(const char* aP
         rv = localDir->AppendNative(NS_LITERAL_CSTRING("NativeMessagingHosts"));
       }
 #else
-      rv = localDir->AppendNative(NS_LITERAL_CSTRING(".mozilla"));
+      rv = localDir->AppendNative(NS_LITERAL_CSTRING(".waterfox"));
       if (NS_SUCCEEDED(rv)) {
         rv = localDir->AppendNative(NS_LITERAL_CSTRING("native-messaging-hosts"));
       }
@@ -480,9 +480,9 @@ nsXREDirProvider::GetFile(const char* aP
   else if (!strcmp(aProperty, XRE_SYS_SHARE_EXTENSION_PARENT_DIR)) {
 #ifdef ENABLE_SYSTEM_EXTENSION_DIRS
 #if defined(__OpenBSD__) || defined(__FreeBSD__)
-    static const char *const sysLExtDir = "/usr/local/share/mozilla/extensions";
+    static const char *const sysLExtDir = "/usr/local/share/waterfox/extensions";
 #else
-    static const char *const sysLExtDir = "/usr/share/mozilla/extensions";
+    static const char *const sysLExtDir = "/usr/share/waterfox/extensions";
 #endif
     return NS_NewNativeLocalFile(nsDependentCString(sysLExtDir),
                                  false, aFile);
@@ -1673,7 +1673,7 @@ nsXREDirProvider::AppendSysUserExtension
 
 #elif defined(XP_UNIX)
 
-  static const char* const sXR = ".mozilla";
+  static const char* const sXR = ".waterfox";
   rv = aFile->AppendNative(nsDependentCString(sXR));
   NS_ENSURE_SUCCESS(rv, rv);
 
@@ -1752,7 +1752,7 @@ nsXREDirProvider::AppendProfilePath(nsIF
   // XXXsmaug ...and the rest of the profile creation!
   MOZ_ASSERT(!aAppName,
              "Profile creation for external applications is not implemented!");
-  rv = aFile->AppendNative(nsDependentCString("mozilla"));
+  rv = aFile->AppendNative(nsDependentCString("waterfox"));
   NS_ENSURE_SUCCESS(rv, rv);
 #elif defined(XP_UNIX)
   nsAutoCString folder;
