diff -up Waterfox-ff4597172229f8d71940c3885b74b903b7b1821a/media/webrtc/trunk/gtest/moz.build.l4vl2 Waterfox-ff4597172229f8d71940c3885b74b903b7b1821a/media/webrtc/trunk/gtest/moz.build
--- Waterfox-ff4597172229f8d71940c3885b74b903b7b1821a/media/webrtc/trunk/gtest/moz.build.l4vl2	2018-12-17 10:39:21.000000000 -0200
+++ Waterfox-ff4597172229f8d71940c3885b74b903b7b1821a/media/webrtc/trunk/gtest/moz.build	2019-01-08 17:34:47.260911090 -0200
@@ -48,6 +48,8 @@ if CONFIG['MOZ_SYSTEM_JPEG']:
 else:
     USE_LIBS += ['media_libjpeg']
 
+OS_LIBS += CONFIG['MOZ_LIBV4L2_LIBS']
+
 if CONFIG['MOZ_SYSTEM_LIBVPX']:
     OS_LIBS += CONFIG['MOZ_LIBVPX_LIBS']
 else:
