From e94437a0f3444da9f42022090ba797725998928a Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 6 Jul 2017 17:27:01 +0200
Subject: [PATCH] Fixed build of libtgvoip under Fedora as shared library.

---
 libtgvoip.gyp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/libtgvoip.gyp b/libtgvoip.gyp
index 05bdb6c..e3820b2 100644
--- a/libtgvoip.gyp
+++ b/libtgvoip.gyp
@@ -4,7 +4,8 @@
     'targets': [
       {
         'target_name': 'libtgvoip',
-        'type': 'static_library',
+        'type': 'shared_library',
+        'product_extension': 'so.<!(echo -n "$VOIPVER")',
         'dependencies': [],
         'defines': [
           'WEBRTC_APM_DEBUG_DUMP=0',
@@ -46,8 +47,10 @@
           '<(tgvoip_src_loc)/OpusEncoder.cpp',
           '<(tgvoip_src_loc)/OpusEncoder.h',
           '<(tgvoip_src_loc)/threading.h',
+          '<(tgvoip_src_loc)/TgVoip.cpp',
           '<(tgvoip_src_loc)/VoIPController.cpp',
           '<(tgvoip_src_loc)/VoIPGroupController.cpp',
+          '<(tgvoip_src_loc)/TgVoip.h',
           '<(tgvoip_src_loc)/VoIPController.h',
           '<(tgvoip_src_loc)/PrivateDefines.h',
           '<(tgvoip_src_loc)/VoIPServerConfig.cpp',
@@ -78,8 +81,6 @@
           '<(tgvoip_src_loc)/video/VideoPacketSender.h',
           '<(tgvoip_src_loc)/video/VideoFEC.cpp',
           '<(tgvoip_src_loc)/video/VideoFEC.h',
-          '<(tgvoip_src_loc)/json11.cpp',
-          '<(tgvoip_src_loc)/json11.hpp',
 
           # Windows
           '<(tgvoip_src_loc)/os/windows/NetworkSocketWinsock.cpp',
@@ -730,7 +731,9 @@
 
 
         ],
-        'libraries': [],
+        'libraries': [
+            'json11',
+        ],
         'configurations': {
           'Debug': {},
           'Release': {},
-- 
2.24.1

