--- libtgvoip-2.4/libtgvoip.gyp.orig	2018-12-11 16:52:12.654488241 -0200
+++ libtgvoip-2.4/libtgvoip.gyp	2018-12-11 16:52:34.025413950 -0200
@@ -721,7 +721,10 @@
 
 
         ],
-        'libraries': [],
+        'libraries': [
+            '<!(pkg-config --libs libcrypto)',
+            '<!(pkg-config --libs opus)',
+        ],
         'configurations': {
           'Debug': {},
           'Release': {},
