From e94437a0f3444da9f42022090ba797725998928a Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 6 Jul 2017 17:27:01 +0200
Subject: [PATCH] Fixed build of libtgvoip under Fedora as shared library.

---
 libtgvoip.gyp | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/libtgvoip.gyp b/libtgvoip.gyp
index 8c62b56..8fe9c5b 100644
--- a/libtgvoip.gyp
+++ b/libtgvoip.gyp
@@ -4,7 +4,8 @@
     'targets': [
       {
         'target_name': 'libtgvoip',
-        'type': 'static_library',
+        'type': 'shared_library',
+        'product_extension': 'so.<!(echo -n "$VOIPVER")',
         'dependencies': [],
         'defines': [
           'WEBRTC_APM_DEBUG_DUMP=0',
@@ -14,11 +15,11 @@
         'variables': {
           'tgvoip_src_loc': '.',
           'official_build_target%': '',
-          'linux_path_opus_include%': '<(DEPTH)/../../../Libraries/opus/include',
         },
         'include_dirs': [
           '<(tgvoip_src_loc)/webrtc_dsp',
-          '<(linux_path_opus_include)',
+          '<!(rpm --eval "%{_includedir}")',
+          '<!(rpm --eval "%{_includedir}/opus")',
         ],
         'direct_dependent_settings': {
           'include_dirs': [
@@ -70,8 +71,6 @@
           '<(tgvoip_src_loc)/video/VideoSource.h',
           '<(tgvoip_src_loc)/video/VideoRenderer.cpp',
           '<(tgvoip_src_loc)/video/VideoRenderer.h',
-          '<(tgvoip_src_loc)/json11.cpp',
-          '<(tgvoip_src_loc)/json11.hpp',
 
           # Windows
           '<(tgvoip_src_loc)/os/windows/NetworkSocketWinsock.cpp',
@@ -722,7 +721,9 @@
 
 
         ],
-        'libraries': [],
+        'libraries': [
+            'json11',
+        ],
         'configurations': {
           'Debug': {},
           'Release': {},
-- 
2.20.1

