From 64abf100f979aa9682fe2c61fc290fe037aa9f87 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 11 Nov 2021 20:36:39 -0300
Subject: [PATCH] Use system lzma-sdk

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e14392..f0324f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,7 @@ if(CMAKE_PROJECT_NAME STREQUAL "chdr")
 endif()
 option(INSTALL_STATIC_LIBS "Install static libraries" OFF)
 option(WITH_SYSTEM_ZLIB "Use system provided zlib library" OFF)
+option(WITH_SYSTEM_LZMA "Use system provided lzma-sdk library" OFF)
 
 option(BUILD_LTO "Compile libchdr with link-time optimization if supported" OFF)
 if(BUILD_LTO)
@@ -25,9 +26,16 @@ include(GNUInstallDirs)
 
 
 # lzma
+if (WITH_SYSTEM_LZMA)
+  include(FindPkgConfig)
+  pkg_check_modules(LZMA REQUIRED lzmasdk-c)
+  list(APPEND CHDR_INCLUDES ${LZMA_INCLUDE_DIRS})
+  list(APPEND CHDR_LIBS ${LZMA_LIBRARIES})
+else()
 add_subdirectory(deps/lzma-22.01 EXCLUDE_FROM_ALL)
   list(APPEND CHDR_LIBS lzma)
   list(APPEND CHDR_INCLUDES lzma)
+endif()
 
 # zlib
 if (WITH_SYSTEM_ZLIB)
-- 
2.41.0

