From 4bbd826e84887394a839d523e19d757fd3c1f882 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 6 Nov 2020 19:42:31 -0300
Subject: [PATCH] Use system lzma-sdk

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 87f7c90..532c2dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ option(BUILD_STATIC_LIBS "Build libchdr also as a static library" ON)
 option(BUILD_SHARED_LIBS "Build libchdr also as a shared library" ON)
 option(WITH_SYSTEM_FLAC "Use system provided FLAC library" OFF)
 option(WITH_SYSTEM_ZLIB "Use system provided zlib library" OFF)
+option(WITH_SYSTEM_LZMA "Use system provided lzma-sdk library" OFF)
 
 if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden -fPIC -O3 -flto")
@@ -64,9 +65,15 @@ else()
 endif()
 
 # lzma
+if (WITH_SYSTEM_LZMA)
+  pkg_check_modules(LZMA REQUIRED lzmasdk-c)
+  list(APPEND PLATFORM_INCLUDES ${LZMA_INCLUDE_DIRS})
+  list(APPEND PLATFORM_LIBS ${LZMA_LIBRARIES})
+else()
 add_subdirectory(deps/lzma-19.00 EXCLUDE_FROM_ALL)
   list(APPEND CHDR_LIBS lzma)
   list(APPEND CHDR_INCLUDES lzma)
+endif()
 
 # zlib
 if (WITH_SYSTEM_ZLIB)
-- 
2.28.0

