From e4ce525656ce8d73de076a701f070aa793cc00b7 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 11 Nov 2021 20:36:39 -0300
Subject: [PATCH] Use system lzma-sdk

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e13345d..9a15c09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,7 @@ endif()
 option(INSTALL_STATIC_LIBS "Install static libraries" OFF)
 option(WITH_SYSTEM_ZLIB "Use system provided zlib library" OFF)
 option(WITH_SYSTEM_ZSTD "Use system provided zstd library" OFF)
+option(WITH_SYSTEM_LZMA "Use system provided lzma-sdk library" OFF)
 
 option(BUILD_LTO "Compile libchdr with link-time optimization if supported" OFF)
 if(BUILD_LTO)
@@ -34,11 +35,18 @@ include(GNUInstallDirs)
 
 
 # lzma
+if (WITH_SYSTEM_LZMA)
+  include(FindPkgConfig)
+  pkg_check_modules(LZMA REQUIRED lzmasdk-c)
+  list(APPEND CHDR_INCLUDES ${LZMA_INCLUDE_DIRS})
+  list(APPEND CHDR_LIBS ${LZMA_LIBRARIES})
+else()
 if(NOT TARGET lzma)
   add_subdirectory(deps/lzma-24.05 EXCLUDE_FROM_ALL)
 endif()
 list(APPEND CHDR_LIBS lzma)
 list(APPEND CHDR_INCLUDES lzma)
+endif()
 
 # zlib
 if (WITH_SYSTEM_ZLIB)
-- 
2.50.1

