From 861022f370c297fb79abbe2e63a68096fc2943fb Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 23 Mar 2021 18:07:46 -0300
Subject: [PATCH] Use system lzma-sdk

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee21eaa..a1f456f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,7 @@ set(CHDR_VERSION_MINOR 1)
 option(BUILD_SHARED_LIBS "Build libchdr also as a shared library" ON)
 option(INSTALL_STATIC_LIBS "Install static libraries" OFF)
 option(WITH_SYSTEM_ZLIB "Use system provided zlib library" OFF)
+option(WITH_SYSTEM_LZMA "Use system provided lzma-sdk library" OFF)
 
 if(CMAKE_C_COMPILER_ID MATCHES "GNU")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden -fPIC -O3 -flto")
@@ -47,9 +48,15 @@ endif()
 
 
 # lzma
+if (WITH_SYSTEM_LZMA)
+  pkg_check_modules(LZMA REQUIRED lzmasdk-c)
+  list(APPEND PLATFORM_INCLUDES ${LZMA_INCLUDE_DIRS})
+  list(APPEND PLATFORM_LIBS ${LZMA_LIBRARIES})
+else()
 add_subdirectory(deps/lzma-19.00 EXCLUDE_FROM_ALL)
   list(APPEND CHDR_LIBS lzma)
   list(APPEND CHDR_INCLUDES lzma)
+endif()
 
 # zlib
 if (WITH_SYSTEM_ZLIB)
-- 
2.30.2

