From 019d963dd21bd1e81517dff83ab6caafff3fd3ac Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 6 Nov 2020 19:37:51 -0300
Subject: [PATCH] Optional static library

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 725a2b5..92a8f99 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@ project(chdr C)
 set(CHDR_VERSION_MAJOR 0)
 set(CHDR_VERSION_MINOR 1)
 
+option(BUILD_STATIC_LIBS "Build libchdr also as a static library" ON)
 option(BUILD_SHARED_LIBS "Build libchdr also as a shared library" ON)
 option(WITH_SYSTEM_FLAC "Use system provided FLAC library" OFF)
 option(WITH_SYSTEM_ZLIB "Use system provided zlib library" OFF)
@@ -92,10 +93,12 @@ set(CHDR_SOURCES
 
 list(APPEND CHDR_INCLUDES include ${CMAKE_CURRENT_BINARY_DIR}/include)
 
+if (BUILD_STATIC_LIBS)
 add_library(chdr-static STATIC ${CHDR_SOURCES})
 target_include_directories(chdr-static PRIVATE ${CHDR_INCLUDES} ${PLATFORM_INCLUDES})
 target_compile_definitions(chdr-static PRIVATE ${CHDR_DEFS})
 target_link_libraries(chdr-static ${CHDR_LIBS} ${PLATFORM_LIBS})
+endif()
 
 if (BUILD_SHARED_LIBS)
   set(LIBS ${PLATFORM_LIBS})
-- 
2.28.0

