diff -up VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup	2019-01-14 12:49:14.000000000 -0200
+++ VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/Makefile.kmk	2019-01-15 15:33:52.925031712 -0200
@@ -190,10 +190,6 @@ ifdef VBOX_WITH_VIDEOHWACCEL
 else ifdef VBOX_GUI_USE_QGL
  VirtualBox_DEFS          += VBOX_GUI_USE_QGL
 endif
-ifdef VBOX_WITH_LIBCURL
- VirtualBox_DEFS          += VBOX_GUI_WITH_NETWORK_MANAGER
- VBOX_GUI_WITH_NETWORK_MANAGER := 1
-endif
 if  "$(KBUILD_TYPE)" != "release" || defined(VBOX_GUI_WITH_EXTRADATA_MANAGER_UI)
  VirtualBox_DEFS          += VBOX_GUI_WITH_EXTRADATA_MANAGER_UI
 endif
@@ -262,10 +258,6 @@ ifdef VBOX_GUI_WITH_SHARED_LIBRARY
  else ifdef VBOX_GUI_USE_QGL
   VirtualBoxVM_DEFS          += VBOX_GUI_USE_QGL
  endif
- ifdef VBOX_WITH_LIBCURL
-  VirtualBoxVM_DEFS          += VBOX_GUI_WITH_NETWORK_MANAGER
-  VBOX_GUI_WITH_NETWORK_MANAGER := 1
- endif
  if  "$(KBUILD_TYPE)" != "release" || defined(VBOX_GUI_WITH_EXTRADATA_MANAGER_UI)
   VirtualBoxVM_DEFS          += VBOX_GUI_WITH_EXTRADATA_MANAGER_UI
  endif
@@ -318,10 +310,6 @@ ifdef VBOX_GUI_WITH_SHARED_LIBRARY
  else ifdef VBOX_GUI_USE_QGL
   VBoxGlobal_DEFS        += VBOX_GUI_USE_QGL
  endif
- ifdef VBOX_WITH_LIBCURL
-  VBoxGlobal_DEFS        += VBOX_GUI_WITH_NETWORK_MANAGER
-  VBOX_GUI_WITH_NETWORK_MANAGER := 1
- endif
  if  "$(KBUILD_TYPE)" != "release" || defined(VBOX_GUI_WITH_EXTRADATA_MANAGER_UI)
   VBoxGlobal_DEFS        += VBOX_GUI_WITH_EXTRADATA_MANAGER_UI
  endif
diff -up VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp.noup VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp
--- VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp.noup	2019-01-14 12:49:34.000000000 -0200
+++ VirtualBox-6.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp	2019-01-15 15:35:02.134832166 -0200
@@ -39,9 +39,9 @@
 #include "UIModalWindowManager.h"
 #include "UIExtraDataManager.h"
 #include "UIMedium.h"
-#ifdef VBOX_OSE
+#ifdef VBOX_GUI_WITH_NETWORK_MANAGER
 # include "UIDownloaderUserManual.h"
-#endif /* VBOX_OSE */
+#endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 #include "VBoxAboutDlg.h"
 #include "UIHostComboEditor.h"
 #ifdef VBOX_WS_MAC
@@ -2761,7 +2761,7 @@ void UIMessageCenter::sltShowHelpAboutDi
 
 void UIMessageCenter::sltShowHelpHelpDialog()
 {
-#ifndef VBOX_OSE
+#ifndef VBOX_GUI_WITH_NETWORK_MANAGER
     /* For non-OSE version we just open it: */
     sltShowUserManual(vboxGlobal().helpFile());
 #else /* #ifndef VBOX_OSE */
@@ -2790,7 +2790,7 @@ void UIMessageCenter::sltShowHelpHelpDia
         /* Start downloading: */
         pDl->start();
     }
-#endif /* #ifdef VBOX_OSE */
+#endif /* #ifdef VBOX_GUI_WITH_NETWORK_MANAGER */
 }
 
 void UIMessageCenter::sltResetSuppressedMessages()
