--- ./configure.orig	2017-03-08 23:58:22.033228287 +0000
+++ ./configure	2017-03-08 23:58:27.288283387 +0000
@@ -447,7 +447,7 @@ check_gcc()
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
              -o \( $cc_maj -eq 5 -a $cc_min -gt 9 \) \
              -o \( $cc_maj -eq 6 -a $cc_min -gt 9 \) \
-             -o $cc_maj -gt 6 ]; then
+             -o $cc_maj -gt 7 ]; then
         log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x, gcc 5.x or gcc 6.x"
         fail really
       else
