From 839e1713f79b6ef493edfb0579c474e700de9050 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 29 Jan 2022 16:29:41 -0300
Subject: [PATCH] glslang: build as static

---
 3rdparty/glslang/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/3rdparty/glslang/CMakeLists.txt b/3rdparty/glslang/CMakeLists.txt
index 3db3d21..0517630 100644
--- a/3rdparty/glslang/CMakeLists.txt
+++ b/3rdparty/glslang/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_library(glslang
+add_library(glslang STATIC
 	glslang/glslang/CInterface/glslang_c_interface.cpp
 	glslang/glslang/GenericCodeGen/CodeGen.cpp
 	glslang/glslang/GenericCodeGen/Link.cpp
-- 
2.34.1

