From af96e2bdc1b07ea648dce3a76997ebff65358eb7 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 21 Oct 2020 20:12:00 -0300
Subject: [PATCH] Use shared fmt

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bc43a8..47eacf4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,8 +42,13 @@ if(NOT NO_TRANSLATION)
 endif()
 
 # 3rdpary Libraries
-if(EXISTS "${CMAKE_SOURCE_DIR}/3rdparty/fmt/fmt")
+find_package(fmt 7.0.3)
+if(fmt_FOUND)
+  message(STATUS "Using shared fmt ${fmt_VERSION}")
+else()
+  if(EXISTS "${CMAKE_SOURCE_DIR}/3rdparty/fmt/fmt")
     add_subdirectory(3rdparty/fmt/fmt)
+  endif()
 endif()
 
 # make common
-- 
2.28.0

