From 76df10e0eda7ee5c1121b42d7977d4d0c51b4a16 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 24 Sep 2021 22:08:46 -0300
Subject: [PATCH] common: build as static

---
 common/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 9444b2f..d09f667 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -5,7 +5,7 @@ if(NOT TOP_CMAKE_WAS_SOURCED)
 	It is advice to delete all wrongly generated cmake stuff => CMakeFiles & CMakeCache.txt")
 endif(NOT TOP_CMAKE_WAS_SOURCED)
 
-add_library(common)
+add_library(common STATIC)
 
 # x86emitter sources
 target_sources(common PRIVATE
-- 
2.31.1

