From 289a2a602189013835682c2559c93270a48e90c2 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 16 Sep 2023 18:28:41 -0300
Subject: [PATCH] Lower the SDL2 requirement a bit

---
 cmake/SearchForStuff.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 92e554f..874d89d 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -96,7 +96,7 @@ endif(WIN32)
 find_package(Threads REQUIRED)
 
 # Also need SDL2.
-find_package(SDL2 2.28.2 REQUIRED)
+find_package(SDL2 2.26.3 REQUIRED)
 
 set(ACTUALLY_ENABLE_TESTS ${ENABLE_TESTS})
 if(ENABLE_TESTS)
-- 
2.41.0

