From 8d58fbf24cf285c252493ebd44cec67dcbd7f18f Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 29 Jan 2022 16:30:13 -0300
Subject: [PATCH] imgui: build as static

---
 3rdparty/imgui/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/3rdparty/imgui/CMakeLists.txt b/3rdparty/imgui/CMakeLists.txt
index 6226953..44c2d41 100644
--- a/3rdparty/imgui/CMakeLists.txt
+++ b/3rdparty/imgui/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_library(imgui
+add_library(imgui STATIC
 	include/imconfig.h
 	include/imgui.h
 	include/imgui_internal.h
-- 
2.34.1

