From 48c78e81eca7ed69264e47494eecd9aca5b69336 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 2 Oct 2020 15:51:59 -0300
Subject: [PATCH] Disable setcap on installation

---
 pcsx2/CMakeLists.txt | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index bca3c21..c08b65c 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -808,11 +808,3 @@ if (APPLE)
 	)
 	add_dependencies(pcsx2-postprocess-bundle ${Output})
 endif()
-
-if(dev9ghzdrk)
-    if(PACKAGE_MODE)
-        install(CODE "execute_process(COMMAND /bin/bash -c \"echo 'Enabling networking capability on Linux...';set -x; [ -f ${BIN_DIR}/${Output} ] && sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' ${BIN_DIR}/${Output}; set +x\")")
-    else()
-        install(CODE "execute_process(COMMAND /bin/bash -c \"echo 'Enabling networking capability on Linux...';set -x; [ -f ${CMAKE_SOURCE_DIR}/bin/${Output} ] && sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' ${CMAKE_SOURCE_DIR}/bin/${Output}; set +x\")")
-    endif()
-endif()
\ No newline at end of file
-- 
2.28.0

