From 5f053a90983510aecc2209cd53f16b8aaa4b4b79 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 31 Aug 2020 09:41:53 -0300
Subject: [PATCH] Disable setcap on installation

---
 pcsx2/CMakeLists.txt | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 5026ba9..5415b49 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -748,11 +748,3 @@ if (APPLE)
 	)
 	add_dependencies(pcsx2-postprocess-bundle ${Output})
 endif()
-
-if(dev9ghzdrk)
-    if(PACKAGE_MODE)
-        install(CODE "execute_process(COMMAND /bin/bash -c \"echo 'Enabling networking capability on Linux...';set -x; [ -f ${BIN_DIR}/${Output} ] && sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' ${BIN_DIR}/${Output}; set +x\")")
-    else()
-        install(CODE "execute_process(COMMAND /bin/bash -c \"echo 'Enabling networking capability on Linux...';set -x; [ -f ${CMAKE_SOURCE_DIR}/bin/${Output} ] && sudo setcap 'CAP_NET_RAW+eip CAP_NET_ADMIN+eip' ${CMAKE_SOURCE_DIR}/bin/${Output}; set +x\")")
-    endif()
-endif()
-- 
2.26.2

