From e60b9580f755cdf348afc1cad2d8f69831c14f19 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 3 Jan 2023 18:40:09 -0300
Subject: [PATCH] Set datadir to RPM packaging

---
 pcsx2/Pcsx2Config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
index f3782b2..2c249b7 100644
--- a/pcsx2/Pcsx2Config.cpp
+++ b/pcsx2/Pcsx2Config.cpp
@@ -1529,7 +1529,7 @@ void EmuFolders::SetResourcesDirectory()
 {
 #ifndef __APPLE__
 	// On Windows/Linux, these are in the binary directory.
-	Resources = Path::Combine(AppRoot, "resources");
+	Resources = "_RPM_DATADIR_/resources";
 #else
 	// On macOS, this is in the bundle resources directory.
 	Resources = Path::Canonicalize(Path::Combine(AppRoot, "../Resources"));
-- 
2.40.1

