From d3b6bb5842b65a9708e6d0d2b4df9f845ff272a7 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 4 Jul 2024 19:25:28 -0300
Subject: [PATCH] cmake: use system discord-rpc

---
 cmake/SearchForStuff.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 202d968..00d0959 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -125,6 +125,11 @@ else()
 add_subdirectory(3rdparty/libzip EXCLUDE_FROM_ALL)
 endif()
 add_subdirectory(3rdparty/rcheevos EXCLUDE_FROM_ALL)
+find_package(DiscordRPC 3.4.0)
+if(DiscordRPC_FOUND)
+	message(STATUS "Using the system DiscordRPC")
+	alias_library(discord-rpc DiscordRPC::discord-rpc)
+else()
 pkg_search_module(rapidjson IMPORTED_TARGET RapidJSON)
 if(rapidjson_FOUND)
 	message(STATUS "Using the system rapidjson")
@@ -133,6 +138,7 @@ else()
 add_subdirectory(3rdparty/rapidjson EXCLUDE_FROM_ALL)
 endif()
 add_subdirectory(3rdparty/discord-rpc EXCLUDE_FROM_ALL)
+endif()
 add_subdirectory(3rdparty/freesurround EXCLUDE_FROM_ALL)
 
 find_package(xbyak)
-- 
2.45.2

