From a122b674ebc3451aaaec5d8f7a49816524074d71 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 2 Apr 2024 17:39:57 -0300
Subject: [PATCH] RPM fixes

---
 include/librashader_ld.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/librashader_ld.h b/include/librashader_ld.h
index dd9eaf0..1dc89fc 100644
--- a/include/librashader_ld.h
+++ b/include/librashader_ld.h
@@ -73,7 +73,7 @@ typedef void *_LIBRASHADER_IMPL_HANDLE;
         }                                                \
     }
 typedef void *_LIBRASHADER_IMPL_HANDLE;
-#define _LIBRASHADER_LOAD dlopen("librashader.so", RTLD_LAZY)
+#define _LIBRASHADER_LOAD dlopen("librashader.so._RPM_SO_VER_", RTLD_LAZY)
 #endif
 
 #include "librashader.h"
-- 
2.44.0

