--- VK9-0.26.0/meson.build.orig	2018-07-30 10:09:55.998262057 -0300
+++ VK9-0.26.0/meson.build	2018-07-30 10:29:18.641873518 -0300
@@ -11,9 +11,12 @@ endif
 
 add_global_arguments( ['-DUNICODE', '-D_UNICODE'], language : 'cpp')
 
+vk9_include_directories = include_directories('./include')
+vk9_library_path = meson.source_root() + '/lib'
+
 boost_dep = dependency('boost', modules : ['system', 'filesystem', 'thread', 'program_options', 'log', 'log_setup'])
-vulkan_dep= dependency('vulkan-1')
-eigen_dep = dependency('eigen')
+vulkan_dep = vk9_compiler.find_library('vulkan-1', dirs : vk9_library_path)
+eigen_dep = dependency('eigen3')
 
 vulkan_defs = ['-DVK_USE_PLATFORM_WIN32_KHR', '-DVK_PROTOTYPES', '-DVULKAN_HPP_NO_EXCEPTIONS',
 	'-DVULKAN_HPP_NO_SMART_HANDLE', '-DVK_KHR_push_descriptor', '-DVK_EXT_debug_report']
