diff -up mesa-22.3.0-rc3/src/glx/drisw_glx.c.4~ mesa-22.3.0-rc3/src/glx/drisw_glx.c
--- mesa-22.3.0-rc3/src/glx/drisw_glx.c.4~	2022-11-18 00:08:00.761471167 +0100
+++ mesa-22.3.0-rc3/src/glx/drisw_glx.c	2022-11-18 00:10:24.383355981 +0100
@@ -1055,7 +1055,9 @@ driswCreateScreen(int screen, struct glx
 {
    const struct drisw_display *pdpyp = (struct drisw_display *)priv->driswDisplay;
    if (pdpyp->zink && !debug_get_bool_option("LIBGL_KOPPER_DISABLE", false)) {
-      return driswCreateScreenDriver(screen, priv, "zink");
+      struct glx_screen * scr = driswCreateScreenDriver(screen, priv, "zink");
+      if (scr)
+         return scr;
    }
 
    return driswCreateScreenDriver(screen, priv, "swrast");
