diff -up mesa-25.2.0-rc1/meson.build.1~ mesa-25.2.0-rc1/meson.build
--- mesa-25.2.0-rc1/meson.build.1~	2025-07-16 16:42:48.000000000 +0200
+++ mesa-25.2.0-rc1/meson.build	2025-07-16 22:51:23.284411698 +0200
@@ -1750,6 +1750,7 @@ dep_llvm = dependency(
   optional_modules : llvm_optional_modules,
   required : with_llvm,
   static : not _shared_llvm,
+  method: 'config-tool',
   fallback : ['llvm', 'dep_llvm'],
   include_type : 'system',
 )
