From 517b63292051a2b5572f0b45cc8ff1fc9f03dbef Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 13 May 2023 17:37:23 -0300
Subject: [PATCH] fix build with bundled openh264

---
 cmake/external.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/external.cmake b/cmake/external.cmake
index 4147d90..3698044 100644
--- a/cmake/external.cmake
+++ b/cmake/external.cmake
@@ -129,7 +129,7 @@ endfunction()
 
 # libopenh264
 function(link_libopenh264 target_name)
-    if (TG_OWT_PACKAGED_BUILD)
+    if (TG_OWT_PACKAGED_BUILD_DISABLED)
         find_package(PkgConfig REQUIRED)
         pkg_check_modules(LIBOPENH264 openh264)
         set(LIBOPENH264_FOUND ${LIBOPENH264_FOUND} PARENT_SCOPE)
-- 
2.40.1

