From bbd0b092c9a7327e73d394ed63b40240502d2bd9 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 1 Feb 2022 10:24:16 -0300
Subject: [PATCH] add missing libdrm includedir

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81fb51f..fde392c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,6 +124,10 @@ if (TG_OWT_USE_PIPEWIRE)
             ${DRM_INCLUDE_DIRS}
         )
     else()
+        target_include_directories(tg_owt
+        PRIVATE
+            ${DRM_INCLUDE_DIRS}
+        )
         link_pipewire(tg_owt)
     endif()
 endif()
-- 
2.34.1

