From 9435f28cec54d7f75e6afd060f5aaed2f24e930f Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 11 Jan 2021 11:46:43 -0300
Subject: [PATCH] Fix undefined reference

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 333c41c..68e01b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1970,6 +1970,15 @@ PRIVATE
     stats/rtc_stats_report.cc
     stats/rtc_stats.cc
     stats/rtcstats_objects.cc
+
+    api/numerics/samples_stats_counter.cc
+    api/numerics/samples_stats_counter.h
+    modules/video_coding/deprecated/nack_module.cc
+    modules/video_coding/deprecated/nack_module.h
+    modules/rtp_rtcp/source/deprecated/deprecated_rtp_sender_egress.cc
+    modules/rtp_rtcp/source/deprecated/deprecated_rtp_sender_egress.h
+    video/video_stream_decoder_impl.cc
+    video/video_stream_decoder_impl.h
 )
 if (NOT arm_use_neon)
     add_sublibrary(avx2
-- 
2.29.2

