--- src/xdgmenumaker.orig	2018-10-03 23:02:43.160073634 -0300
+++ src/xdgmenumaker	2018-10-03 23:04:42.331946100 -0300
@@ -139,6 +139,10 @@ de = dentry.DesktopEntry(filename=deskto
                          'xdgmenumaker-system.directory')
 system = de.getName().encode('utf-8')
 system_icon = de.getIcon()
+de = dentry.DesktopEntry(filename=desktop_dir +
+                         'Wine.directory')
+wine = de.getName().encode('utf-8')
+wine_icon = de.getIcon()
 de = dentry.DesktopEntry(filename=desktop_dir + 'xdgmenumaker-other.directory')
 other = de.getName().encode('utf-8')
 other_icon = de.getIcon()
@@ -381,7 +385,8 @@ def clean_up_categories(categories):
         "Office":       office,
         "System":       system,
         "Settings":     settings,
-        "Utility":      accessories
+        "Utility":      accessories,
+        "X-Wine":       wine
     }
     category = other
     for candidate in categories:
@@ -522,6 +527,8 @@ def category_icon(category):
         icon = settings_icon
     elif category == system:
         icon = system_icon
+    elif category == wine:
+        icon = wine_icon
     elif category == other:
         icon = other_icon
     else:
