From ab90b9bd7392299f9257341ced7a4be2aed45d12 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Wed, 30 Mar 2022 08:33:19 -0300
Subject: [PATCH] Use system qhexedit

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8d2407..af6c74e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,8 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release CACHE STRING "" FORCE)
 endif()
 
+option(USE_SYSTEM_QHEXEDIT "Dynamically link against system qhexedit" OFF)
+
 #Set Deployed Binary Name
 if(WIN32 OR APPLE)
     set(BIN_NAME "Black_Chocobo")
@@ -78,5 +80,11 @@ endif()
 add_subdirectory(deploy)
 add_subdirectory(icons)
 add_subdirectory(translations)
+if(USE_SYSTEM_QHEXEDIT)
+find_package(PkgConfig REQUIRED)
+pkg_search_module(QHEXEDIT REQUIRED qhexedit2-qt6)
+add_library(QHEXEDIT INTERFACE IMPORTED GLOBAL)
+else()
 add_subdirectory(qhexedit)
+endif()
 add_subdirectory(src)
-- 
2.41.0

