From f56f201f10b451fc90e23008bb274bc379481ac3 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 13 Sep 2022 00:08:32 -0300
Subject: [PATCH] cmake: do not install dbg files

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 879151c..8d6e5a6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -104,7 +104,7 @@ macro(MAKE_LIBRARY LIB_TARGET HEADER_INSTALL_DIR)
         COMPONENT ff7tk_headers
     )
 
-    if(UNIX)
+    if(UNIX_DISABLED)
         if(NOT APPLE)
             add_custom_command(TARGET ${LIB_TARGET} POST_BUILD
                 COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:${LIB_TARGET}> $<TARGET_FILE:${LIB_TARGET}>.dbg
-- 
2.37.3

