From 52f04f0d38ff45b163deb97893eeaf1104ed77df Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 2 Jun 2019 14:20:59 -0300
Subject: [PATCH] Decrease volume step_big to 5

---
 pulsemixer | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pulsemixer b/pulsemixer
index f4f825a..3ec10f5 100755
--- a/pulsemixer
+++ b/pulsemixer
@@ -1723,7 +1723,7 @@ class Config():
 
         class General:
             step = 1
-            step_big = 10
+            step_big = 5
             server = None
 
         self._more_keys = {'KEY_ESC': 27, 'KEY_TAB': 9, 'C': -96, 'M': 128}
-- 
2.21.0

