diff -up pulsemixer-1.4.0/pulsemixer.orig pulsemixer-1.4.0/pulsemixer
--- pulsemixer-1.4.0/pulsemixer.orig	2017-12-16 11:36:43.000000000 -0200
+++ pulsemixer-1.4.0/pulsemixer	2018-05-15 22:04:45.245977131 -0300
@@ -1597,9 +1597,9 @@ class Screen():
             elif c == curses.KEY_RIGHT or c == ord('l') or any([c & i for i in self.SCROLL_UP]):
                 bar.move(1, side)
             elif c == curses.KEY_SLEFT or c == ord('H'):
-                bar.move(-10, side)
+                bar.move(-5, side)
             elif c == curses.KEY_SRIGHT or c == ord('L'):
-                bar.move(10, side)
+                bar.move(5, side)
             elif c in self.DIGITS:
                 percent = int(chr(c)) * 10
                 bar.set(100 if percent == 0 else percent, side)
