--- _upstream/VirtualBox/VirtualBox-fdist/VirtualBox.spec	2017-01-18 19:16:50.898654090 -0200
+++ VirtualBox/VirtualBox.spec	2017-01-25 22:41:39.369964469 -0200
@@ -25,7 +25,7 @@
 Name:       VirtualBox
 Version:    5.1.14
 #Release:   1%%{?prerel:.%%{prerel}}%%{?dist}
-Release:    1%{?dist}
+Release:    1.chinfo%{?dist}
 Summary:    A general-purpose full virtualizer for PC hardware
 
 License:    GPLv2 or (GPLv2 and CDDL)
@@ -40,6 +40,9 @@ Source11:   vboxservice.service
 Source12:   96-vbox.preset
 Source20:   os_mageia.png
 Source21:   os_mageia_64.png
+Source100:   vboxlogo-bmp.sh
+Source101:   README.vboxlogo
+Source102:   vboxlogo.bmp
 Patch1:     VirtualBox-OSE-4.1.4-noupdate.patch
 Patch2:     VirtualBox-5.1.0-strings.patch
 Patch18:    VirtualBox-OSE-4.0.2-aiobug.patch
@@ -58,7 +61,9 @@ Patch37:    smap.diff
 # from Mageia
 Patch50:    VirtualBox-5.1.0-add-Mageia-support.patch
 Patch51:    VirtualBox-5.1.0-revert-VBox.sh.patch
-
+# Chinforinfula
+Patch100:   virtualbox-5.1.0-noupdate.patch
+Patch101:   virtualbox-5.1.0-default-to-linux26.patch
 
 BuildRequires:  kBuild >= 0.1.9998
 BuildRequires:  SDL-devel xalan-c-devel
@@ -278,10 +283,15 @@ rm -r src/libs/zlib-1.2.8/
 %patch37 -p1 -b .kernel4.7-smap
 %patch50 -p1 -b .mageia-support
 %patch51 -p1 -b .revert-VBox.sh
+%patch100 -p1 -b .noup
+%patch101 -p1 -b .defsys
 
 # CRLF->LF
 sed -i 's/\r//' COPYING
 
+cat %{SOURCE102} \
+  > src/VBox/Devices/Graphics/BIOS/ose_logo.bmp
+
 %build
 ./configure --disable-kmods \
 %if %{with webservice}
@@ -314,7 +324,10 @@ kmk %{_smp_mflags}    \
     KBUILD_VERBOSE=2   \
     TOOL_YASM_AS=yasm   \
     PATH_OUT="$PWD/obj"      \
-    VBOX_PATH_APP_PRIVATE=%{_libdir}/virtualbox         \
+    VBOX_PATH_APP_PRIVATE_ARCH=%{_libdir}/virtualbox         \
+    VBOX_PATH_SHARED_LIBS=%{_libdir}/virtualbox         \
+    VBOX_WITH_RUNPATH=%{_libdir}/virtualbox         \
+    VBOX_PATH_APP_PRIVATE=%{_datadir}/virtualbox         \
     VBOX_PATH_APP_DOCS=%{_docdir}/VirtualBox        \
     VBOX_WITH_TESTCASES= \
     VBOX_WITH_VALIDATIONKIT= \
@@ -336,7 +349,12 @@ kmk %{_smp_mflags}    \
     VBOX_PATH_DOCBOOK_DTD=/usr/share/sgml/docbook/xml-dtd-4.5/ \
     VBOX_PATH_DOCBOOK=/usr/share/sgml/docbook/xsl-stylesheets/ \
     VBOX_JAVA_HOME=%{_prefix}/lib/jvm/java \
-    VBOX_BUILD_PUBLISHER=%{publisher}
+    VBOX_BUILD_PUBLISHER=%{publisher} \
+    VBOX_WITH_REGISTRATION_REQUEST= \
+    VBOX_GUI_WITH_NETWORK_MANAGER= \
+    VBOX_WITH_UPDATE_REQUEST= \
+    VBOX_WITH_UPDATE=
+
 
 #    VBOX_WITH_ADDITION_DRIVERS = \
 #    VBOX_WITH_INSTALLER = 1 \
@@ -355,9 +373,10 @@ install -d %{buildroot}%{_sbindir}
 install -d %{buildroot}%{_libdir}
 install -d %{buildroot}%{_libdir}/virtualbox
 install -d %{buildroot}%{_libdir}/virtualbox/components
-install -d %{buildroot}%{_libdir}/virtualbox/nls
+install -d %{buildroot}%{_datadir}/virtualbox/nls
 install -d %{buildroot}%{_libdir}/virtualbox/ExtensionPacks
 install -d %{buildroot}%{_libdir}/virtualbox/sdk
+install -d %{buildroot}%{_datadir}/virtualbox
 install -d %{buildroot}%{_datadir}/pixmaps
 install -d %{buildroot}%{_datadir}/mime/packages
 install -d %{buildroot}%{_datadir}/icons
@@ -430,11 +449,25 @@ ln -s VBox %{buildroot}%{_bindir}/vboxwe
 %endif
 ln -s ../..%{_libdir}/virtualbox/vbox-img %{buildroot}%{_bindir}/vbox-img
 
+
+# rdesktop-vrdp
+install -p -m 0755 obj/bin/rdesktop-vrdp %{buildroot}%{_libdir}/virtualbox/
+cp -rp obj/bin/rdesktop-vrdp-keymaps %{buildroot}%{_libdir}/virtualbox/
+ln -sf rdesktop-vrdp-keymaps %{buildroot}%{_libdir}/virtualbox/keymaps
+
+cat > %{buildroot}%{_bindir}/rdesktop-vrdp <<'EOF'
+#!/bin/sh
+set -e
+cd %{_libdir}/virtualbox
+exec %{_libdir}/virtualbox/rdesktop-vrdp "$@"
+EOF
+chmod 0755 %{buildroot}%{_bindir}/rdesktop-vrdp
+
 # Components , preserve symlinks
 cp -a obj/bin/components/* %{buildroot}%{_libdir}/virtualbox/components/
 
 # Language files
-install -p -m 0755 -t %{buildroot}%{_libdir}/virtualbox/nls \
+install -p -m 0755 -t %{buildroot}%{_datadir}/virtualbox/nls \
     obj/bin/nls/*
 
 # SDK
@@ -445,6 +478,9 @@ popd
 cp -rp obj/bin/sdk/. %{buildroot}%{_libdir}/virtualbox/sdk
 rm -rf %{buildroot}%{_libdir}/virtualbox/sdk/installer
 
+ln -s ../..%{_libdir}/virtualbox/vboxshell.py \
+  %{buildroot}%{_bindir}/vboxshell
+
 # Icons
 install -p -m 0644 -t %{buildroot}%{_datadir}/pixmaps \
     obj/bin/VBox.png
@@ -537,7 +573,7 @@ install -p -m 0644 -D %{SOURCE7} %{build
 
 # Menu entry
 desktop-file-install --dir=%{buildroot}%{_datadir}/applications \
-    --remove-key=Encoding obj/bin/virtualbox.desktop
+    --remove-key=Encoding --add-category="Qt" obj/bin/virtualbox.desktop
 desktop-file-validate %{buildroot}%{_datadir}/applications/virtualbox.desktop
 
 #    --remove-key=DocPath
@@ -547,9 +583,6 @@ desktop-file-validate %{buildroot}%{_dat
 #  mv ExtensionPacks/VNC $RPM_BUILD_ROOT/usr/lib/virtualbox/ExtensionPacks
 #fi
 #set_selinux_permissions /usr/lib/virtualbox /usr/share/virtualbox
-#for i in rdesktop-vrdp.tar.gz rdesktop-vrdp-keymaps; do
-#    mv $i $RPM_BUILD_ROOT/usr/share/virtualbox; done
-#mv rdesktop-vrdp $RPM_BUILD_ROOT/usr/bin
 #
 # vboxautostart-service
 
@@ -706,6 +739,11 @@ getent group vboxsf >/dev/null || groupa
 %{_udevrulesdir}/60-vboxdrv.rules
 %{_prefix}/lib/modules-load.d/%{name}.conf
 %{_prefix}/lib/udev/VBoxCreateUSBNode.sh
+%{_bindir}/rdesktop-vrdp
+%{_libdir}/virtualbox/rdesktop-vrdp
+%{_libdir}/virtualbox/keymaps
+%{_libdir}/virtualbox/rdesktop-vrdp-keymaps
+%{_datadir}/virtualbox
 
 %files
 %{_bindir}/VirtualBox
@@ -713,7 +751,7 @@ getent group vboxsf >/dev/null || groupa
 %{_libdir}/virtualbox/VBoxTestOGL
 %{_libdir}/virtualbox/VirtualBox.so
 %{_libdir}/virtualbox/VBoxDbg.so
-%{_libdir}/virtualbox/nls
+%{_datadir}/virtualbox/nls
 %{_datadir}/pixmaps/*
 %{_datadir}/applications/*.desktop
 
@@ -730,6 +768,8 @@ getent group vboxsf >/dev/null || groupa
 %exclude %{_libdir}/virtualbox/sdk/bindings/xpcom/python
 
 %files -n python-%{name}
+%{_bindir}/vboxshell
+%{_libdir}/virtualbox/*.py*
 %{_libdir}/virtualbox/*.py*
 %{python2_sitelib}/virtualbox
 %{python2_sitelib}/vboxapi*
@@ -760,8 +800,18 @@ getent group vboxsf >/dev/null || groupa
 %{_datadir}/%{name}-kmod-%{version}
 
 %changelog
-* Tue Jan 17 2017 Sérgio Basto <sergio@serjux.com> - 5.1.14-1
-- Update VBox to 5.1.14
+* Wed Jan 18 2017 Phantom X <megaphantomx at bol dot com dot br> - 5.1.14-1
+- 5.1.14
+- Sync with RPMfusion
+
+* Sat Jan 07 2017 Phantom X <megaphantomx at bol dot com dot br> - 5.1.12-1
+- 5.1.12
+
+* Mon Nov 28 2016 Sérgio Basto <sergio@serjux.com> - 5.1.10-2
+- Use systemd-detect-virt to detect if we can install
+  VirtualBox-guest-additions, which fix an old problem which made not possible
+  install VirtualBox in a guest machine.
+  Removed conflicts between main package and guest-additions.
 
 * Tue Nov 22 2016 Sérgio Basto <sergio@serjux.com> - 5.1.10-1
 - New upstream release, 5.1.10
