--- ./configure.orig	2016-03-05 02:37:09.491895482 +0000
+++ ./configure	2016-03-05 02:53:57.136090980 +0000
@@ -441,7 +441,7 @@ check_gcc()
              -o \( $cc_maj -eq 4 -a $cc_min -lt 2 -a "$OS"  = "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
              -o \( $cc_maj -eq 5 -a $cc_min -gt 9 \) \
-             -o \( $cc_maj -eq 6 -a $cc_min -gt 1 \) \
+             -o \( $cc_maj -eq 6 -a $cc_min -gt 2 \) \
              -o $cc_maj -gt 6 ]; then
         log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x, gcc 5.x or gcc 6.x"
         fail really
