diff -rupN --no-dereference binutils-2.37/bfd/bfd-in2.h binutils-2.37-new/bfd/bfd-in2.h
--- binutils-2.37/bfd/bfd-in2.h	2021-07-08 13:37:19.000000000 +0200
+++ binutils-2.37-new/bfd/bfd-in2.h	2021-07-24 21:59:03.850951615 +0200
@@ -30,11 +30,6 @@
 #ifndef __BFD_H_SEEN__
 #define __BFD_H_SEEN__
 
-/* PR 14072: Ensure that config.h is included first.  */
-#if !defined PACKAGE && !defined PACKAGE_VERSION
-#error config.h must be included before this header
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
diff -rupN --no-dereference binutils-2.37/bfd/bfd-in.h binutils-2.37-new/bfd/bfd-in.h
--- binutils-2.37/bfd/bfd-in.h	2021-07-08 13:37:19.000000000 +0200
+++ binutils-2.37-new/bfd/bfd-in.h	2021-07-24 21:59:03.851951606 +0200
@@ -23,11 +23,6 @@
 #ifndef __BFD_H_SEEN__
 #define __BFD_H_SEEN__
 
-/* PR 14072: Ensure that config.h is included first.  */
-#if !defined PACKAGE && !defined PACKAGE_VERSION
-#error config.h must be included before this header
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
