From 98cece80775488bfd037207925f5c28f17989fac Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 18 Apr 2022 09:44:34 -0300
Subject: [PATCH] appearance: fix black border

---
 gtk/gtkwindow.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index c1bc10f..d6076bc 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -3163,11 +3163,7 @@ gtk_window_set_decorated (GtkWindow *window,
 gboolean
 gtk_window_get_decorated (GtkWindow *window)
 {
-  GtkWindowPrivate *priv = gtk_window_get_instance_private (window);
-
-  g_return_val_if_fail (GTK_IS_WINDOW (window), TRUE);
-
-  return priv->decorated;
+  return FALSE;
 }
 
 /**
-- 
2.35.1

