From 4be0f1fe8480bffa99d131c3470f81edf2b7d580 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 19 Oct 2025 09:10:20 -0300
Subject: [PATCH] staging: fixup 2

---
 server/queue.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/server/queue.c b/server/queue.c
index c8fb5cc..258a500 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -3404,8 +3404,6 @@ DECL_HANDLER(get_message)
         reply->wparam = timer->id;
         reply->lparam = timer->lparam;
         get_message_defaults( queue, &reply->x, &reply->y, &reply->time );
-        if (!(req->flags & PM_NOYIELD) && current->process->idle_event)
-            set_event( current->process->idle_event );
         goto found_msg;
     }
 
@@ -3419,8 +3417,6 @@ DECL_HANDLER(get_message)
         get_posted_message( queue, 0, get_win, WM_HOTKEY, WM_HOTKEY, req->flags, reply ))
         return;
 
-    if (get_win == -1 && current->process->idle_event) set_event( current->process->idle_event );
-
     SHARED_WRITE_BEGIN( queue_shm, queue_shm_t )
     {
         shared->wake_mask = req->wake_mask;
-- 
2.51.0

