From 9cde0f037ce7a79c6286ea3708dd8009f0451ca4 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 3 Apr 2023 11:44:09 -0300
Subject: [PATCH] proton win10 default fixup 1

---
 loader/wine.inf.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 8884e02..54989c4 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -2817,7 +2817,6 @@ HKLM,%CurrentVersionNT%,"DigitalProductId",1,00,00,00,00,00,00,00,00,00,00,00,\
 00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,\
 00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,\
 00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00
-HKLM,%CurrentVersionNT%,"EditionId",2,"Professional"
 HKLM,%CurrentVersionNT%,"ProductName",2,"Windows 10 Pro"
 HKLM,%Control%\ProductOptions,"ProductType",2,"WinNT"
 HKLM,%Control%\Windows,"CSDVersion",0x10003,0
-- 
2.40.0

