From a20c54461325eaefbcc9005935299111627a2d3a Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 28 Jan 2023 11:49:51 -0300
Subject: [PATCH] sharedgpures: fixup

---
 dlls/winevulkan/make_vulkan | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dlls/winevulkan/make_vulkan b/dlls/winevulkan/make_vulkan
index 2725143..53ed8b6 100755
--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -102,7 +102,6 @@ UNSUPPORTED_EXTENSIONS = [
     "VK_KHR_external_semaphore_win32",
     # Relates to external_semaphore and needs type conversions in bitflags.
     "VK_KHR_shared_presentable_image", # Needs WSI work.
-    "VK_KHR_video_queue", # TODO Video extensions use separate headers + xml
     "VK_KHR_win32_keyed_mutex",
     "VK_NV_external_memory_rdma", # Needs shared resources work.
 
-- 
2.39.1

