From 6f645ae6d4bcf6239abd73336b05e3e8b4c998a4 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 6 Jan 2024 11:44:45 -0300
Subject: [PATCH] ntdll: kernel soft dirty flags fixup 2

---
 dlls/ntdll/unix/virtual.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index ce885aa..05ffa37 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -3362,6 +3362,13 @@ void virtual_init(void)
     pthread_mutex_init( &virtual_mutex, &attr );
     pthread_mutexattr_destroy( &attr );
 
+#ifdef __aarch64__
+    host_addr_space_limit = get_host_addr_space_limit();
+    TRACE( "host addr space limit: %p\n", host_addr_space_limit );
+#else
+    host_addr_space_limit = address_space_limit;
+#endif
+
     if (!((env_var = getenv("WINE_DISABLE_KERNEL_WRITEWATCH")) && atoi(env_var))
             && (pagemap_reset_fd = open("/proc/self/pagemap_reset", O_RDONLY)) != -1)
     {
-- 
2.43.0

