From 60d1ef5a36798ab41cc3d900810f84d72d56b902 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 23 Mar 2024 09:54:31 -0300
Subject: [PATCH] tkg: no childwindow fixup 1

---
 dlls/winex11.drv/x11drv_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 4410e88..04c4040 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -71,6 +71,8 @@ Window root_window;
 BOOL usexvidmode = TRUE;
 BOOL usexrandr = TRUE;
 BOOL usexcomposite = TRUE;
+BOOL use_xfixes = FALSE;
+BOOL use_xpresent = FALSE;
 BOOL use_take_focus = TRUE;
 BOOL use_primary_selection = FALSE;
 BOOL use_system_cursors = TRUE;
-- 
2.44.0

