From d2d0038d3c0d01e7fe6a2132df39f2478f0a270c Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 9 Aug 2026 19:52:08 -0300
Subject: [PATCH] mr11476 fix up 1

---
 include/wine/server_protocol.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/wine/server_protocol.h b/include/wine/server_protocol.h
index 2dab6ee..1b050da 100644
--- a/include/wine/server_protocol.h
+++ b/include/wine/server_protocol.h
@@ -7179,6 +7179,6 @@ union generic_reply
     struct alpc_create_port_reply alpc_create_port_reply;
 };
 
-#define SERVER_PROTOCOL_VERSION 958
+#define SERVER_PROTOCOL_VERSION 957
 
 #endif /* __WINE_WINE_SERVER_PROTOCOL_H */
-- 
2.55.0

