From d589d28e9ed1e4c4f53b48ab53cb3f95cb47c092 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sun, 21 Apr 2024 16:44:59 -0300
Subject: [PATCH] tkg ntsync5 cpu topology fixup 1

---
 server/process.c | 2 ++
 server/process.h | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/server/process.c b/server/process.c
index 692fa33..b0e64e2 100644
--- a/server/process.c
+++ b/server/process.c
@@ -706,6 +706,8 @@ struct process *create_process( int fd, struct process *parent, unsigned int fla
     process->fast_sync       = NULL;
     memset( &process->image_info, 0, sizeof(process->image_info) );
     list_init( &process->rawinput_entry );
+    process->esync_fd        = -1;
+    process->fsync_idx       = 0;
     list_init( &process->kernel_object );
     list_init( &process->thread_list );
     list_init( &process->locks );
diff --git a/server/process.h b/server/process.h
index 2140427..f37c26f 100644
--- a/server/process.h
+++ b/server/process.h
@@ -85,8 +85,8 @@ struct process
     const struct rawinput_device *rawinput_kbd;   /* rawinput keyboard device, if any */
     struct list          rawinput_entry;  /* entry in the rawinput process list */
     struct list          kernel_object;   /* list of kernel object pointers */
-    pe_image_info_t      image_info;      /* main exe image info */
-    struct fast_sync    *fast_sync;       /* fast synchronization object */
+    int                  esync_fd;        /* esync file descriptor (signaled on exit) */
+    unsigned int         fsync_idx;
 };
 
 /* process functions */
-- 
2.44.0

