From 5848030e0d5f83ae9d13e5e46677f7ba09e69263 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 1 Nov 2021 22:37:17 -0300
Subject: [PATCH] bcrypt: add fcntl.h to gnutls

---
 dlls/bcrypt/gnutls.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/bcrypt/gnutls.c b/dlls/bcrypt/gnutls.c
index 2a43c67..6a4f3f5 100644
--- a/dlls/bcrypt/gnutls.c
+++ b/dlls/bcrypt/gnutls.c
@@ -25,6 +25,7 @@
 #include "config.h"
 #include "wine/port.h"
 
+#include <fcntl.h>
 #include <stdarg.h>
 
 #ifdef HAVE_GNUTLS_CIPHER_INIT
-- 
2.33.1

