From ba46024c875d1ca7cd92c1419fcf0aa36d34071d Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 23 Mar 2024 09:58:34 -0300
Subject: [PATCH] tkg: no childwindow fixup 2

---
 dlls/winex11.drv/x11drv_main.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 53c169c..91760e9 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -71,8 +71,6 @@ Window root_window;
 BOOL usexvidmode = TRUE;
 BOOL usexrandr = TRUE;
 BOOL usexcomposite = TRUE;
-BOOL use_xfixes = FALSE;
-BOOL use_xpresent = FALSE;
 BOOL use_take_focus = FALSE;
 BOOL use_primary_selection = FALSE;
 BOOL use_system_cursors = TRUE;
-- 
2.44.0

