From f2127aef062cd468b749d7184ce06e77ad26468c Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sun, 11 Jun 2023 10:34:15 -0300
Subject: [PATCH] proton-tkg-additions: fixup 1

---
 dlls/winex11.drv/mouse.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index 4782347..50b2718 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -1700,8 +1700,8 @@ BOOL X11DRV_ClipCursor( const RECT *clip, BOOL reset )
             struct x11drv_thread_data *data = x11drv_thread_data();
             if (data && data->clip_hwnd)
             {
-                if (EqualRect( clip, &clip_rect )) return TRUE;
-                if (clip_fullscreen_window( NtUserGetForegroundWindow(), TRUE )) return TRUE;
+                if (EqualRect( clip, &clip_rect ) || clip_fullscreen_window( foreground, TRUE ))
+                    return TRUE;
             }
         }
     }
-- 
2.41.0

