From d36b5e1f0bc4009fd2308a1be93c4c360b40ae72 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sun, 23 Jul 2023 20:33:31 -0300
Subject: [PATCH] fastsync-staging-protonify: fixup 1

---
 server/named_pipe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/server/named_pipe.c b/server/named_pipe.c
index 07ac400..0e3359f 100644
--- a/server/named_pipe.c
+++ b/server/named_pipe.c
@@ -183,7 +183,7 @@ static const struct object_ops pipe_server_ops =
     NULL,                         /* unlink_name */
     pipe_server_open_file,        /* open_file */
     no_kernel_obj_list,           /* get_kernel_obj_list */
-    async_close_obj_handle,       /* close_handle */
+    no_close_handle,              /* close_handle */
     pipe_server_destroy           /* destroy */
 };
 
@@ -229,7 +229,7 @@ static const struct object_ops pipe_client_ops =
     NULL,                         /* unlink_name */
     no_open_file,                 /* open_file */
     no_kernel_obj_list,           /* get_kernel_obj_list */
-    async_close_obj_handle,       /* close_handle */
+    no_close_handle,              /* close_handle */
     pipe_end_destroy              /* destroy */
 };
 
-- 
2.41.0

