From 217c86a45cc302e275240e5b46e29f52968cf9a0 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 13 Jan 2025 15:33:12 -0300
Subject: [PATCH] tkg ntsync7 staging protonify fixup 1

---
 include/config.h.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/config.h.in b/include/config.h.in
index 563b63a..6911d77 100644
--- a/include/config.h.in
+++ b/include/config.h.in
@@ -321,7 +321,7 @@
 /* Define to 1 if you have the 'posix_fallocate' function. */
 #undef HAVE_POSIX_FALLOCATE
 
-/* Define to 1 if you have the `ppoll' function. */
+/* Define to 1 if you have the 'ppoll' function. */
 #undef HAVE_PPOLL
 
 /* Define to 1 if you have the 'prctl' function. */
@@ -387,7 +387,7 @@
 /* Define to 1 if you have the `shm_open' function. */
 #undef HAVE_SHM_OPEN
 
-/* Define to 1 if `si_fd' is a member of `siginfo_t'. */
+/* Define to 1 if 'si_fd' is a member of 'siginfo_t'. */
 #undef HAVE_SIGINFO_T_SI_FD
 
 /* Define to 1 if you have the 'sigprocmask' function. */
-- 
2.47.1

