From e462a5222c4e653b79b33e41b687de297368adcb Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 21 Nov 2022 19:18:16 -0300
Subject: [PATCH] staging: update nvapi and nvencodeapi autoconf

---
 configure.ac | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2dc7cdd..040ba7c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -189,19 +189,16 @@ case $host in
     ;;
 esac
 
+enable_nvapi=${enable_nvapi:-i386}
+enable_nvapi64=${enable_nvapi64:-aarch64,x86_64}
+enable_nvencodeapi=${enable_nvencodeapi:-i386}
+enable_nvencodeapi64=${enable_nvencodeapi64:-aarch64,x86_64}
 enable_win16=${enable_win16:-i386}
 enable_win64=${enable_win64:-no}
 enable_wow64=${enable_wow64:-aarch64,x86_64}
 enable_wow64win=${enable_wow64win:-aarch64,x86_64}
 enable_wow64cpu=${enable_wow64cpu:-x86_64}
 
-if test "x$enable_win64" != "xyes"
-then
-  enable_win32="yes"
-else
-  enable_win32="no"
-fi
-
 dnl Disable winetest too if tests are disabled
 enable_winetest=${enable_winetest:-$enable_tests}
 
@@ -2939,14 +2936,14 @@ WINE_CONFIG_MAKEFILE(dlls/ntoskrnl.exe)
 WINE_CONFIG_MAKEFILE(dlls/ntoskrnl.exe/tests)
 WINE_CONFIG_MAKEFILE(dlls/ntprint)
 WINE_CONFIG_MAKEFILE(dlls/ntprint/tests)
-WINE_CONFIG_MAKEFILE(dlls/nvapi,enable_win32)
+WINE_CONFIG_MAKEFILE(dlls/nvapi)
 WINE_CONFIG_MAKEFILE(dlls/nvapi/tests)
-WINE_CONFIG_MAKEFILE(dlls/nvapi64,enable_win64)
+WINE_CONFIG_MAKEFILE(dlls/nvapi64)
 WINE_CONFIG_MAKEFILE(dlls/nvcuda)
 WINE_CONFIG_MAKEFILE(dlls/nvcuda/tests)
 WINE_CONFIG_MAKEFILE(dlls/nvcuvid)
-WINE_CONFIG_MAKEFILE(dlls/nvencodeapi,enable_win32)
-WINE_CONFIG_MAKEFILE(dlls/nvencodeapi64,enable_win64)
+WINE_CONFIG_MAKEFILE(dlls/nvencodeapi)
+WINE_CONFIG_MAKEFILE(dlls/nvencodeapi64)
 WINE_CONFIG_MAKEFILE(dlls/objsel)
 WINE_CONFIG_MAKEFILE(dlls/odbc32)
 WINE_CONFIG_MAKEFILE(dlls/odbcbcp)
-- 
2.38.1

