From bffa082b92bf6040107f15475bf4d4b6e57e499a Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 14 Jun 2024 23:26:40 -0300
Subject: [PATCH] tkg ntsync5 staging protonify fixup 1

---
 include/config.h.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/config.h.in b/include/config.h.in
index 6456c14..2da6750 100644
--- a/include/config.h.in
+++ b/include/config.h.in
@@ -381,7 +381,7 @@
 /* Define to 1 if you have the `shm_open' function. */
 #undef HAVE_SHM_OPEN
 
-/* Define to 1 if `si_fd' is a member of `siginfo_t'. */
+/* Define to 1 if 'si_fd' is a member of `siginfo_t'. */
 #undef HAVE_SIGINFO_T_SI_FD
 
 /* Define to 1 if you have the `sigprocmask' function. */
-- 
2.45.2

