From c72c7b964358496345d4de1ccdf7a3b58ab882a1 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 6 Jan 2024 11:43:05 -0300
Subject: [PATCH] ntdll: kernel soft dirty flags fixup 1

---
 dlls/ntdll/unix/virtual.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 857eae4..8dfdf79 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -3334,13 +3334,6 @@ void virtual_init(void)
     pthread_mutex_init( &virtual_mutex, &attr );
     pthread_mutexattr_destroy( &attr );
 
-#ifdef __aarch64__
-    host_addr_space_limit = get_host_addr_space_limit();
-    TRACE( "host addr space limit: %p\n", host_addr_space_limit );
-#else
-    host_addr_space_limit = address_space_limit;
-#endif
-
     if (preload_info && *preload_info)
         for (i = 0; (*preload_info)[i].size; i++)
             mmap_add_reserved_area( (*preload_info)[i].addr, (*preload_info)[i].size );
-- 
2.43.0

