From a82275f3a19ed41ed5b2e03176da709b32fbafcc Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 5 Apr 2024 23:29:15 -0300
Subject: [PATCH] tkg: proton tkg staging fixup 1

---
 dlls/wbemprox/builtin.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/wbemprox/builtin.c b/dlls/wbemprox/builtin.c
index fa4c6ee..0617e8b 100644
--- a/dlls/wbemprox/builtin.c
+++ b/dlls/wbemprox/builtin.c
@@ -2950,10 +2950,11 @@ static enum fill_status fill_networkadapter( struct table *table, const struct e
         rec->description          = wcsdup( aa->Description );
         rec->device_id            = wcsdup( device_id );
         rec->guid                 = get_networkadapter_guid( &aa->Luid );
-        rec->index                = aa->IfIndex;
-        rec->interface_index      = aa->IfIndex;
+        rec->index                = aa->u.s.IfIndex;
+        rec->interface_index      = aa->u.s.IfIndex;
         rec->mac_address          = get_mac_address( aa->PhysicalAddress, aa->PhysicalAddressLength );
         rec->manufacturer         = L"The Wine Project";
+        rec->netconnection_id     = NULL; /* FIXME Windows seems to fill this when it's connected and in use */
         rec->name                 = wcsdup( aa->FriendlyName );
         rec->netenabled           = connection_status ? -1 : 0;
         rec->netconnection_status = connection_status;
-- 
2.44.0

