From ff5f2c6967ea9ef335d0c9a1d4e1e9915f3cecae Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 14 Aug 2021 10:21:16 -0300
Subject: [PATCH] proton-tkg-staging temporary fix

---
 dlls/ws2_32/unixlib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ws2_32/unixlib.c b/dlls/ws2_32/unixlib.c
index 9dd91be..df5a54c 100644
--- a/dlls/ws2_32/unixlib.c
+++ b/dlls/ws2_32/unixlib.c
@@ -909,7 +909,7 @@ static NTSTATUS unix_gethostbyname( void *args )
     int locerr;
     int ret;
 
-    if (!strcmp(name, "download-alt.easyanticheat.net"))
+    if (!strcmp(params->name, "download-alt.easyanticheat.net"))
     {
         ERR("HACK: failing download-alt.easyanticheat.net resolution.\n");
         return HOST_NOT_FOUND;
-- 
2.31.1

