From 3404b4bd5a852e3f847e4b7dbf96dc91e51ba663 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 3 Apr 2023 10:55:06 -0300
Subject: [PATCH] sharedgpufences fences fixup 1

---
 dlls/winevulkan/make_vulkan | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dlls/winevulkan/make_vulkan b/dlls/winevulkan/make_vulkan
index 5c3ca3b..5fe4f62 100755
--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -100,7 +100,6 @@ UNSUPPORTED_EXTENSIONS = [
     "VK_GOOGLE_display_timing",
     "VK_KHR_external_fence_win32",
     "VK_KHR_external_semaphore_win32",
-    "VK_KHR_map_memory2", # Needs wow64 handling and is useless for now.
     # Relates to external_semaphore and needs type conversions in bitflags.
     "VK_KHR_shared_presentable_image", # Needs WSI work.
     "VK_KHR_video_queue", # TODO Video extensions use separate headers + xml
-- 
2.40.0

