From 3a3577faf6f47842db5386340260e0d89c4852fb Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 6 Dec 2024 19:47:47 -0300
Subject: [PATCH] mr6072 fixup 2

---
 server/registry.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/server/registry.c b/server/registry.c
index 2ed15bd..29c40b5 100644
--- a/server/registry.c
+++ b/server/registry.c
@@ -1887,11 +1887,18 @@ void init_registry(void)
                                     'C','u','r','r','e','n','t','V','e','r','s','i','o','n','\\',
                                     'P','e','r','f','l','i','b','\\',
                                     '0','0','9'};
+    static const WCHAR controlset[] = {'S','y','s','t','e','m','\\',
+                                       'C','u','r','r','e','n','t','C','o','n','t','r','o','l','S','e','t'};
+    static const WCHAR controlset001_path[] = {'\\','R','e','g','i','s','t','r','y','\\',
+                                               'M','a','c','h','i','n','e','\\',
+                                               'S','y','s','t','e','m','\\',
+                                               'C','o','n','t','r','o','l','S','e','t','0','0','1'};
     static const WCHAR software[] = {'S','o','f','t','w','a','r','e',};
     static const struct unicode_str root_name = { REGISTRY, sizeof(REGISTRY) };
     static const struct unicode_str HKLM_name = { HKLM, sizeof(HKLM) };
     static const struct unicode_str HKU_name = { HKU_default, sizeof(HKU_default) };
     static const struct unicode_str perflib_name = { perflib, sizeof(perflib) };
+    static const struct unicode_str controlset_name = { controlset, sizeof(controlset) };
 
     WCHAR *current_user_path;
     struct unicode_str current_user_str, name;
-- 
2.47.1

