From 8740baa06a3388ecd7db847d8efb2de9ac368d6d Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Fri, 17 May 2024 18:45:41 -0300
Subject: [PATCH] tkg: proton cpu topology overrides fixup 2

---
 dlls/ntdll/unix/system.c | 36 +++++++++++++++++++-----------------
 1 file changed, 19 insertions(+), 17 deletions(-)

diff --git a/dlls/ntdll/unix/system.c b/dlls/ntdll/unix/system.c
index 8fb43b7..212dd8c 100644
--- a/dlls/ntdll/unix/system.c
+++ b/dlls/ntdll/unix/system.c
@@ -1147,7 +1147,7 @@ static NTSTATUS create_logical_proc_info(void)
 
             for (j = 0; j < 4; j++)
             {
-                CACHE_DESCRIPTOR cache;
+                CACHE_DESCRIPTOR cache = { .Associativity = 8, .LineSize = 64, .Type = CacheUnified, .Size = 64 * 1024 };
                 ULONG_PTR mask = 0;
 
                 snprintf(name, sizeof(name), cache_info, cpu_id, j, "shared_cpu_map");
@@ -1161,23 +1161,25 @@ static NTSTATUS create_logical_proc_info(void)
                 cache.Level = r;
 
                 snprintf(name, sizeof(name), cache_info, cpu_id, j, "ways_of_associativity");
-                f = fopen(name, "r");
-                if(!f) continue;
-                fscanf(f, "%u", &r);
-                fclose(f);
-                cache.Associativity = r;
+                if ((f = fopen(name, "r")))
+                {
+                    fscanf(f, "%u", &r);
+                    fclose(f);
+                    cache.Associativity = r;
+                }
 
                 snprintf(name, sizeof(name), cache_info, cpu_id, j, "coherency_line_size");
-                f = fopen(name, "r");
-                if(!f) continue;
-                fscanf(f, "%u", &r);
-                fclose(f);
-                cache.LineSize = r;
+                if ((f = fopen(name, "r")))
+                {
+                    fscanf(f, "%u", &r);
+                    fclose(f);
+                    cache.LineSize = r;
+                }
 
                 snprintf(name, sizeof(name), cache_info, cpu_id, j, "size");
-                f = fopen(name, "r");
-                if(!f) continue;
-                fscanf(f, "%u%c", &r, &op);
+                if ((f = fopen(name, "r")))
+                {
+                    fscanf(f, "%u%c", &r, &op);
                     fclose(f);
                     if(op != 'K')
                         WARN("unknown cache size %u%c\n", r, op);
@@ -1185,9 +1187,9 @@ static NTSTATUS create_logical_proc_info(void)
                 }
 
                 snprintf(name, sizeof(name), cache_info, cpu_id, j, "type");
-                f = fopen(name, "r");
-                if(!f) continue;
-                fscanf(f, "%s", name)
+                if ((f = fopen(name, "r")))
+                {
+                    fscanf(f, "%s", name);
                     fclose(f);
                     if (!memcmp(name, "Data", 5))
                         cache.Type = CacheData;
-- 
2.45.0

