From e7a7bdcab7d1561266ba5909df1677806239232b Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 5 Mar 2022 00:48:11 -0300
Subject: [PATCH] tkg proton staging: restore dxgi for patch

---
 dlls/dxgi/Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dlls/dxgi/Makefile.in b/dlls/dxgi/Makefile.in
index e2c58f2..d582639 100644
--- a/dlls/dxgi/Makefile.in
+++ b/dlls/dxgi/Makefile.in
@@ -1,10 +1,10 @@
 EXTRADEFS = -DWINE_NO_LONG_TYPES
 MODULE    = dxgi.dll
 IMPORTLIB = dxgi
-IMPORTS   = gdi32 dxguid uuid wined3d user32
-EXTRAINCL = $(VKD3D_CFLAGS)
+IMPORTS   = gdi32 dxguid uuid wined3d user32 win32u
+EXTRAINCL = $(VKD3D_PE_CFLAGS)
 
-EXTRADLLFLAGS = -Wl,--image-base,0x7bb00000 -mcygwin
+EXTRADLLFLAGS = -Wl,--image-base,0x7bb00000
 
 C_SRCS = \
 	adapter.c \
-- 
2.35.1

