From 4268dec478813d3425696461d4307bb9e1ce2753 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 28 Jan 2023 11:51:58 -0300
Subject: [PATCH] sharedgpures: fixup 2

---
 dlls/winevulkan/make_vulkan | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/winevulkan/make_vulkan b/dlls/winevulkan/make_vulkan
index be2485b..1f34c7a 100755
--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -101,6 +101,7 @@ UNSUPPORTED_EXTENSIONS = [
     "VK_KHR_external_fence_win32",
     # Relates to external_semaphore and needs type conversions in bitflags.
     "VK_KHR_shared_presentable_image", # Needs WSI work.
+    "VK_KHR_video_queue", # TODO Video extensions use separate headers + xml
     "VK_NV_external_memory_rdma", # Needs shared resources work.
 
     # Extensions for other platforms
-- 
2.39.1

