From 29978a14969c167876eb8ffadfe7c8c5ac81cf11 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Sat, 12 Sep 2020 10:50:54 -0300
Subject: [PATCH] hotfix-rebased f2d60c1

---
 tools/makedep.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/makedep.c b/tools/makedep.c
index 26ffac2..ca1945d 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -217,6 +217,7 @@ struct makefile
     struct strarray crossobj_files;
     struct strarray unixobj_files;
     struct strarray res_files;
+    struct strarray font_files;
     struct strarray c2man_files;
     struct strarray debug_files;
     struct strarray dlldata_files;
@@ -2937,7 +2938,7 @@ static void output_source_sfd( struct makefile *make, struct incl_file *source,
         output( "%s: %s\n", ttf_file, source->filename );
         output( "\t%s -script %s %s $@\n",
                 fontforge, top_src_dir_path( make, "fonts/genttf.ff" ), source->filename );
-        if (!(source->file->flags & FLAG_SFD_FONTS)) output( "all: %s\n", ttf_file );
+        if (!(source->file->flags & FLAG_SFD_FONTS)) strarray_add( &make->font_files, ttf_obj );
     }
     if (source->file->flags & FLAG_INSTALL)
     {
@@ -4001,6 +4002,7 @@ static void output_sources( struct makefile *make )
         {
             output( "all:" );
             output_filenames_obj_dir( make, make->all_targets );
+            output_filenames_obj_dir( make, make->font_files );
             output( "\n" );
         }
         output_install_rules( make, INSTALL_LIB, "install-lib" );
-- 
2.26.2

