From 887d1dbaba833d4fd8ca113534c0113af19712b9 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 3 Apr 2023 11:45:44 -0300
Subject: [PATCH] proton win10 default fixup 2

---
 loader/wine.inf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 22bd540..3bc09df 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -2818,6 +2818,7 @@ HKLM,%CurrentVersionNT%,"DigitalProductId",2,00,00,00,00,00,00,00,00,00,00,00,\
 00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,\
 00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00
 HKLM,%CurrentVersionNT%,"DisplayVersion",2,"21H1"
+HKLM,%CurrentVersionNT%,"EditionId",2,"Professional"
 HKLM,%CurrentVersionNT%,"ProductName",2,"Windows 10 Pro"
 HKLM,%CurrentVersionNT%,"ReleaseId",,"2009"
 HKLM,%Control%\ProductOptions,"ProductType",2,"WinNT"
-- 
2.40.0

