From 5a113787434801f161f1178886f55d7f8b528a19 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 22 May 2023 09:44:41 -0300
Subject: [PATCH] proton tkg staging fixup

---
 dlls/mscoree/metahost.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/mscoree/metahost.c b/dlls/mscoree/metahost.c
index 309ad63..18a6fca 100644
--- a/dlls/mscoree/metahost.c
+++ b/dlls/mscoree/metahost.c
@@ -1812,6 +1812,8 @@ static MonoAssembly* CDECL wine_mono_assembly_preload_hook_v2_fn(MonoAssemblyNam
         }
     }
 
+    /* FIXME: We should search the given paths before the GAC. */
+
     if ((search_flags & ASSEMBLY_SEARCH_GAC) != 0)
     {
         stringnameW_size = MultiByteToWideChar(CP_UTF8, 0, stringname, -1, NULL, 0);
-- 
2.40.1

