diff -up wine-3.15/patches/patchinstall.sh.orig wine-3.15/patches/patchinstall.sh
--- wine-3.15/patches/patchinstall.sh.orig	2018-09-01 12:43:37.304114734 -0300
+++ wine-3.15/patches/patchinstall.sh	2018-09-01 12:45:52.729631681 -0300
@@ -7627,33 +7627,26 @@ fi
 # |
 # | This patchset fixes the following Wine bugs:
 # |   *	[#37042] Implement exclusive mode in PulseAudio backend
-# |   *	[#28282] Sound constantly crackling in a lot of games
 # |
 # | Modified files:
-# |   *	dlls/winepulse.drv/Makefile.in, dlls/winepulse.drv/mmdevdrv.c
+# |   *	dlls/mmdevapi/tests/render.c, dlls/winepulse.drv/Makefile.in, dlls/winepulse.drv/mmdevdrv.c
 # |
 if test "$enable_winepulse_PulseAudio_Support" -eq 1; then
 	patch_apply winepulse-PulseAudio_Support/0001-winepulse.drv-Use-a-separate-mainloop-and-ctx-for-pu.patch
-	patch_apply winepulse-PulseAudio_Support/0002-winepulse-Don-t-rely-on-pulseaudio-callbacks-for-tim.patch
-	patch_apply winepulse-PulseAudio_Support/0003-winepulse-expose-audio-devices-directly-to-programs.patch
+	patch_apply winepulse-PulseAudio_Support/0002-winepulse-expose-audio-devices-directly-to-programs.patch
+	patch_apply winepulse-PulseAudio_Support/0003-winepulse-implement-exclusive-mode.patch
 	patch_apply winepulse-PulseAudio_Support/0004-winepulse-fix-segfault-in-pulse_rd_loop.patch
 	patch_apply winepulse-PulseAudio_Support/0005-winepulse-implement-GetPropValue.patch
 	patch_apply winepulse-PulseAudio_Support/0006-winepulse-fetch-actual-program-name-if-possible.patch
 	patch_apply winepulse-PulseAudio_Support/0007-winepulse-return-PKEY_AudioEndpoint_PhysicalSpeakers.patch
-	patch_apply winepulse-PulseAudio_Support/0008-winepulse-Fix-up-recording.patch
-	patch_apply winepulse-PulseAudio_Support/0009-winepulse.drv-Fix-getting-the-same-timing-info.patch
-	patch_apply winepulse-PulseAudio_Support/0010-winepulse-Update-last-time-on-underrun.patch
 	(
 		printf '%s\n' '+    { "Sebastian Lackner", "winepulse.drv: Use a separate mainloop and ctx for pulse_test_connect.", 1 },';
-		printf '%s\n' '+    { "Andrew Eikum", "winepulse: Don'\''t rely on pulseaudio callbacks for timing.", 1 },';
 		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Expose audio devices directly to programs.", 1 },';
+		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Implement exclusive mode.", 1 },';
 		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Fix segfault in pulse_rd_loop.", 1 },';
 		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Implement GetPropValue.", 1 },';
 		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Fetch actual program name if possible.", 1 },';
 		printf '%s\n' '+    { "Mark Harmstone", "winepulse: Return PKEY_AudioEndpoint_PhysicalSpeakers device prop.", 1 },';
-		printf '%s\n' '+    { "Andrew Eikum", "winepulse: Fix up recording.", 1 },';
-		printf '%s\n' '+    { "Zhiyi Zhang", "winepulse.drv: Fix getting the same timing info.", 1 },';
-		printf '%s\n' '+    { "Andrew Eikum", "winepulse: Update last time on underrun.", 1 },';
 	) >> "$patchlist"
 fi
 
diff -up wine-3.15/patches/winepulse-PulseAudio_Support_old/definition.orig wine-3.15/patches/winepulse-PulseAudio_Support_old/definition
--- wine-3.15/patches/winepulse-PulseAudio_Support_old/definition.orig	2018-08-31 21:16:27.000000000 -0300
+++ wine-3.15/patches/winepulse-PulseAudio_Support_old/definition	2018-09-01 12:46:20.672754609 -0300
@@ -4,4 +4,3 @@ Fixes: Fix possible segfault in pulse_rd
 Fixes: Add support for GetPropValue to PulseAudio backend
 Fixes: Use actual program name if available to describe PulseAudio streams
 Fixes: Expose PKEY_AudioEndpoint_PhysicalSpeakers device property in PulseAudio driver
-Disabled: true
