From 1a51fb0e20696ebc915a71f29993c7587a0f3e91 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 18 Jan 2021 18:03:56 -0300
Subject: [PATCH] Fix gcc 10 i686 miscompilation

---
 Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index b52495f..a06d505 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -139,3 +139,7 @@ TAGS etags:
 tags ctags:
 	rm -f tags
 	(test -d .git && git ls-files || find -L $(srcdir) -name '*.[ch]' -print) | xargs ctags -a $(TAGSFLAGS)
+
+dlls/winevulkan/%.o dlls/winex11.drv/%.o: CFLAGS   += -fno-tree-dce
+
+dlls/d3d8/%.o dlls/d3d9/%.o dlls/d3d11/%.o dlls/vulkan-1/%.o: CROSSCFLAGS   += -fno-tree-dce
-- 
2.29.2

