From 3096c667f754d501e87da919674f67a028f2c9be Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Fri, 3 Apr 2020 10:07:20 -0300
Subject: [PATCH] Fix gcc 10 i686 miscompilation

---
 dlls/winevulkan/Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/winevulkan/Makefile.in b/dlls/winevulkan/Makefile.in
index 251de5c..d2ace62 100644
--- a/dlls/winevulkan/Makefile.in
+++ b/dlls/winevulkan/Makefile.in
@@ -1,6 +1,7 @@
 MODULE    = winevulkan.dll
 IMPORTLIB = winevulkan
 IMPORTS   = user32 gdi32 advapi32
+CFLAGS   += -fno-tree-dce
 
 C_SRCS = \
 	vulkan.c \
-- 
2.26.0

