@@ -, +, @@ 
 wow64.
--- a/dlls/ntdll/unix/virtual.c	
+++ a/dlls/ntdll/unix/virtual.c	
@@ -5302,7 +5302,10 @@ void virtual_set_large_address_space(void)
                 free_reserved_memory( 0, (char *)0x7ffe0000 );
 #endif
         }
-        else user_space_wow_limit = (is_large_address_aware() ? limit_4g : limit_2g) - 1;
+        else {
+            user_space_wow_limit = (is_large_address_aware() ? limit_4g : limit_2g) - 1;
+            mmap((void*)(0xffffffff-0x11000+1), 0x11000, PROT_NONE, MAP_FIXED | MAP_NORESERVE | MAP_PRIVATE | MAP_ANON, -1, 0);
+        }
     }
     else
     {
