--- _upstream/wine/wine-fdist/wine.spec	2017-01-25 19:26:49.208079929 -0200
+++ wine/wine.spec	2017-01-25 19:28:53.058893617 -0200
@@ -1,6 +1,7 @@
 # Compiling the preloader fails with hardening enabled
 %undefine _hardened_build
 
+%global rcrev 0
 %global no64bit   0
 %global winegecko 2.47
 %global winemono  4.6.4
@@ -10,6 +11,11 @@
 # uncomment to enable; comment-out to disable.
 %if 0%{?fedora}
 %global compholio 1
+%global compholiover 2.0
+
+# build with wine-d3d9-patches (nine), see:  https://github.com/sarnex/wine-d3d9-patches
+%global nine 1
+%global ninever 2.0
 %endif # 0%{?fedora}
 
 # binfmt macros for RHEL
@@ -20,16 +26,21 @@
 %{nil}
 %endif
 
+%if 0%{?rcrev}
+%global rctag .rc%rcrev
+%global rctagtarball -rc%rcrev
+%endif
+
 Name:           wine
 Version:        2.0
-Release:        1%{?dist}
+Release:        100%{?rctag}.chinfo%{?dist}
 Summary:        A compatibility layer for windows applications
 
 Group:          Applications/Emulators
 License:        LGPLv2+
 URL:            http://www.winehq.org/
-Source0:        http://downloads.sourceforge.net/wine/wine-%{version}.tar.bz2
-Source10:       http://downloads.sourceforge.net/wine/wine-%{version}.tar.bz2.sign
+Source0:        http://downloads.sourceforge.net/wine/wine-%{version}%{?rctagtarball}.tar.bz2
+Source10:       http://downloads.sourceforge.net/wine/wine-%{version}%{?rctagtarball}.tar.bz2.sign
 
 Source1:        wine.init
 Source2:        wine.systemd
@@ -48,6 +59,10 @@ Source106:      wine-winhelp.desktop
 Source107:      wine-wineboot.desktop
 Source108:      wine-wordpad.desktop
 Source109:      wine-oleview.desktop
+Source110:      wine-iexplore.desktop
+Source111:      wine-inetcpl.desktop
+Source112:      wine-joycpl.desktop
+Source113:      wine-taskmgr.desktop
 
 # build fixes
 
@@ -68,10 +83,24 @@ Source501:      wine-tahoma.conf
 Source502:      wine-README-tahoma
 
 Patch511:       wine-cjk.patch
+Patch599:       0003-winemenubuilder-silence-an-err.patch
+
+### AUR - https://aur.archlinux.org/packages/wine-gaming-nine
+# Steam patch, Crossover Hack version
+# https://bugs.winehq.org/show_bug.cgi?id=39403
+Patch600:       steam.patch
+# Wbemprox videocontroller query fix v2
+# https://bugs.winehq.org/show_bug.cgi?id=38879
+Patch601:       wbemprox_query_v2.patch
+# Keybind patch reversion
+Patch602:       keybindings.patch
 
 # wine compholio patches for wine-staging
 # pulseaudio-patch is covered by that patch-set, too.
-Source900: https://github.com/compholio/wine-compholio/archive/v%{version}.tar.gz#/wine-staging-%{version}.tar.gz
+Source900: https://github.com/compholio/wine-compholio/archive/v%{compholiover}.tar.gz#/wine-staging-%{compholiover}.tar.gz
+
+# wine-d3d9
+Source910: https://github.com/sarnex/wine-d3d9-patches/archive/wine-d3d9-%{ninever}.tar.gz
 
 %if !%{?no64bit}
 ExclusiveArch:  %{ix86} x86_64 %{arm} aarch64
@@ -148,9 +177,15 @@ BuildRequires:  libattr-devel
 BuildRequires:  libva-devel
 %endif # 0%{?compholio}
 
+%if 0%{?nine}
+BuildRequires:  mesa-libd3d-devel
+%endif # 0%{?nine}
+
+
 %if 0%{?fedora} >= 10 || 0%{?rhel} >= 6
 BuildRequires:  openal-soft-devel
 BuildRequires:  icoutils
+BuildRequires:  librsvg2-tools
 %endif
 
 Requires:       wine-common = %{version}-%{release}
@@ -274,6 +309,9 @@ Requires:       unixODBC(x86-32)
 %if 0%{?compholio}
 Requires:       libva(x86-32)
 %endif
+%if 0%{?nine}
+Requires:       mesa-libd3d(x86-32)
+%endif
 %endif
 
 %ifarch x86_64
@@ -296,6 +334,9 @@ Requires:       unixODBC(x86-64)
 %if 0%{?compholio}
 Requires:       libva(x86-64)
 %endif
+%if 0%{?nine}
+Requires:       mesa-libd3d(x86-64)
+%endif
 %endif
 
 %ifarch %{arm} aarch64
@@ -315,6 +356,9 @@ Requires:       unixODBC
 %if 0%{?compholio}
 Requires:       libva
 %endif
+%if 0%{?nine}
+Requires:       mesa-libd3d
+%endif
 %endif
 
 # removed as of 1.7.35
@@ -660,8 +704,16 @@ This package adds the opencl driver for
 %endif
 
 %prep
-%setup -q -n wine-%{version}
+%setup -q -n wine-%{version}%{?rctagtarball}
 %patch511 -p1 -b.cjk
+%patch599 -p1
+%patch600 -p1
+%patch601 -p1
+%patch602 -p1 -R
+
+# extract nine patches
+mkdir nine
+gzip -dc %{SOURCE910} | tar -xf - --strip-components=1 -C nine
 
 # setup and apply wine-staging patches
 gzip -dc %{SOURCE900} | tar -xf - --strip-components=1
@@ -672,12 +724,33 @@ make -C patches DESTDIR="`pwd`" install
 # fix parallelized build
 sed -i -e 's!^loader server: libs/port libs/wine tools.*!& include!' Makefile.in
 
+%if 0%{?nine}
+patch -p1 -i nine/staging-helper.patch
+%endif # 0%{?nine}
+
 %else # 0%{?compholio}
 
 rm -rf patches/
 
+%if 0%{?nine}
+patch -p1 -i nine/d3d9-helper.patch
+%endif # 0%{?nine}
+
 %endif # 0%{?compholio}
 
+%if 0%{?nine}
+patch -p1 -i nine/wine-d3d9.patch
+
+autoreconf -ivf
+
+%endif # 0%{?nine}
+
+sed -i \
+  -e 's|-lncurses |-lncursesw |g' \
+  -e 's|"-lncurses"|"-lncursesw"|g' \
+  -e 's|OpenCL/opencl.h|CL/opencl.h|g' \
+  configure
+
 %build
 
 # disable fortify as it breaks wine
@@ -700,6 +773,7 @@ export CFLAGS="`echo $TEMP_CFLAGS | sed
  --enable-win64 \
 %endif
 %{?compholio: --with-xattr} \
+%{?nine: --with-d3d9-nine} \
  --disable-tests
 
 make %{?_smp_mflags} TARGETFLAGS=""
@@ -818,6 +892,28 @@ install -p -m 644 programs/wordpad/wordp
  %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/wordpad.svg
 sed -i -e "$PROGRAM_ICONFIX" %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/wordpad.svg
 
+install -p -m 644 programs/iexplore/iexplore.svg \
+ %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/iexplore.svg
+sed -i -e "$PROGRAM_ICONFIX" %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/iexplore.svg
+
+install -p -m 644 dlls/joy.cpl/joy.svg \
+  %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/joycpl.svg
+sed -i -e '3s/368/64/' %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/joycpl.svg
+
+install -p -m 644 programs/taskmgr/taskmgr.svg \
+ %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/taskmgr.svg
+sed -i -e "$PROGRAM_ICONFIX" %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/taskmgr.svg
+
+for file in %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/*.svg ;do
+  basefile=$(basename ${file} .svg)
+  for res in 16 22 24 32 36 48 64 72 96 128 192 256 512 ;do
+    dir=%{buildroot}%{_datadir}/icons/hicolor/${res}x${res}/apps
+    mkdir -p ${dir}
+    rsvg-convert -w ${res} -h ${res} ${file} \
+      -o ${dir}/${basefile}.png
+  done
+done
+
 %endif
 
 # install desktop files
@@ -863,6 +959,22 @@ desktop-file-install \
 
 desktop-file-install \
   --dir=%{buildroot}%{_datadir}/applications \
+  %{SOURCE110}
+
+desktop-file-install \
+  --dir=%{buildroot}%{_datadir}/applications \
+  %{SOURCE111}
+
+desktop-file-install \
+  --dir=%{buildroot}%{_datadir}/applications \
+  %{SOURCE112}
+
+desktop-file-install \
+  --dir=%{buildroot}%{_datadir}/applications \
+  %{SOURCE113}
+
+desktop-file-install \
+  --dir=%{buildroot}%{_datadir}/applications \
   --delete-original \
   %{buildroot}%{_datadir}/applications/wine.desktop
 
@@ -1836,6 +1948,10 @@ fi
 %endif
 %endif
 
+%if 0%{?nine}
+%{_libdir}/wine/d3d9-nine.dll.so
+%endif
+
 # 16 bit and other non 64bit stuff
 %ifnarch x86_64 %{arm} aarch64
 %{_libdir}/wine/winevdm.exe.so
@@ -2023,6 +2139,10 @@ fi
 %{_datadir}/fonts/wine-wingdings-fonts
 
 %files desktop
+%{_datadir}/applications/wine-iexplore.desktop
+%{_datadir}/applications/wine-inetcpl.desktop
+%{_datadir}/applications/wine-joycpl.desktop
+%{_datadir}/applications/wine-taskmgr.desktop
 %{_datadir}/applications/wine-notepad.desktop
 %{_datadir}/applications/wine-winefile.desktop
 %{_datadir}/applications/wine-winemine.desktop
@@ -2038,6 +2158,7 @@ fi
 %{_datadir}/desktop-directories/Wine.directory
 %config %{_sysconfdir}/xdg/menus/applications-merged/wine.menu
 %if 0%{?fedora} >= 10
+%{_datadir}/icons/hicolor/*/apps/*png
 %{_datadir}/icons/hicolor/scalable/apps/*svg
 %endif
 
@@ -2114,17 +2235,28 @@ fi
 %endif
 
 %changelog
-* Wed Jan 25 2017 Michael Cronenworth <mike@cchtml.com> 2.0-1
-- version update
+* Wed Jan 25 2017 Phantom X <megaphantomx at bol dot com dot br> - 2.0-100.chinfo
+- 2.0 final
 
-* Mon Jan 23 2017 Michael Cronenworth <mike@cchtml.com> 2.0-0.1.rc6
-- version update
+* Mon Jan 23 2017 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.7.rc7.chinfo
+- 2.0-rc6
 
-* Mon Jan 16 2017 Michael Cronenworth <mike@cchtml.com> 2.0-0.1.rc5
-- version update
+* Mon Jan 16 2017 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.6.rc5.chinfo
+- 2.0-rc5
 
-* Mon Jan 09 2017 Michael Cronenworth <mike@cchtml.com> 2.0-0.1.rc4
-- version update
+* Mon Jan 09 2017 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.5.rc4.chinfo
+- 2.0-rc4
+
+* Wed Dec 28 2016 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.4.rc3.nine
+- rebuilt
+
+* Wed Dec 28 2016 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.3.rc3.nine
+- Drop epoch.
+
+* Wed Dec 28 2016 Phantom X <megaphantomx at bol dot com dot br> - 2.0-0.2.rc3.nine
+- nine patches
+- extra patches
+- joy.cpl desktop file
 
 * Tue Dec 27 2016 Michael Cronenworth <mike@cchtml.com> 2.0-0.1.rc3
 - version update
