From 194e09baeca42b369b391d87c9605338b8fbc978 Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Wed, 29 Jun 2022 11:06:16 +0200
Subject: [PATCH] makefiles: Add separate variables for Unix flags and
 libraries.

Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 dlls/winegstreamer/Makefile.in     |  4 ++--
 35 files changed, 64 insertions(+), 59 deletions(-)

diff --git a/dlls/winegstreamer/Makefile.in b/dlls/winegstreamer/Makefile.in
index 462b54f2557..1cdad72cbcb 100644
--- a/dlls/winegstreamer/Makefile.in
+++ b/dlls/winegstreamer/Makefile.in
@@ -3,8 +3,8 @@ UNIXLIB   = winegstreamer.so
 IMPORTLIB = winegstreamer
 IMPORTS   = strmbase ole32 oleaut32 msdmo
 DELAYIMPORTS = mfplat
-EXTRAINCL = $(GSTREAMER_CFLAGS)
-EXTRALIBS = $(GSTREAMER_LIBS) $(PTHREAD_LIBS)
+UNIX_CFLAGS  = $(GSTREAMER_CFLAGS)
+UNIX_LIBS    = $(GSTREAMER_LIBS) $(PTHREAD_LIBS)
 
 C_SRCS = \
 	color_convert.c \
-- 
2.20.1

