diff --git a/patches/patchinstall.sh b/patches/patchinstall.sh
index 93484b40..a20bcaca 100755
--- a/patches/patchinstall.sh
+++ b/patches/patchinstall.sh
@@ -368,6 +368,7 @@ patch_enable_all ()
 	enable_wined3d_DXTn="$1"
 	enable_wined3d_Dual_Source_Blending="$1"
 	enable_wined3d_Indexed_Vertex_Blending="$1"
+	enable_wined3d_Persistent_Buffer_Allocator="$1"
 	enable_wined3d_QUERY_Stubs="$1"
 	enable_wined3d_Restore_DirectX10_Support="$1"
 	enable_wined3d_Silence_FIXMEs="$1"
@@ -1281,6 +1282,9 @@ patch_enable ()
 		wined3d-Indexed_Vertex_Blending)
 			enable_wined3d_Indexed_Vertex_Blending="$2"
 			;;
+        wined3d-Persistent_Buffer_Allocator)
+			enable_wined3d_Persistent_Buffer_Allocator="$2"
+			;;
 		wined3d-QUERY_Stubs)
 			enable_wined3d_QUERY_Stubs="$2"
 			;;
@@ -8409,6 +8413,32 @@ if test "$enable_xaudio2_CommitChanges" -eq 1; then
 	) >> "$patchlist"
 fi
 
+# Patchset  Persistent_Buffer_Allocator
+# |
+# | This patchset adds wine-PBA
+# |     https://github.com/acomminos/wine-pba
+# | with additional envvars patch
+# |     https://github.com/Firerat/wine-pba/tree/knobs_and_switches
+# | 
+# | Modified files:
+# |     * dlls/wined3d/Makefile.in, dlls/wined3d/buffer.c, dlls/wined3d/buffer_heap.c, dlls/wined3d/context.c, dlls/wined3d/cs.c, dlls/wined3d/device.c, dlls/wined3d/directx.c, dlls/wined3d/resource.c, dlls/wined3d/state.c, dlls/wined3d/texture.c, dlls/wined3d/utils.c, dlls/wined3d/wined3d_gl.h, dlls/wined3d/wined3d_private.h
+# |
+if test "$enable_wined3d_Persistent_Buffer_Allocator" -eq 1; then
+    patch_apply wined3d-Persistent_Buffer_Allocator/0001-wined3d-Initial-implementation-of-a-persistent-mappe.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0002-wined3d-Add-support-for-backing-dynamic-wined3d_buff.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0003-wined3d-Use-ARB_multi_bind-to-speed-up-UBO-updates.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0004-wined3d-Use-GL_CLIENT_STORAGE_BIT-for-persistent-map.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0005-wined3d-Disable-persistently-mapped-shader-resource-.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0006-wined3d-Perform-initial-allocation-of-persistent-buf.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0007-wined3d-Avoid-freeing-persistent-buffer-heap-element.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0008-wined3d-Add-DISABLE_PBA-envvar-some-PBA-cleanup.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0009-wined3d-Add-quirk-to-use-GL_CLIENT_STORAGE_BIT-for-m.patch
+    patch_apply wined3d-Persistent_Buffer_Allocator/0010-wined3d-knobs-and-switches.patch
+	(
+		printf '%s\n' '+    { "Andrew Comminos", "wined3d: Persistent Buffer Allocator", 1 },';
+		printf '%s\n' '+    { "Firerat", "wined3d: add envvars to Persistent Buffer Allocator", 1 },';
+	) >> "$patchlist"
+fi
 
 if test "$enable_patchlist" -eq 1; then
 
