--- _upstream/VirtualBox-kmod/VirtualBox-kmod-fdist/VirtualBox-kmod.spec	2017-01-18 19:17:26.883026684 -0200
+++ VirtualBox-kmod/VirtualBox-kmod.spec	2017-02-02 23:41:48.903560182 -0200
@@ -28,7 +28,7 @@
 Name:           VirtualBox-kmod
 Version:        5.1.14
 #Release:        1%%{?prerel:.%%{prerel}}%%{?dist}
-Release:        1%{?dist}
+Release:        1.chinfo%{?dist}
 
 Summary:        Kernel module for VirtualBox
 Group:          System Environment/Kernel
@@ -41,7 +41,7 @@ Source1:        VirtualBox-kmod-excludek
 BuildRequires:  %{AkmodsBuildRequires}
 
 # needed for plague to make sure it builds for i586 and i686
-ExclusiveArch:  i686 x86_64
+ExclusiveArch:  %{ix86} x86_64
 
 # get the proper build-sysbuild package from the repo, which
 # tracks in all the kernel-devel packages
@@ -106,8 +106,11 @@ DIRS=$(ls %{name}-%{version} |wc -l)
 
 
 %changelog
-* Wed Jan 18 2017 Sérgio Basto <sergio@serjux.com> - 5.1.14-1
-- Update VBox to 5.1.14
+* Wed Jan 18 2017 Phantom X <megaphantomx at bol dot com dot br> - 5.1.14-1
+- 5.1.14
+
+* Sat Jan 07 2017 Phantom X <megaphantomx at bol dot com dot br> - 5.1.12-1.chinfo
+- 5.1.12
 
 * Tue Nov 22 2016 Sérgio Basto <sergio@serjux.com> - 5.1.10-1
 - New upstream vesion, 5.1.10
